<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run the derfinder analysis on a chromosome — analyzeChr • derfinder</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Run the derfinder analysis on a chromosome — analyzeChr" />
<meta property="og:description" content="This is a major wrapper for running several key functions from this package.
It is meant to be used after loadCoverage has been used for a
specific chromosome. The steps run include makeModels,
preprocessCoverage, calculateStats, calculatePvalues
and annotating with annotateTranscripts and
matchGenes." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinder</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.23.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/derfinder-quickstart.html">derfinder quick start guide</a>
    </li>
    <li>
      <a href="../articles/derfinder-users-guide.html">derfinder users guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lcolladotor/derfinder/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run the derfinder analysis on a chromosome</h1>
    <small class="dont-index">Source: <a href='https://github.com/lcolladotor/derfinder/blob/master/R/analyzeChr.R'><code>R/analyzeChr.R</code></a></small>
    <div class="hidden name"><code>analyzeChr.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This is a major wrapper for running several key functions from this package.
It is meant to be used after <a href='loadCoverage.html'>loadCoverage</a> has been used for a
specific chromosome. The steps run include <a href='makeModels.html'>makeModels</a>,
<a href='preprocessCoverage.html'>preprocessCoverage</a>, <a href='calculateStats.html'>calculateStats</a>, <a href='calculatePvalues.html'>calculatePvalues</a>
and annotating with <a href='https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html'>annotateTranscripts</a> and
<a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a>.</p>
    </div>

    <pre class="usage"><span class='fu'>analyzeChr</span>(
  <span class='no'>chr</span>,
  <span class='no'>coverageInfo</span>,
  <span class='no'>models</span>,
  <span class='kw'>cutoffPre</span> <span class='kw'>=</span> <span class='fl'>5</span>,
  <span class='kw'>cutoffFstat</span> <span class='kw'>=</span> <span class='fl'>1e-08</span>,
  <span class='kw'>cutoffType</span> <span class='kw'>=</span> <span class='st'>"theoretical"</span>,
  <span class='kw'>nPermute</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>seeds</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/integer.html'>as.integer</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span>(<span class='st'>"-"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/Sys.time.html'>Sys.Date</a></span>())) + <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span>(<span class='no'>nPermute</span>),
  <span class='no'>groupInfo</span>,
  <span class='kw'>txdb</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>writeOutput</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>runAnnotation</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>lowMemDir</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span>(<span class='no'>chr</span>, <span class='st'>"chunksDir"</span>),
  <span class='kw'>smooth</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>weights</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>smoothFunction</span> <span class='kw'>=</span> <span class='kw pkg'>bumphunter</span><span class='kw ns'>::</span><span class='no'><a href='https://rdrr.io/pkg/bumphunter/man/locfitByCluster.html'>locfitByCluster</a></span>,
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>chr</th>
      <td><p>Used for naming the output files when <code>writeOutput=TRUE</code> and
the resulting <code>GRanges</code> object.</p></td>
    </tr>
    <tr>
      <th>coverageInfo</th>
      <td><p>A list containing a DataFrame --<code>$coverage</code>-- with
the coverage data and a logical Rle --<code>$position</code>-- with the positions
that passed the cutoff. This object is generated using <a href='loadCoverage.html'>loadCoverage</a>.
You should have specified a cutoff value for <a href='loadCoverage.html'>loadCoverage</a> unless
that you are using <code>colsubset</code> which will force a filtering step
with <a href='filterData.html'>filterData</a> when running <a href='preprocessCoverage.html'>preprocessCoverage</a>.</p></td>
    </tr>
    <tr>
      <th>models</th>
      <td><p>The output from <a href='makeModels.html'>makeModels</a>.</p></td>
    </tr>
    <tr>
      <th>cutoffPre</th>
      <td><p>This argument is passed to <a href='preprocessCoverage.html'>preprocessCoverage</a>
(<code>cutoff</code>).</p></td>
    </tr>
    <tr>
      <th>cutoffFstat</th>
      <td><p>This is used to determine the cutoff argument of
<a href='calculatePvalues.html'>calculatePvalues</a> and it's behaviour is determined by
<code>cutoffType</code>.</p></td>
    </tr>
    <tr>
      <th>cutoffType</th>
      <td><p>If set to <code>empirical</code>, the <code>cutoffFstat</code>
(example: 0.99) quantile is used via <a href='https://rdrr.io/r/stats/quantile.html'>quantile</a>. If set to
<code>theoretical</code>, the theoretical <code>cutoffFstats</code> (example: 1e-08) is
calculated via <a href='https://rdrr.io/r/stats/Fdist.html'>qf</a>. If set to <code>manual</code>, <code>cutoffFstats</code> is
passed to <a href='calculatePvalues.html'>calculatePvalues</a> without any other calculation.</p></td>
    </tr>
    <tr>
      <th>nPermute</th>
      <td><p>The number of permutations. Note that for a full chromosome,
a small amount (10) of permutations is sufficient. If set to 0, no
permutations are performed and thus no null regions are used, however, the
<code>$regions</code> component is created.</p></td>
    </tr>
    <tr>
      <th>seeds</th>
      <td><p>An integer vector of length <code>nPermute</code> specifying the
seeds to be used for each permutation. If <code>NULL</code> no seeds are used.</p></td>
    </tr>
    <tr>
      <th>groupInfo</th>
      <td><p>A factor specifying the group membership of each sample
that can later be used with the plotting functions in the
<code>derfinderPlot</code> package.</p></td>
    </tr>
    <tr>
      <th>txdb</th>
      <td><p>This argument is passed to
<a href='https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html'>annotateTranscripts</a>. If <code>NULL</code>,
TxDb.Hsapiens.UCSC.hg19.knownGene
is used.</p></td>
    </tr>
    <tr>
      <th>writeOutput</th>
      <td><p>If <code>TRUE</code>, output Rdata files are created at each
step inside a directory with the chromosome name (example: 'chr21' if
<code>chrnum='21'</code>). One Rdata file is created for each component described
in the return section.</p></td>
    </tr>
    <tr>
      <th>runAnnotation</th>
      <td><p>If <code>TRUE</code> <a href='https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html'>annotateTranscripts</a>
and <a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a> are run. Otherwise these steps are skipped.</p></td>
    </tr>
    <tr>
      <th>lowMemDir</th>
      <td><p>If specified, each chunk is saved into a separate Rdata
file under <code>lowMemDir</code> and later loaded in
<a href='https://rdrr.io/pkg/derfinderHelper/man/fstats.apply.html'>fstats.apply</a> when
running <a href='calculateStats.html'>calculateStats</a> and <a href='calculatePvalues.html'>calculatePvalues</a>. Using this option
helps reduce the memory load as each fork in <a href='https://rdrr.io/pkg/BiocParallel/man/bplapply.html'>bplapply</a>
loads only the data needed for the chunk processing. The downside is a bit
longer computation time due to input/output.</p></td>
    </tr>
    <tr>
      <th>smooth</th>
      <td><p>Whether to smooth the F-statistics (<code>fstats</code>) or not. This
is by default <code>FALSE</code>. For RNA-seq data we recommend using <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>weights</th>
      <td><p>Weights used by the smoother as described in
<a href='https://rdrr.io/pkg/bumphunter/man/smoother.html'>smoother</a>.</p></td>
    </tr>
    <tr>
      <th>smoothFunction</th>
      <td><p>A function to be used for smoothing the F-statistics.
Two functions are provided by the <code>bumphunter</code> package:
<a href='https://rdrr.io/pkg/bumphunter/man/loessByCluster.html'>loessByCluster</a> and <a href='https://rdrr.io/pkg/bumphunter/man/runmedByCluster.html'>runmedByCluster</a>. If
you are using your own custom function, it has to return a named list with
an element called <code>$fitted</code> that contains the smoothed F-statistics and
an element claled <code>$smoothed</code> that is a logical vector indicating
whether the F-statistics were smoothed or not. If they are not smoothed, the
original values will be used.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to other methods and/or advanced arguments.
Advanced arguments:</p><dl>
<dt>verbose </dt><dd><p>If <code>TRUE</code> basic status updates will be printed along
the way. Default <code>TRUE</code>.</p></dd>
<dt>scalefac </dt><dd><p>This argument is passed to <a href='preprocessCoverage.html'>preprocessCoverage</a>.</p></dd>
<dt>chunksize </dt><dd><p>This argument is passed to <a href='preprocessCoverage.html'>preprocessCoverage</a>.</p></dd>
<dt>returnOutput </dt><dd><p>If <code>TRUE</code>, it returns a list with the results
from each step. Otherwise, it returns <code>NULL</code>. Default: the opposite of
<code>writeOutput</code>.</p></dd>

</dl><p>Passed to <a href='extendedMapSeqlevels.html'>extendedMapSeqlevels</a>, <a href='preprocessCoverage.html'>preprocessCoverage</a>,
<a href='calculateStats.html'>calculateStats</a>, <a href='calculatePvalues.html'>calculatePvalues</a>,
<a href='https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html'>annotateTranscripts</a>, <a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a>,
and <a href='define_cluster.html'>define_cluster</a>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>If <code>returnOutput=TRUE</code>, a list with six components:</p><dl>
<dt>timeinfo </dt><dd><p>The wallclock timing information for each step.</p></dd>
<dt>optionsStats </dt><dd><p>The main options used when running this function.</p></dd>
<dt>coveragePrep </dt><dd><p>The output from <a href='preprocessCoverage.html'>preprocessCoverage</a>.</p></dd>
<dt>fstats</dt><dd><p>The output from <a href='calculateStats.html'>calculateStats</a>.</p></dd>
<dt>regions</dt><dd><p>The output from <a href='calculatePvalues.html'>calculatePvalues</a>.</p></dd>
<dt>annotation</dt><dd><p>The output from <a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a>.</p></dd>

</dl><p>These are the same components that are written to Rdata files if
<code>writeOutput=TRUE</code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>If you are working with data from an organism different from 'Homo sapiens'
specify so by setting the global 'species' and 'chrsStyle' options. For
example:
<code><a href='https://rdrr.io/r/base/options.html'>options(species = 'arabidopsis_thaliana')</a></code>
<code><a href='https://rdrr.io/r/base/options.html'>options(chrsStyle = 'NCBI')</a></code></p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='makeModels.html'>makeModels</a>, <a href='preprocessCoverage.html'>preprocessCoverage</a>,
<a href='calculateStats.html'>calculateStats</a>, <a href='calculatePvalues.html'>calculatePvalues</a>,
<a href='https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html'>annotateTranscripts</a>, <a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Collapse the coverage information</span>
<span class='no'>collapsedFull</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='collapseFullCoverage.html'>collapseFullCoverage</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='no'>genomeData</span>$<span class='no'>coverage</span>),
    <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)</div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:56 collapseFullCoverage: Sorting fullCov</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:56 collapseFullCoverage: Collapsing chromosomes information by sample</span></div><div class='input'>
<span class='co'>## Calculate library size adjustments</span>
<span class='no'>sampleDepths</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='sampleDepth.html'>sampleDepth</a></span>(<span class='no'>collapsedFull</span>,
    <span class='kw'>probs</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>0.5</span>), <span class='kw'>nonzero</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
    <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)</div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:56 sampleDepth: Calculating sample quantiles</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:56 sampleDepth: Calculating sample adjustments</span></div><div class='input'>
<span class='co'>## Build the models</span>
<span class='no'>groupInfo</span> <span class='kw'>&lt;-</span> <span class='no'>genomeInfo</span>$<span class='no'>pop</span>
<span class='no'>adjustvars</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span>(<span class='no'>genomeInfo</span>$<span class='no'>gender</span>)
<span class='no'>models</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='makeModels.html'>makeModels</a></span>(<span class='no'>sampleDepths</span>, <span class='kw'>testvars</span> <span class='kw'>=</span> <span class='no'>groupInfo</span>, <span class='kw'>adjustvars</span> <span class='kw'>=</span> <span class='no'>adjustvars</span>)

<span class='co'>## Analyze the chromosome</span>
<span class='no'>results</span> <span class='kw'>&lt;-</span> <span class='fu'>analyzeChr</span>(
    <span class='kw'>chr</span> <span class='kw'>=</span> <span class='st'>"21"</span>, <span class='kw'>coverageInfo</span> <span class='kw'>=</span> <span class='no'>genomeData</span>, <span class='kw'>models</span> <span class='kw'>=</span> <span class='no'>models</span>,
    <span class='kw'>cutoffFstat</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>cutoffType</span> <span class='kw'>=</span> <span class='st'>"manual"</span>, <span class='kw'>groupInfo</span> <span class='kw'>=</span> <span class='no'>groupInfo</span>, <span class='kw'>mc.cores</span> <span class='kw'>=</span> <span class='fl'>1</span>,
    <span class='kw'>writeOutput</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>returnOutput</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"regular"</span>,
    <span class='kw'>runAnnotation</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>
)</div><div class='output co'>#&gt; <span class='message'>extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:56 analyzeChr: Pre-processing the coverage data</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 analyzeChr: Calculating statistics</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 calculateStats: calculating the F-statistics</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 analyzeChr: Calculating pvalues</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 analyzeChr: Using the following manual cutoff for the F-statistics 1</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 calculatePvalues: identifying data segments</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 findRegions: segmenting information</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 findRegions: identifying candidate regions</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 findRegions: identifying region clusters</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 calculatePvalues: calculating F-statistics for permutation 1 and seed 20200528</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 findRegions: segmenting information</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:57 findRegions: identifying candidate regions</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:58 calculatePvalues: calculating the p-values</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:58 calculatePvalues: skipping q-value calculation.</span></div><div class='output co'>#&gt; <span class='message'>2020-05-27 05:46:58 analyzeChr: Annotating regions</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>results</span>)</div><div class='output co'>#&gt; [1] "timeinfo"     "optionsStats" "coveragePrep" "fstats"       "regions"     
#&gt; [6] "annotation"  </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


