<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Find non-zero regions in a Rle — findRegions • derfinder</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Find non-zero regions in a Rle — findRegions" />
<meta property="og:description" content="Find genomic regions for which a numeric vector is above (or below)
predefined thresholds. In other words, this function finds the candidate
Differentially Expressed Regions (candidate DERs). This is similar to
regionFinder and is a helper function for
calculatePvalues." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinder</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.23.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/derfinder-quickstart.html">derfinder quick start guide</a>
    </li>
    <li>
      <a href="../articles/derfinder-users-guide.html">derfinder users guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lcolladotor/derfinder/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Find non-zero regions in a Rle</h1>
    <small class="dont-index">Source: <a href='https://github.com/lcolladotor/derfinder/blob/master/R/findRegions.R'><code>R/findRegions.R</code></a></small>
    <div class="hidden name"><code>findRegions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Find genomic regions for which a numeric vector is above (or below)
predefined thresholds. In other words, this function finds the candidate
Differentially Expressed Regions (candidate DERs). This is similar to
<a href='https://rdrr.io/pkg/bumphunter/man/regionFinder.html'>regionFinder</a> and is a helper function for
<a href='calculatePvalues.html'>calculatePvalues</a>.</p>
    </div>

    <pre class="usage"><span class='fu'>findRegions</span>(
  <span class='kw'>position</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='no'>fstats</span>,
  <span class='no'>chr</span>,
  <span class='kw'>oneTable</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>maxClusterGap</span> <span class='kw'>=</span> <span class='fl'>300L</span>,
  <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/quantile.html'>quantile</a></span>(<span class='no'>fstats</span>, <span class='fl'>0.99</span>, <span class='kw'>na.rm</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>),
  <span class='kw'>segmentIR</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>smooth</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>weights</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>smoothFunction</span> <span class='kw'>=</span> <span class='kw pkg'>bumphunter</span><span class='kw ns'>::</span><span class='no'><a href='https://rdrr.io/pkg/bumphunter/man/locfitByCluster.html'>locfitByCluster</a></span>,
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>position</th>
      <td><p>A logical Rle of genomic positions. This is generated in
<a href='loadCoverage.html'>loadCoverage</a>. Note that it gets updated in <a href='preprocessCoverage.html'>preprocessCoverage</a>
if <code>colsubset</code> is not <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>fstats</th>
      <td><p>A numeric Rle with the F-statistics. Usually obtained using
<a href='calculateStats.html'>calculateStats</a>.</p></td>
    </tr>
    <tr>
      <th>chr</th>
      <td><p>A single element character vector specifying the chromosome name.</p></td>
    </tr>
    <tr>
      <th>oneTable</th>
      <td><p>If <code>TRUE</code> only one GRanges is returned.
Otherwise, a GRangesList with two components is returned: one for the
regions with positive values and one for the negative values.</p></td>
    </tr>
    <tr>
      <th>maxClusterGap</th>
      <td><p>This determines the maximum gap between candidate DERs.
It should be greater than <code>maxRegionGap</code> (0 by default).</p></td>
    </tr>
    <tr>
      <th>cutoff</th>
      <td><p>Threshold applied to the <code>fstats</code> used to determine the
regions.</p></td>
    </tr>
    <tr>
      <th>segmentIR</th>
      <td><p>An IRanges object with the genomic positions that are
potentials DERs. This is used in <a href='calculatePvalues.html'>calculatePvalues</a> to speed up
permutation calculations.</p></td>
    </tr>
    <tr>
      <th>smooth</th>
      <td><p>Whether to smooth the F-statistics (<code>fstats</code>) or not. This
is by default <code>FALSE</code>. For RNA-seq data we recommend using <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>weights</th>
      <td><p>Weights used by the smoother as described in
<a href='https://rdrr.io/pkg/bumphunter/man/smoother.html'>smoother</a>.</p></td>
    </tr>
    <tr>
      <th>smoothFunction</th>
      <td><p>A function to be used for smoothing the F-statistics.
Two functions are provided by the <code>bumphunter</code> package:
<a href='https://rdrr.io/pkg/bumphunter/man/loessByCluster.html'>loessByCluster</a> and <a href='https://rdrr.io/pkg/bumphunter/man/runmedByCluster.html'>runmedByCluster</a>. If
you are using your own custom function, it has to return a named list with
an element called <code>$fitted</code> that contains the smoothed F-statistics and
an element claled <code>$smoothed</code> that is a logical vector indicating
whether the F-statistics were smoothed or not. If they are not smoothed, the
original values will be used.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to other methods and/or advanced arguments.
Advanced arguments:</p><dl>
<dt>verbose </dt><dd><p>If <code>TRUE</code> basic status updates will be printed along
the way.</p></dd>
<dt>basic </dt><dd><p>If <code>TRUE</code> a DataFrame is returned that has only basic
information on the candidate DERs. This is used in <a href='calculatePvalues.html'>calculatePvalues</a>
to speed up permutation calculations. Default: <code>FALSE</code>.</p></dd>
<dt>maxRegionGap </dt><dd><p>This determines the maximum number of gaps between two
genomic positions to be considered part of the same candidate region. The
default is 0L.</p></dd>

</dl><p>Passed to <a href='extendedMapSeqlevels.html'>extendedMapSeqlevels</a> and the internal function
<code>.getSegmentsRle</code> that has by default <code>verbose = FALSE</code>.</p>
<p>When <code>smooth = TRUE</code>, <code>...</code> is passed to the internal function
<code>.smootherFstats</code>. This internal function has the advanced argument
<code>maxClusterGap</code> (same as above) and passes <code>...</code> to
<a href='define_cluster.html'>define_cluster</a> and the formal arguments of <code>smoothFun</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Either a GRanges or a GRangesList as determined by <code>oneTable</code>.
Each of them has the following metadata variables.</p><dl>
<dt>value </dt><dd><p>The mean of the values of <code>y</code> for the given region.</p></dd>
<dt>area </dt><dd><p>The absolute value of the sum of the values of <code>y</code> for
the given region.</p></dd>
<dt>indexStart </dt><dd><p>The start position of the region in terms of the index
for <code>y</code>.</p></dd>
<dt>indexEnd </dt><dd><p>The end position of the region in terms of the index for
<code>y</code>.</p></dd>
<dt>cluster </dt><dd><p>The cluser ID.</p></dd>
<dt>clusterL </dt><dd><p>The total length of the cluster.</p></dd>

</dl>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p><a href='https://rdrr.io/pkg/bumphunter/man/regionFinder.html'>regionFinder</a> adapted to Rle world.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Rafael A. Irizarry, Martin Aryee, Hector Corrada Bravo, Kasper
D. Hansen and Harris A. Jaffee. bumphunter: Bump Hunter. R package version
1.1.10.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='calculatePvalues.html'>calculatePvalues</a></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Preprocess the data</span>
<span class='no'>prep</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='preprocessCoverage.html'>preprocessCoverage</a></span>(<span class='no'>genomeData</span>,
    <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='fl'>0</span>, <span class='kw'>scalefac</span> <span class='kw'>=</span> <span class='fl'>32</span>, <span class='kw'>chunksize</span> <span class='kw'>=</span> <span class='fl'>1e3</span>,
    <span class='kw'>colsubset</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
)

<span class='co'>## Get the F statistics</span>
<span class='no'>fstats</span> <span class='kw'>&lt;-</span> <span class='no'>genomeFstats</span>

<span class='co'>## Find the regions</span>
<span class='no'>regs</span> <span class='kw'>&lt;-</span> <span class='fu'>findRegions</span>(<span class='no'>prep</span>$<span class='no'>position</span>, <span class='no'>fstats</span>, <span class='st'>"chr21"</span>, <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:06 findRegions: identifying potential segments</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:06 findRegions: segmenting information</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:06 .getSegmentsRle: segmenting with cutoff(s) 19.7936614060235</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:06 findRegions: identifying candidate regions</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:06 findRegions: identifying region clusters</span></div><div class='input'><span class='no'>regs</span></div><div class='output co'>#&gt; GRanges object with 3 ranges and 6 metadata columns:
#&gt;      seqnames            ranges strand |     value      area indexStart
#&gt;         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt;
#&gt;   up    chr21 47409011-47409016      * |   20.2182  121.3091         82
#&gt;   up    chr21          47409051      * |   20.3485   20.3485        122
#&gt;   up    chr21 47409522-47409533      * |   21.6702  260.0421        128
#&gt;       indexEnd cluster clusterL
#&gt;      &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt;
#&gt;   up        87       1       41
#&gt;   up       122       1       41
#&gt;   up       139       2       12
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'><span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>## Once you have the regions you can proceed to annotate them</span>
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"bumphunter"</span>)
<span class='no'>genes</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html'>annotateTranscripts</a></span>(<span class='kw pkg'>TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class='kw ns'>::</span><span class='no'><a href='https://rdrr.io/pkg/TxDb.Hsapiens.UCSC.hg19.knownGene/man/package.html'>TxDb.Hsapiens.UCSC.hg19.knownGene</a></span>)
<span class='no'>annotation</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a></span>(<span class='no'>regs</span>, <span class='no'>genes</span>)
<span class='no'>annotation</span>
}

<span class='co'># Find regions with smoothing the F-statistics by bumphunter::runmedByCluster</span>
<span class='no'>regs_smooth</span> <span class='kw'>&lt;-</span> <span class='fu'>findRegions</span>(<span class='no'>prep</span>$<span class='no'>position</span>, <span class='no'>fstats</span>, <span class='st'>"chr21"</span>,
    <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
    <span class='kw'>smoothFunction</span> <span class='kw'>=</span> <span class='kw pkg'>bumphunter</span><span class='kw ns'>::</span><span class='no'><a href='https://rdrr.io/pkg/bumphunter/man/runmedByCluster.html'>runmedByCluster</a></span>
)</div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:06 findRegions: identifying potential segments</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:06 findRegions: segmenting information</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:06 .getSegmentsRle: segmenting with cutoff(s) 19.7936614060235</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:06 findRegions: identifying candidate regions</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:06 findRegions: identifying region clusters</span></div><div class='input'><span class='co'>## Compare against the object regs obtained earlier</span>
<span class='no'>regs_smooth</span></div><div class='output co'>#&gt; GRanges object with 3 ranges and 6 metadata columns:
#&gt;      seqnames            ranges strand |     value      area indexStart
#&gt;         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt;
#&gt;   up    chr21 47409011-47409016      * |   20.2182  121.3091         82
#&gt;   up    chr21          47409051      * |   20.3485   20.3485        122
#&gt;   up    chr21 47409522-47409533      * |   21.6702  260.0421        128
#&gt;       indexEnd cluster clusterL
#&gt;      &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt;
#&gt;   up        87       1       41
#&gt;   up       122       1       41
#&gt;   up       139       2       12
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


