<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Load the unfiltered coverage information from a group of BAM files and a
list of chromosomes — fullCoverage • derfinder</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Load the unfiltered coverage information from a group of BAM files and a
list of chromosomes — fullCoverage" />
<meta property="og:description" content="For a group of samples this function reads the coverage information for
several chromosomes directly from the BAM files. Per chromosome, it merges
the unfiltered coverage by sample into a DataFrame. The end result is a list
with one such DataFrame objects per chromosome." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinder</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.21.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/derfinder-quickstart.html">derfinder quick start guide</a>
    </li>
    <li>
      <a href="../articles/derfinder-users-guide.html">derfinder users guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lcolladotor/derfinder/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Load the unfiltered coverage information from a group of BAM files and a
list of chromosomes</h1>
    <small class="dont-index">Source: <a href='https://github.com/lcolladotor/derfinder/blob/master/R/fullCoverage.R'><code>R/fullCoverage.R</code></a></small>
    <div class="hidden name"><code>fullCoverage.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For a group of samples this function reads the coverage information for
several chromosomes directly from the BAM files. Per chromosome, it merges
the unfiltered coverage by sample into a DataFrame. The end result is a list
with one such DataFrame objects per chromosome.</p>
    </div>

    <pre class="usage"><span class='fu'>fullCoverage</span>(
  <span class='no'>files</span>,
  <span class='no'>chrs</span>,
  <span class='kw'>bai</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>chrlens</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>outputs</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>files</th>
      <td><p>A character vector with the full path to the sample BAM files
(or BigWig files).
The names are used for the column names of the DataFrame. Check
<a href='rawFiles.html'>rawFiles</a> for constructing <code>files</code>. <code>files</code> can also be a
<code>BamFileList</code> object created with <a href='https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html'>BamFileList</a> or a
<code>BigWigFileList</code> object created with <a href='https://rdrr.io/pkg/rtracklayer/man/BigWigFile.html'>BigWigFileList</a>.</p></td>
    </tr>
    <tr>
      <th>chrs</th>
      <td><p>The chromosome of the files to read. The format has to match the
one used in the input files.</p></td>
    </tr>
    <tr>
      <th>bai</th>
      <td><p>The full path to the BAM index files. If <code>NULL</code> it is
assumed that the BAM index files are in the same location as the BAM files
and that they have the .bai extension. Ignored if <code>files</code> is a
<code>BamFileList</code> object.</p></td>
    </tr>
    <tr>
      <th>chrlens</th>
      <td><p>The chromosome lengths in base pairs. If it's <code>NULL</code>,
the chromosome length is extracted from the BAM files. Otherwise, it should
have the same length as <code>chrs</code>.</p></td>
    </tr>
    <tr>
      <th>outputs</th>
      <td><p>This argument is passed to the <code>output</code> argument of
<a href='loadCoverage.html'>loadCoverage</a>. If <code>NULL</code> or <code>'auto'</code> it is then recycled.</p></td>
    </tr>
    <tr>
      <th>cutoff</th>
      <td><p>This argument is passed to <a href='filterData.html'>filterData</a>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to other methods and/or advanced arguments.
Advanced arguments:</p><dl>
<dt>verbose </dt><dd><p>If <code>TRUE</code> basic status updates will be printed along
the way.</p></dd>
<dt>mc.cores </dt><dd><p>How many cores to use for reading the chromosome
information. There's no benefit of using a number greater than the number
of chromosomes. Also note that your harddisk speed will limit how much
you get from using a higher <code>mc.cores</code> value.</p></dd>
<dt>mc.cores.load </dt><dd><p>Controls the number of cores to be used per chr for
loading the data which is only useful in the scenario that you are loading
in genome tiles. If not supplied, it uses <code>mc.cores</code> for
<a href='loadCoverage.html'>loadCoverage</a>. Default: 1. If you are using genome tiles, the total
number of cores you'll use will be <code>mc.cores</code> times
<code>mc.cores.load</code>.</p></dd>

</dl><p>Passed to <a href='loadCoverage.html'>loadCoverage</a>, <a href='define_cluster.html'>define_cluster</a> and
<a href='extendedMapSeqlevels.html'>extendedMapSeqlevels</a>.
Note that <a href='filterData.html'>filterData</a> is used internally
by <a href='loadCoverage.html'>loadCoverage</a> (and hence fullCoverage) and has the important
arguments <code>totalMapped</code> and <code>targetSize</code> which can be used to
normalize the coverage by library size. See <a href='getTotalMapped.html'>getTotalMapped</a> for
calculating <code>totalMapped</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list with one element per chromosome.</p><dl>
produced by loadCoverage.

</dl>

    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='loadCoverage.html'>loadCoverage</a>, <a href='filterData.html'>filterData</a>, <a href='getTotalMapped.html'>getTotalMapped</a></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>datadir</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"genomeData"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"derfinder"</span>)
<span class='no'>files</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='rawFiles.html'>rawFiles</a></span>(
    <span class='kw'>datadir</span> <span class='kw'>=</span> <span class='no'>datadir</span>, <span class='kw'>samplepatt</span> <span class='kw'>=</span> <span class='st'>"*accepted_hits.bam$"</span>,
    <span class='kw'>fileterm</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
)
<span class='co'>## Shorten the column names</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>files</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span>(<span class='st'>"_accepted_hits.bam"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>files</span>))

<span class='co'>## Read and filter the data, only for 1 file</span>
<span class='no'>fullCov</span> <span class='kw'>&lt;-</span> <span class='fu'>fullCoverage</span>(<span class='kw'>files</span> <span class='kw'>=</span> <span class='no'>files</span>[<span class='fl'>1</span>], <span class='kw'>chrs</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"21"</span>, <span class='st'>"22"</span>))</div><div class='output co'>#&gt; <span class='message'>2020-04-27 15:00:57 fullCoverage: processing chromosome 21</span></div><div class='output co'>#&gt; <span class='message'>2020-04-27 15:00:57 loadCoverage: finding chromosome lengths</span></div><div class='output co'>#&gt; <span class='message'>2020-04-27 15:00:57 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009101_accepted_hits.bam</span></div><div class='output co'>#&gt; <span class='message'>2020-04-27 15:00:57 loadCoverage: applying the cutoff to the merged data</span></div><div class='output co'>#&gt; <span class='message'>2020-04-27 15:00:57 filterData: originally there were 48129895 rows, now there are 48129895 rows. Meaning that 0 percent was filtered.</span></div><div class='output co'>#&gt; <span class='message'>2020-04-27 15:00:57 fullCoverage: processing chromosome 22</span></div><div class='output co'>#&gt; <span class='message'>2020-04-27 15:00:57 loadCoverage: finding chromosome lengths</span></div><div class='output co'>#&gt; <span class='message'>2020-04-27 15:00:57 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009101_accepted_hits.bam</span></div><div class='output co'>#&gt; <span class='message'>2020-04-27 15:00:57 loadCoverage: applying the cutoff to the merged data</span></div><div class='output co'>#&gt; <span class='message'>2020-04-27 15:00:57 filterData: originally there were 51304566 rows, now there are 51304566 rows. Meaning that 0 percent was filtered.</span></div><div class='output co'>#&gt; <span class='message'>extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span></div><div class='input'><span class='no'>fullCov</span></div><div class='output co'>#&gt; $chr21
#&gt; DataFrame with 48129895 rows and 1 column
#&gt;          ERR009101
#&gt;              &lt;Rle&gt;
#&gt; 1                0
#&gt; 2                0
#&gt; 3                0
#&gt; 4                0
#&gt; 5                0
#&gt; ...            ...
#&gt; 48129891         0
#&gt; 48129892         0
#&gt; 48129893         0
#&gt; 48129894         0
#&gt; 48129895         0
#&gt; 
#&gt; $chr22
#&gt; DataFrame with 51304566 rows and 1 column
#&gt;          ERR009101
#&gt;              &lt;Rle&gt;
#&gt; 1                0
#&gt; 2                0
#&gt; 3                0
#&gt; 4                0
#&gt; 5                0
#&gt; ...            ...
#&gt; 51304562         0
#&gt; 51304563         0
#&gt; 51304564         0
#&gt; 51304565         0
#&gt; 51304566         0
#&gt; </div><div class='input'><span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>## You can then use filterData() to filter the data if you want to.</span>
<span class='co'>## Use bplapply() if you want to do so with multiple cores as shown below.</span>
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"BiocParallel"</span>)
<span class='no'>p</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/SnowParam-class.html'>SnowParam</a></span>(<span class='fl'>2L</span>)
<span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/bplapply.html'>bplapply</a></span>(<span class='no'>fullCov</span>, <span class='kw'>function</span>(<span class='no'>x</span>) {
    <span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"derfinder"</span>)
    <span class='fu'><a href='filterData.html'>filterData</a></span>(<span class='no'>x</span>, <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='fl'>0</span>)
}, <span class='kw'>BPPARAM</span> <span class='kw'>=</span> <span class='no'>p</span>)
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


