<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Extract coverage information for a set of regions — getRegionCoverage • derfinder</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Extract coverage information for a set of regions — getRegionCoverage" />
<meta property="og:description" content="This function extracts the raw coverage information calculated by
fullCoverage at each base for a set of regions found with
calculatePvalues. It can further calculate the mean coverage per
sample for each region." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinder</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.21.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/derfinder-quickstart.html">derfinder quick start guide</a>
    </li>
    <li>
      <a href="../articles/derfinder-users-guide.html">derfinder users guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lcolladotor/derfinder/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Extract coverage information for a set of regions</h1>
    <small class="dont-index">Source: <a href='https://github.com/lcolladotor/derfinder/blob/master/R/getRegionCoverage.R'><code>R/getRegionCoverage.R</code></a></small>
    <div class="hidden name"><code>getRegionCoverage.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function extracts the raw coverage information calculated by
<a href='fullCoverage.html'>fullCoverage</a> at each base for a set of regions found with
<a href='calculatePvalues.html'>calculatePvalues</a>. It can further calculate the mean coverage per
sample for each region.</p>
    </div>

    <pre class="usage"><span class='fu'>getRegionCoverage</span>(
  <span class='kw'>fullCov</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='no'>regions</span>,
  <span class='kw'>totalMapped</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>targetSize</span> <span class='kw'>=</span> <span class='fl'>8e+07</span>,
  <span class='kw'>files</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>fullCov</th>
      <td><p>A list where each element is the result from
<a href='loadCoverage.html'>loadCoverage</a> used with <code>returnCoverage = TRUE</code>. Can be generated
using <a href='fullCoverage.html'>fullCoverage</a>. Alternatively, specify <code>files</code> to extract
the coverage information from the regions of interest. This can be
helpful if you do not wish to store <code>fullCov</code> for memory reasons.</p></td>
    </tr>
    <tr>
      <th>regions</th>
      <td><p>The <code>$regions</code> output from <a href='calculatePvalues.html'>calculatePvalues</a>. It
is important that the seqlengths information is provided.</p></td>
    </tr>
    <tr>
      <th>totalMapped</th>
      <td><p>The total number of reads mapped for each sample.
Providing this data adjusts the coverage to reads in <code>targetSize</code>
library. By default, to reads per 80 million reads.</p></td>
    </tr>
    <tr>
      <th>targetSize</th>
      <td><p>The target library size to adjust the coverage to. Used
only when <code>totalMapped</code> is specified.</p></td>
    </tr>
    <tr>
      <th>files</th>
      <td><p>A character vector with the full path to the sample BAM files
(or BigWig files).
The names are used for the column names of the DataFrame. Check
<a href='rawFiles.html'>rawFiles</a> for constructing <code>files</code>. <code>files</code> can also be a
<code>BamFileList</code> object created with <a href='https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html'>BamFileList</a> or a
<code>BigWigFileList</code> object created with <a href='https://rdrr.io/pkg/rtracklayer/man/BigWigFile.html'>BigWigFileList</a>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to other methods and/or advanced arguments.
Advanced arguments:</p><dl'>
<dt>verbose </dt><dd><p>If <code>TRUE</code> basic status updates will be printed along
the way.</p></dd>

</dl><p>Passed to <a href='extendedMapSeqlevels.html'>extendedMapSeqlevels</a> and <a href='define_cluster.html'>define_cluster</a>.</p>
<p>When <code>fullCov</code> is <code>NULL</code>, <code>...</code> has the advanced argument
<code>protectWhich</code> (default 30000) from <a href='loadCoverage.html'>loadCoverage</a>. Also
<code>...</code> is passed to <a href='fullCoverage.html'>fullCoverage</a> for loading the data on the fly.
This can be useful for loading the data from a specific region (or small
sets of regions) without having to load in memory the output the coverage
information from all the genome.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a list of data.frame where each data.frame has the coverage
information (nrow = width of region, ncol = number of samples) for a given
region. The names of the list correspond to the region indexes in
<code>regions</code></p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>When <code>fullCov</code> is the output of <a href='loadCoverage.html'>loadCoverage</a> with
<code>cutoff</code> non-NULL, getRegionCoverage assumes that the regions
come from the same data. Meaning that <a href='filterData.html'>filterData</a> was not used again.
This ensures that the regions are a subset of the data available in
<code>fullCov</code>.</p>
<p>If <code>fullCov</code> is <code>NULL</code> and <code>files</code> is specified, this function
will attempt to read the coverage from the files. Note that if you used
'totalMapped' and 'targetSize' before, you will have to specify them again
to get the same results.</p>
<p>You should use at most one core per chromosome.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='fullCoverage.html'>fullCoverage</a>, <a href='calculatePvalues.html'>calculatePvalues</a></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Obtain fullCov object</span>
<span class='no'>fullCov</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='st'>"21"</span> <span class='kw'>=</span> <span class='no'>genomeDataRaw</span>$<span class='no'>coverage</span>)

<span class='co'>## Assign chr lengths using hg19 information, use only first two regions</span>
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"GenomicRanges"</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='input'><span class='no'>regions</span> <span class='kw'>&lt;-</span> <span class='no'>genomeRegions</span>$<span class='no'>regions</span>[<span class='fl'>1</span>:<span class='fl'>2</span>]
<span class='fu'>seqlengths</span>(<span class='no'>regions</span>) <span class='kw'>&lt;-</span> <span class='fu'>seqlengths</span>(<span class='fu'>getChromInfoFromUCSC</span>(<span class='st'>"hg19"</span>,
    <span class='kw'>as.Seqinfo</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
))[
    <span class='fu'>mapSeqlevels</span>(<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='fu'>seqlengths</span>(<span class='no'>regions</span>)), <span class='st'>"UCSC"</span>)
]

<span class='co'>## Finally, get the region coverage</span>
<span class='no'>regionCov</span> <span class='kw'>&lt;-</span> <span class='fu'>getRegionCoverage</span>(<span class='kw'>fullCov</span> <span class='kw'>=</span> <span class='no'>fullCov</span>, <span class='kw'>regions</span> <span class='kw'>=</span> <span class='no'>regions</span>)</div><div class='output co'>#&gt; <span class='message'>extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span></div><div class='output co'>#&gt; <span class='message'>2020-04-17 19:54:55 getRegionCoverage: processing chr21</span></div><div class='output co'>#&gt; <span class='message'>2020-04-17 19:54:55 getRegionCoverage: done processing chr21</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


