<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Load the coverage information from a group of BAM files — loadCoverage • derfinder</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Load the coverage information from a group of BAM files — loadCoverage" />
<meta property="og:description" content="For a group of samples this function reads the coverage information for a
specific chromosome directly from the BAM files. It then merges them into a
DataFrame and removes the bases that do not pass the cutoff." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinder</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.23.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/derfinder-quickstart.html">derfinder quick start guide</a>
    </li>
    <li>
      <a href="../articles/derfinder-users-guide.html">derfinder users guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lcolladotor/derfinder/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Load the coverage information from a group of BAM files</h1>
    <small class="dont-index">Source: <a href='https://github.com/lcolladotor/derfinder/blob/master/R/loadCoverage.R'><code>R/loadCoverage.R</code></a></small>
    <div class="hidden name"><code>loadCoverage.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For a group of samples this function reads the coverage information for a
specific chromosome directly from the BAM files. It then merges them into a
DataFrame and removes the bases that do not pass the cutoff.</p>
    </div>

    <pre class="usage"><span class='fu'>loadCoverage</span>(
  <span class='no'>files</span>,
  <span class='no'>chr</span>,
  <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>filter</span> <span class='kw'>=</span> <span class='st'>"one"</span>,
  <span class='kw'>chrlen</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>output</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>bai</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>files</th>
      <td><p>A character vector with the full path to the sample BAM files
(or BigWig files).
The names are used for the column names of the DataFrame. Check
<a href='rawFiles.html'>rawFiles</a> for constructing <code>files</code>. <code>files</code> can also be a
<a href='https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html'>BamFileList</a>, <a href='https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html'>BamFile</a>,
<a href='https://rdrr.io/pkg/rtracklayer/man/BigWigFile.html'>BigWigFileList</a>, or <a href='https://rdrr.io/pkg/rtracklayer/man/BigWigFile.html'>BigWigFile</a> object.</p></td>
    </tr>
    <tr>
      <th>chr</th>
      <td><p>Chromosome to read. Should be in the format matching the one used
in the raw data.</p></td>
    </tr>
    <tr>
      <th>cutoff</th>
      <td><p>This argument is passed to <a href='filterData.html'>filterData</a>.</p></td>
    </tr>
    <tr>
      <th>filter</th>
      <td><p>Has to be either <code>'one'</code> (default) or <code>'mean'</code>. In
the first case, at least one sample has to have coverage above <code>cutoff</code>.
In the second case, the mean coverage has to be greater than <code>cutoff</code>.</p></td>
    </tr>
    <tr>
      <th>chrlen</th>
      <td><p>The chromosome length in base pairs. If it's <code>NULL</code>, the
chromosome length is extracted from the BAM files.</p></td>
    </tr>
    <tr>
      <th>output</th>
      <td><p>If <code>NULL</code> then no output is saved in disk. If <code>auto</code>
then an automatic name is constructed using UCSC names (chrXCovInfo.Rdata
for example). If another character is specified, then that name is used for
the output file.</p></td>
    </tr>
    <tr>
      <th>bai</th>
      <td><p>The full path to the BAM index files. If <code>NULL</code> it is
assumed that the BAM index files are in the same location as the BAM files
and that they have the .bai extension. Ignored if <code>files</code> is a
<code>BamFileList</code> object or if <code>inputType=='BigWig'</code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to other methods and/or advanced arguments.
Advanced arguments:</p><dl>
<dt>verbose </dt><dd><p>If <code>TRUE</code> basic status updates will be printed along
the way.</p></dd>
<dt>inputType </dt><dd><p>Has to be either <code>bam</code> or <code>BigWig</code>. It specifies
the format of the raw data files. By default it's set to <code>bam</code> before
trying to guess it from the file names.</p></dd>
<dt>tilewidth </dt><dd><p>When specified, <a href='https://rdrr.io/pkg/GenomicRanges/man/tileGenome.html'>tileGenome</a> is used
to break up the chromosome into chunks. We don't recommend this for BAM
files as the coverage in the borders of the chunks might be slightly off.</p></dd>
<dt>which </dt><dd><p><code>NULL</code> by default. When a <code>GRanges</code> is specified,
this specific region of the genome is loaded instead of the full chromosome.</p></dd>
<dt>fileStyle </dt><dd><p>The naming style of the chromosomes in the input files.
If the global option <code>chrsStyle</code> is not set, the naming style won't be
changed. This option is useful when you want to use a specific naming style
but the raw files use another style.</p></dd>
<dt>protectWhich </dt><dd><p>When not <code>NULL</code> and <code>which</code> is specified,
this argument specifies by how much the ranges in <code>which</code> will be
expanded. This helps get the same base level coverage you would get from
reading the coverage for a full chromosome from BAM files. Otherwise some
reads might be excluded and thus the base level coverage will be lower.
<code>NULL</code> by default.</p></dd>
<dt>drop.D </dt><dd><p>Whether to drop the bases with 'D' in the CIGAR strings
or to include them. Only used with BAM files. <code>FALSE</code> by default.</p></dd>
<dt>sampleNames </dt><dd><p>Column names to be used the samples. By default it's
<code><a href='https://rdrr.io/r/base/names.html'>names(files)</a></code>.</p></dd>

</dl><p>Passed to <a href='extendedMapSeqlevels.html'>extendedMapSeqlevels</a>, <a href='define_cluster.html'>define_cluster</a>,
<a href='https://rdrr.io/pkg/Rsamtools/man/ScanBamParam-class.html'>scanBamFlag</a> and <a href='filterData.html'>filterData</a>.
Note that <a href='filterData.html'>filterData</a> is used internally
by loadCoverage and has the important arguments <code>totalMapped</code>
and <code>targetSize</code> which can be used to normalize the coverage by
library size. See <a href='getTotalMapped.html'>getTotalMapped</a> for calculating <code>totalMapped</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list with two components.</p><dl>
<dt>coverage </dt><dd><p>is a DataFrame object where each column represents a
sample. The number of rows depends on the number of base pairs that passed
the cutoff and the information stored is the coverage at that given base.</p></dd>
<dt>position </dt><dd><p>is a logical Rle with the positions of the chromosome
that passed the cutoff.</p></dd>

</dl>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The <code>...</code> argument can be used to control which alignments to consider
when reading from BAM files. See <a href='https://rdrr.io/pkg/Rsamtools/man/ScanBamParam-class.html'>scanBamFlag</a>.</p>
<p>Parallelization for loading the data in chunks is used only used when
<code>tilewidth</code> is specified. You may use up to one core per tile.</p>
<p>If you set the advanced argument <code>drop.D = TRUE</code>, bases with CIGAR
string "D" (deletion from reference) will be excluded from the base-level
coverage calculation.</p>
<p>If you are working with data from an organism different from 'Homo sapiens'
specify so by setting the global 'species' and 'chrsStyle' options. For
example:
<code><a href='https://rdrr.io/r/base/options.html'>options(species = 'arabidopsis_thaliana')</a></code>
<code><a href='https://rdrr.io/r/base/options.html'>options(chrsStyle = 'NCBI')</a></code></p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='fullCoverage.html'>fullCoverage</a>, <a href='getTotalMapped.html'>getTotalMapped</a></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>datadir</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"genomeData"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"derfinder"</span>)
<span class='no'>files</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='rawFiles.html'>rawFiles</a></span>(
    <span class='kw'>datadir</span> <span class='kw'>=</span> <span class='no'>datadir</span>, <span class='kw'>samplepatt</span> <span class='kw'>=</span> <span class='st'>"*accepted_hits.bam$"</span>,
    <span class='kw'>fileterm</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
)
<span class='co'>## Shorten the column names</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>files</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span>(<span class='st'>"_accepted_hits.bam"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>files</span>))

<span class='co'>## Read and filter the data, only for 2 files</span>
<span class='no'>dataSmall</span> <span class='kw'>&lt;-</span> <span class='fu'>loadCoverage</span>(<span class='kw'>files</span> <span class='kw'>=</span> <span class='no'>files</span>[<span class='fl'>1</span>:<span class='fl'>2</span>], <span class='kw'>chr</span> <span class='kw'>=</span> <span class='st'>"21"</span>, <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='fl'>0</span>)</div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:08 loadCoverage: finding chromosome lengths</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:08 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009101_accepted_hits.bam</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:08 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009102_accepted_hits.bam</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:08 loadCoverage: applying the cutoff to the merged data</span></div><div class='output co'>#&gt; <span class='message'>2020-05-10 16:49:08 filterData: originally there were 48129895 rows, now there are 454 rows. Meaning that 100 percent was filtered.</span></div><div class='input'><span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>## Export to BigWig files</span>
<span class='fu'><a href='createBw.html'>createBw</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='st'>"chr21"</span> <span class='kw'>=</span> <span class='no'>dataSmall</span>))

<span class='co'>## Load data from BigWig files</span>
<span class='no'>dataSmall.bw</span> <span class='kw'>&lt;-</span> <span class='fu'>loadCoverage</span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(
    <span class='kw'>ERR009101</span> <span class='kw'>=</span> <span class='st'>"ERR009101.bw"</span>, <span class='kw'>ERR009102</span> <span class='kw'>=</span>
        <span class='st'>"ERR009102.bw"</span>
), <span class='kw'>chr</span> <span class='kw'>=</span> <span class='st'>"chr21"</span>)

<span class='co'>## Compare them</span>
<span class='fu'><a href='https://rdrr.io/r/base/mapply.html'>mapply</a></span>(<span class='kw'>function</span>(<span class='no'>x</span>, <span class='no'>y</span>) {
    <span class='no'>x</span> - <span class='no'>y</span>
}, <span class='no'>dataSmall</span>$<span class='no'>coverage</span>, <span class='no'>dataSmall.bw</span>$<span class='no'>coverage</span>)

<span class='co'>## Note that the only difference is the type of Rle (integer vs numeric) used</span>
<span class='co'>## to store the data.</span>
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


