<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Obtain the genomic state per region from annotation — makeGenomicState • derfinder</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Obtain the genomic state per region from annotation — makeGenomicState" />
<meta property="og:description" content="This function summarizes the annotation contained in a
TxDb at each given base of the genome based
on annotated transcripts. It groups contiguous base pairs classified as the
same type into regions." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinder</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.21.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/derfinder-quickstart.html">derfinder quick start guide</a>
    </li>
    <li>
      <a href="../articles/derfinder-users-guide.html">derfinder users guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lcolladotor/derfinder/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Obtain the genomic state per region from annotation</h1>
    <small class="dont-index">Source: <a href='https://github.com/lcolladotor/derfinder/blob/master/R/makeGenomicState.R'><code>R/makeGenomicState.R</code></a></small>
    <div class="hidden name"><code>makeGenomicState.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function summarizes the annotation contained in a
<a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>TxDb</a> at each given base of the genome based
on annotated transcripts. It groups contiguous base pairs classified as the
same type into regions.</p>
    </div>

    <pre class="usage"><span class='fu'>makeGenomicState</span>(<span class='no'>txdb</span>, <span class='kw'>chrs</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span>(<span class='fl'>22</span>), <span class='st'>"X"</span>, <span class='st'>"Y"</span>), <span class='no'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>txdb</th>
      <td><p>A <a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>TxDb</a> object with chromosome lengths
(check <code>seqlengths(txdb)</code>). If you are using a
<a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>TxDb</a> object created from a GFF/GTF file, you will
find this <a href='https://support.bioconductor.org/p/93235/'>https://support.bioconductor.org/p/93235/</a> useful.</p></td>
    </tr>
    <tr>
      <th>chrs</th>
      <td><p>The names of the chromosomes to use as denoted in the
<code>txdb</code> object. Check <a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>isActiveSeq</a>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to <a href='extendedMapSeqlevels.html'>extendedMapSeqlevels</a>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>GRangesList</code> object with two elements: <code>fullGenome</code> and
<code>codingGenome</code>. Both have metadata information for the type of region
(theRegion), transcript IDs (tx_id), transcript name (tx_name), and gene ID
(gene_id). <code>fullGenome</code> classifies each region as either being exon,
intron or intergenic. <code>codingGenome</code> classfies the regions as being
promoter, exon, intro, 5UTR, 3UTR or intergenic.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>TxDb</a></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Load the example data base from the GenomicFeatures vignette</span>
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"GenomicFeatures"</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: AnnotationDbi</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biobase</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='input'><span class='no'>samplefile</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"hg19_knownGene_sample.sqlite"</span>,
    <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"GenomicFeatures"</span>
)
<span class='no'>txdb</span> <span class='kw'>&lt;-</span> <span class='fu'>loadDb</span>(<span class='no'>samplefile</span>)

<span class='co'>## Generate genomic state object, only for chr6</span>
<span class='no'>sampleGenomicState</span> <span class='kw'>&lt;-</span> <span class='fu'>makeGenomicState</span>(<span class='no'>txdb</span>, <span class='kw'>chrs</span> <span class='kw'>=</span> <span class='st'>"chr6"</span>)</div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:1 mapping between keys and columns</span></div><div class='input'><span class='co'>#</span>
<span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>## Create the GenomicState object for Hsapiens.UCSC.hg19.knownGene</span>
<span class='no'>txdb</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class='kw ns'>::</span><span class='no'><a href='https://rdrr.io/pkg/TxDb.Hsapiens.UCSC.hg19.knownGene/man/package.html'>TxDb.Hsapiens.UCSC.hg19.knownGene</a></span>

<span class='co'>## Creating this GenomicState object takes around 8 min for all chrs and</span>
<span class='co'>## around 30 secs for chr21</span>
<span class='no'>GenomicState.Hsapiens.UCSC.hg19.knownGene.chr21</span> <span class='kw'>&lt;-</span>
    <span class='fu'>makeGenomicState</span>(<span class='kw'>txdb</span> <span class='kw'>=</span> <span class='no'>txdb</span>, <span class='kw'>chrs</span> <span class='kw'>=</span> <span class='st'>"chr21"</span>)

<span class='co'>## For convinience, this object is already included in derfinder</span>
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"testthat"</span>)
<span class='fu'><a href='https://testthat.r-lib.org/reference/expect_that.html'>expect_that</a></span>(
    <span class='no'>GenomicState.Hsapiens.UCSC.hg19.knownGene.chr21</span>,
    <span class='fu'><a href='https://testthat.r-lib.org/reference/oldskool.html'>is_equivalent_to</a></span>(<span class='no'>genomicState</span>)
)

<span class='co'>## Hsapiens ENSEMBL GRCh37</span>
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"GenomicFeatures"</span>)
<span class='co'>## Can take several minutes and speed will depend on your internet speed</span>
<span class='no'>xx</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbPackage.html'>makeTxDbPackageFromBiomart</a></span>(
    <span class='kw'>version</span> <span class='kw'>=</span> <span class='st'>"0.99"</span>, <span class='kw'>maintainer</span> <span class='kw'>=</span> <span class='st'>"Your Name"</span>,
    <span class='kw'>author</span> <span class='kw'>=</span> <span class='st'>"Your Name"</span>
)
<span class='no'>txdb</span> <span class='kw'>&lt;-</span> <span class='fu'>loadDb</span>(<span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span>(
    <span class='st'>"TxDb.Hsapiens.BioMart.ensembl.GRCh37.p11"</span>, <span class='st'>"inst"</span>,
    <span class='st'>"extdata"</span>, <span class='st'>"TxDb.Hsapiens.BioMart.ensembl.GRCh37.p11.sqlite"</span>
))

<span class='co'>## Creating this GenomicState object takes around 13 min</span>
<span class='no'>GenomicState.Hsapiens.ensembl.GRCh37.p11</span> <span class='kw'>&lt;-</span> <span class='fu'>makeGenomicState</span>(
    <span class='kw'>txdb</span> <span class='kw'>=</span> <span class='no'>txdb</span>,
    <span class='kw'>chrs</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>1</span>:<span class='fl'>22</span>, <span class='st'>"X"</span>, <span class='st'>"Y"</span>)
)

<span class='co'>## Save for later use</span>
<span class='fu'><a href='https://rdrr.io/r/base/save.html'>save</a></span>(<span class='no'>GenomicState.Hsapiens.ensembl.GRCh37.p11</span>,
    <span class='kw'>file</span> <span class='kw'>=</span> <span class='st'>"GenomicState.Hsapiens.ensembl.GRCh37.p11.Rdata"</span>
)
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


