<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Identify regions data by a coverage filter and get a count matrix from
BigWig files — railMatrix • derfinder</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Identify regions data by a coverage filter and get a count matrix from
BigWig files — railMatrix" />
<meta property="og:description" content="Rail (available at http://rail.bio) generates coverage BigWig files. These
files are faster to load in R and to process. Rail creates an un-adjusted
coverage BigWig file per sample and an adjusted summary coverage BigWig file
by chromosome (median or mean). railMatrix reads in the mean (or
median) coverage BigWig file and applies a threshold cutoff to identify
expressed regions (ERs). Then it goes back to the sample coverage BigWig
files and extracts the base level coverage for each sample. Finally it
summarizes this information in a matrix with one row per ERs and one column
per sample. This function is similar to regionMatrix but is faster
thanks to the advantages presented by BigWig files." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinder</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.21.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/derfinder-quickstart.html">derfinder quick start guide</a>
    </li>
    <li>
      <a href="../articles/derfinder-users-guide.html">derfinder users guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lcolladotor/derfinder/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Identify regions data by a coverage filter and get a count matrix from
BigWig files</h1>
    <small class="dont-index">Source: <a href='https://github.com/lcolladotor/derfinder/blob/master/R/railMatrix.R'><code>R/railMatrix.R</code></a></small>
    <div class="hidden name"><code>railMatrix.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Rail (available at http://rail.bio) generates coverage BigWig files. These
files are faster to load in R and to process. Rail creates an un-adjusted
coverage BigWig file per sample and an adjusted summary coverage BigWig file
by chromosome (median or mean). railMatrix reads in the mean (or
median) coverage BigWig file and applies a threshold cutoff to identify
expressed regions (ERs). Then it goes back to the sample coverage BigWig
files and extracts the base level coverage for each sample. Finally it
summarizes this information in a matrix with one row per ERs and one column
per sample. This function is similar to <a href='regionMatrix.html'>regionMatrix</a> but is faster
thanks to the advantages presented by BigWig files.</p>
    </div>

    <pre class="usage"><span class='fu'>railMatrix</span>(
  <span class='no'>chrs</span>,
  <span class='no'>summaryFiles</span>,
  <span class='no'>sampleFiles</span>,
  <span class='no'>L</span>,
  <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>maxClusterGap</span> <span class='kw'>=</span> <span class='fl'>300L</span>,
  <span class='kw'>totalMapped</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>targetSize</span> <span class='kw'>=</span> <span class='fl'>4e+07</span>,
  <span class='kw'>file.cores</span> <span class='kw'>=</span> <span class='fl'>1L</span>,
  <span class='kw'>chrlens</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>chrs</th>
      <td><p>A character vector with the names of the chromosomes.</p></td>
    </tr>
    <tr>
      <th>summaryFiles</th>
      <td><p>A character vector (or BigWigFileList) with the paths to
the summary BigWig files created by Rail. Either mean or median files. These
are library size adjusted by default in Rail. The order of the files in this
vector must match the order in <code>chrs</code>.</p></td>
    </tr>
    <tr>
      <th>sampleFiles</th>
      <td><p>A character vector with the paths to the BigWig files
by sample. These files are unadjusted for library size.</p></td>
    </tr>
    <tr>
      <th>L</th>
      <td><p>The width of the reads used. Either a vector of length 1 or length
equal to the number of samples.</p></td>
    </tr>
    <tr>
      <th>cutoff</th>
      <td><p>The base-pair level cutoff to use. It's behavior is controlled
by <code>filter</code>.</p></td>
    </tr>
    <tr>
      <th>maxClusterGap</th>
      <td><p>This determines the maximum gap between candidate ERs.</p></td>
    </tr>
    <tr>
      <th>totalMapped</th>
      <td><p>A vector with the total number of reads mapped for each
sample. The vector should be in the same order as the samples in <code>data</code>.
Providing this data adjusts the coverage to reads in <code>targetSize</code>
library prior to filtering. See <a href='getTotalMapped.html'>getTotalMapped</a> for
calculating this vector.</p></td>
    </tr>
    <tr>
      <th>targetSize</th>
      <td><p>The target library size to adjust the coverage to. Used
only when <code>totalMapped</code> is specified. By default, it adjusts to
libraries with 40 million reads, which matches the default used in Rail.</p></td>
    </tr>
    <tr>
      <th>file.cores</th>
      <td><p>Number of cores used for loading the BigWig files per chr.</p></td>
    </tr>
    <tr>
      <th>chrlens</th>
      <td><p>The chromosome lengths in base pairs. If it's <code>NULL</code>,
the chromosome length is extracted from the BAM files. Otherwise, it should
have the same length as <code>chrs</code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to other methods and/or advanced arguments.
Advanced arguments:</p><dl'>
<dt>verbose </dt><dd><p>If <code>TRUE</code> basic status updates will be printed along
the way. Default: <code>TRUE</code>.</p></dd>
<dt>verbose.load </dt><dd><p>If <code>TRUE</code> basic status updates will be printed
along the way when loading data. Default: <code>TRUE</code>.</p></dd>
<dt>BPPARAM.railChr </dt><dd><p>A BPPARAM object to use for the chr step. Set to
<a href='https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html'>SerialParam</a> when <code>file.cores = 1</code> and
<a href='https://rdrr.io/pkg/BiocParallel/man/SnowParam-class.html'>SnowParam</a> otherwise.</p></dd>
<dt>chunksize </dt><dd><p>Chunksize to use. Default: 1000.</p></dd>

</dl><p>Passed to <a href='filterData.html'>filterData</a>, <a href='findRegions.html'>findRegions</a> and <a href='define_cluster.html'>define_cluster</a>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list with one entry per chromosome. Then per chromosome, a list
with two components.</p><dl'>
<dt>regions </dt><dd><p>A set of regions based on the coverage filter cutoff as
returned by <a href='findRegions.html'>findRegions</a>.</p></dd>
<dt>coverageMatrix </dt><dd><p>A matrix with the mean coverage by sample for each
candidate region.</p></dd>

</dl>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Given a set of un-filtered coverage data (see <a href='fullCoverage.html'>fullCoverage</a>), create
candidate regions by applying a cutoff on the coverage values,
and obtain a count matrix where the number of rows corresponds to the number
of candidate regions and the number of columns corresponds to the number of
samples. The values are the mean coverage for a given sample for a given
region.</p>
<p>This function uses several other <a href='derfinder-package.html'>derfinder-package</a>
functions. Inspect the code if interested.</p>
<p>You should use at most one core per chromosome.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>## BigWig files are not supported in Windows</span>
<span class='kw'>if</span> (<span class='no'>.Platform</span>$<span class='no'>OS.type</span> <span class='kw'>!=</span> <span class='st'>"windows"</span>) {
    <span class='co'>## Get data</span>
    <span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"derfinderData"</span>)

    <span class='co'>## Identify sample files</span>
    <span class='no'>sampleFiles</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='rawFiles.html'>rawFiles</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"AMY"</span>,
        <span class='kw'>package</span> <span class='kw'>=</span>
            <span class='st'>"derfinderData"</span>
    ), <span class='kw'>samplepatt</span> <span class='kw'>=</span> <span class='st'>"bw"</span>, <span class='kw'>fileterm</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)
    <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>sampleFiles</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span>(<span class='st'>".bw"</span>, <span class='st'>""</span>, <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>sampleFiles</span>))

    <span class='co'>## Create the mean bigwig file. This file is normally created by Rail</span>
    <span class='co'>## but in this example we'll create it manually.</span>
    <span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"GenomicRanges"</span>)
    <span class='no'>fullCov</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='fullCoverage.html'>fullCoverage</a></span>(<span class='kw'>files</span> <span class='kw'>=</span> <span class='no'>sampleFiles</span>, <span class='kw'>chrs</span> <span class='kw'>=</span> <span class='st'>"chr21"</span>)
    <span class='no'>meanCov</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/funprog.html'>Reduce</a></span>(<span class='st'>"+"</span>, <span class='no'>fullCov</span>$<span class='no'>chr21</span>) / <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span>(<span class='no'>fullCov</span>$<span class='no'>chr21</span>)
    <span class='fu'><a href='createBw.html'>createBw</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='st'>"chr21"</span> <span class='kw'>=</span> <span class='fu'>DataFrame</span>(<span class='st'>"meanChr21"</span> <span class='kw'>=</span> <span class='no'>meanCov</span>)),
        <span class='kw'>keepGR</span> <span class='kw'>=</span>
            <span class='fl'>FALSE</span>
    )

    <span class='no'>summaryFile</span> <span class='kw'>&lt;-</span> <span class='st'>"meanChr21.bw"</span>

    <span class='co'>## Get the regions</span>
    <span class='no'>regionMat</span> <span class='kw'>&lt;-</span> <span class='fu'>railMatrix</span>(
        <span class='kw'>chrs</span> <span class='kw'>=</span> <span class='st'>"chr21"</span>, <span class='kw'>summaryFiles</span> <span class='kw'>=</span> <span class='no'>summaryFile</span>,
        <span class='kw'>sampleFiles</span> <span class='kw'>=</span> <span class='no'>sampleFiles</span>, <span class='kw'>L</span> <span class='kw'>=</span> <span class='fl'>76</span>, <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='fl'>5.1</span>,
        <span class='kw'>maxClusterGap</span> <span class='kw'>=</span> <span class='fl'>3000L</span>
    )

    <span class='co'>## Explore results</span>
    <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>regionMat</span>$<span class='no'>chr21</span>)
    <span class='no'>regionMat</span>$<span class='no'>chr21</span>$<span class='no'>regions</span>
    <span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span>(<span class='no'>regionMat</span>$<span class='no'>chr21</span>$<span class='no'>coverageMatrix</span>)
}</div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:20 fullCoverage: processing chromosome chr21</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:20 loadCoverage: finding chromosome lengths</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:20 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB113.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:20 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB123.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:20 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB126.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:20 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB130.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:20 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB135.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:20 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB136.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:20 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB145.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:21 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB153.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:21 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB159.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:21 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB178.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:21 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB92.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:21 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB97.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:21 loadCoverage: applying the cutoff to the merged data</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:21 filterData: originally there were 48129895 rows, now there are 48129895 rows. Meaning that 0 percent was filtered.</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:22 coerceGR: coercing sample meanChr21</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:23 createBwSample: exporting bw for sample meanChr21</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:24 loadCoverage: finding chromosome lengths</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:24 loadCoverage: loading BigWig file meanChr21.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:25 loadCoverage: applying the cutoff to the merged data</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:27 filterData: originally there were 48129895 rows, now there are 48129895 rows. Meaning that 0 percent was filtered.</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 filterData: originally there were 48129895 rows, now there are 15927 rows. Meaning that 99.97 percent was filtered.</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 findRegions: identifying potential segments</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 findRegions: segmenting information</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 .getSegmentsRle: segmenting with cutoff(s) 5.1</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 findRegions: identifying candidate regions</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 findRegions: identifying region clusters</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 railMatrix: processing regions 1 to 204</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB113.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB123.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB126.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB130.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB135.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB136.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB145.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB153.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:28 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB159.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB178.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB92.bw</span></div><div class='output co'>#&gt; <span class='message'>2020-04-19 20:55:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB97.bw</span></div><div class='output co'>#&gt; [1] 204  12</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


