<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>derfinder users guide • derfinder</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="derfinder users guide">
<meta property="og:description" content="derfinder">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinder</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.23.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/derfinder-quickstart.html">derfinder quick start guide</a>
    </li>
    <li>
      <a href="../articles/derfinder-users-guide.html">derfinder users guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lcolladotor/derfinder/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>derfinder users guide</h1>
                        <h4 class="author">Leonardo Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 class="date">14 May 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/lcolladotor/derfinder/blob/master/vignettes/derfinder-users-guide.Rmd"><code>vignettes/derfinder-users-guide.Rmd</code></a></small>
      <div class="hidden name"><code>derfinder-users-guide.Rmd</code></div>

    </div>

    
    
<div id="asking-for-help" class="section level1">
<h1 class="hasAnchor">
<a href="#asking-for-help" class="anchor"></a>Asking for help</h1>
<p>Please read the <a href="http://bioconductor.org/packages/release/bioc/vignettes/derfinder/inst/doc/derfinder-quickstart.html#basics">basics of using <code>derfinder</code></a> in the quick start guide. Thank you.</p>
</div>
<div id="derfinder-users-guide" class="section level1">
<h1 class="hasAnchor">
<a href="#derfinder-users-guide" class="anchor"></a><em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> users guide</h1>
<p>If you haven’t already, please read the <a href="quickstart-derfinder.html">quick start to using derfinder</a> vignette. It explains the basics of using <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em>, how to ask for help, and showcases an example analysis.</p>
<p>The <em>derfinder users guide</em> goes into more depth about what you can do with <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em>. It covers the two implementations of the DER Finder approach <a id="cite-Frazee_2014"></a>(<a href="http://biostatistics.oxfordjournals.org/content/15/3/413.long">Frazee, Sabunciyan, Hansen, Irizarry, et al., 2014</a>). That is, the (A) expressed regions-level and (B) single base-level F-statistics implementations. The vignette also includes advanced material for fine tuning some options, working with non-human data, and example scripts for a high-performance computing cluster.</p>
</div>
<div id="expressed-regions-level" class="section level1">
<h1 class="hasAnchor">
<a href="#expressed-regions-level" class="anchor"></a>Expressed regions-level</h1>
<p>The expressed regions-level implementation is based on summarizing the coverage information for all your samples and applying a cutoff to that summary. For example, by calculating the mean coverage at every base and then checking if it’s greater than some cutoff value of interest. Contiguous bases passing the cutoff become a candidate Expressed Region (ER). We can then construct a matrix summarizing the base-level coverage for each sample for the set of ERs. This matrix can then be using with packages developed for feature counting (<em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em>, <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em>, <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em>, etc) to determine which ERs have differential expression signal. That is, to identify the Differentially Expressed Regions (DERs).</p>
<div id="regionmatrix" class="section level2">
<h2 class="hasAnchor">
<a href="#regionmatrix" class="anchor"></a><code>regionMatrix()</code>
</h2>
<p>Commonly, users have aligned their raw RNA-seq data and saved the alignments in BAM files. Some might have chosen to compress this information into BigWig files. BigWig files are much faster to load than BAM files and are the type of file we prefer to work with. However, we can still identify the expressed regions from BAM files.</p>
<p>The function <code><a href="../reference/regionMatrix.html">regionMatrix()</a></code> will require you to load the data (either from BAM or BigWig files) and store it all in memory. It will then calculate the mean coverage across all samples, apply the cutoff you chose, and determine the expressed regions.</p>
<p>This is the path you will want to follow in most scenarios.</p>
</div>
<div id="railmatrix" class="section level2">
<h2 class="hasAnchor">
<a href="#railmatrix" class="anchor"></a><code>railMatrix()</code>
</h2>
<p><a href="http://rail.bio"><img src="https://github.com/nellore/rail/blob/master/assets/railrnalogodark.png?raw=true" alt="Rail logo"></a></p>
<p>Our favorite method for identifying expressed regions is based on pre-computed summary coverage files (mean or median) as well as coverage files by sample. <a href="http://rail.bio">Rail</a> is a cloud-enabled aligner that will generate:</p>
<ol style="list-style-type: decimal">
<li>Per chromosome, a summary (mean or median) coverage BigWig file. Note that the summary is adjusted for library size and is by default set to libraries with 40 million reads.</li>
<li>Per sample, a coverage BigWig file with the un-adjusted coverage.</li>
</ol>
<p><a href="http://rail.bio">Rail</a> does a great job in creating these files for us, saving time and reducing the memory load needed for this type of analysis with <code>R</code>.</p>
<p>If you have this type of data or can generate it from BAM files with other tools, you will be interested in using the <code><a href="../reference/railMatrix.html">railMatrix()</a></code> function. The output is identical to the one from <code><a href="../reference/regionMatrix.html">regionMatrix()</a></code>. It’s just much faster and memory efficient. The only drawback is that BigWig files are not fully supported in Windows as of <em><a href="https://bioconductor.org/packages/3.12/rtracklayer">rtracklayer</a></em> version 1.25.16.</p>
<p>We highly recommend this approach. <a href="http://rail.bio">Rail</a> has also other significant features such as: scalability, reduced redundancy, integrative analysis, mode agnosticism, and inexpensive cloud implementation. For more information, visit <a href="http://rail.bio">rail.bio</a>.</p>
</div>
</div>
<div id="single-base-level-f-statistics" class="section level1">
<h1 class="hasAnchor">
<a href="#single-base-level-f-statistics" class="anchor"></a>Single base-level F-statistics</h1>
<p>The DER Finder approach was originally implemented by calculating t-statistics between two groups and using a hidden markov model to determine the expression states: not expressed, expressed, differentially expressed (<a href="http://biostatistics.oxfordjournals.org/content/15/3/413.long">Frazee, Sabunciyan, Hansen, Irizarry, et al., 2014</a>). The original software works but had areas where we worked to improve it, which lead to the single base-level F-statistics implementation. Also note that the original software is no longer maintained.</p>
<p>This type of analysis first loads the data and preprocess it in a format that saves time and reduces memory later. It then fits two nested models (an alternative and a null model) with the coverage information for every single base-pair of the genome. Using the two fitted models, it calculates an F-statistic. Basically, it generates a vector along the genome with F-statistics.</p>
<p>A cutoff is then applied to the F-statistics and contiguous base-pairs of the genome passing the cutoff are considered a candidate Differentially Expressed Region (DER).</p>
<p>Calculating F-statistics along the genome is computationally intensive, but doable. The major resource drain comes from assigning p-values to the DERs. To do so, we permute the model matrices and re-calculate the F-statistics generating a set of null DERs. Once we have enough null DERs, we compare the observed DERs against the null DERs to calculate p-values for the observed DERs.</p>
<p>This type of analysis at the chromosome level is done by the <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> function, which is a high level function using many other pieces of <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em>. Once all chromosomes have been processed, <code><a href="../reference/mergeResults.html">mergeResults()</a></code> combines them.</p>
<p>Which implementation of the DER Finder approach you will want to use depends on your specific use case and computational resources available. But in general, we recommend starting with the expressed regions-level implementation.</p>
</div>
<div id="example-data" class="section level1">
<h1 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h1>
<p>In this vignette we we will analyze a small subset of the samples from the <em>BrainSpan Atlas of the Human Brain</em> <a id="cite-brainspan"></a>(<a href="http://www.brainspan.org/">BrainSpan, 2011</a>) publicly available data.</p>
<p>We first load the required packages.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co">## Load libraries</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinder"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinderData"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicRanges"</span>)</pre></body></html></div>
<div id="phenotype-data" class="section level2">
<h2 class="hasAnchor">
<a href="#phenotype-data" class="anchor"></a>Phenotype data</h2>
<p>For this example, we created a small table with the relevant phenotype data for 12 samples: 6 from fetal samples and 6 from adult samples. We chose at random a brain region, in this case the amygdaloid complex. For this example we will only look at data from chromosome 21. Other variables include the age in years and the gender. The data is shown below.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"knitr"</span>)
<span class="co">## Get pheno table</span>
<span class="no">pheno</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="no">brainspanPheno</span>, <span class="no">structure_acronym</span> <span class="kw">==</span> <span class="st">"AMY"</span>)

<span class="co">## Display the main information</span>
<span class="no">p</span> <span class="kw">&lt;-</span> <span class="no">pheno</span>[, -<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">pheno</span>) <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"structure_acronym"</span>,
    <span class="st">"structure_name"</span>, <span class="st">"file"</span>
))]
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">p</span>) <span class="kw">&lt;-</span> <span class="kw">NULL</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">p</span>, <span class="kw">format</span> <span class="kw">=</span> <span class="st">"html"</span>, <span class="kw">row.names</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
gender
</th>
<th style="text-align:left;">
lab
</th>
<th style="text-align:right;">
Age
</th>
<th style="text-align:left;">
group
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB97.AMY
</td>
<td style="text-align:right;">
-0.4423077
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB92.AMY
</td>
<td style="text-align:right;">
-0.3653846
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB178.AMY
</td>
<td style="text-align:right;">
-0.4615385
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB159.AMY
</td>
<td style="text-align:right;">
-0.3076923
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB153.AMY
</td>
<td style="text-align:right;">
-0.5384615
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB113.AMY
</td>
<td style="text-align:right;">
-0.5384615
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB130.AMY
</td>
<td style="text-align:right;">
21.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB136.AMY
</td>
<td style="text-align:right;">
23.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB126.AMY
</td>
<td style="text-align:right;">
30.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB145.AMY
</td>
<td style="text-align:right;">
36.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB123.AMY
</td>
<td style="text-align:right;">
37.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB135.AMY
</td>
<td style="text-align:right;">
40.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
</tbody>
</table>
</div>
<div id="load-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-the-data" class="anchor"></a>Load the data</h2>
<p><em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> offers three functions related to loading raw data. The first one, <code><a href="../reference/rawFiles.html">rawFiles()</a></code>, is a helper function for identifying the full paths to the input files. Next, <code><a href="../reference/loadCoverage.html">loadCoverage()</a></code> loads the base-level coverage data from either BAM or BigWig files for a specific chromosome. Finally, <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code> will load the coverage for a set of chromosomes using <code><a href="../reference/loadCoverage.html">loadCoverage()</a></code>.</p>
<p>We can load the data from <em><a href="https://bioconductor.org/packages/3.12/derfinderData">derfinderData</a></em> <a id="cite-Collado-Torres_2020"></a>(<a href="https://github.com/leekgroup/derfinderData">Collado-Torres, Jaffe, and Leek, 2020</a>) by first identifying the paths to the BigWig files with <code><a href="../reference/rawFiles.html">rawFiles()</a></code> and then loading the data with <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code>. Note that the <em>BrainSpan</em> data is already normalized by the total number of mapped reads in each sample. However, that won’t be the case with most data sets in which case you might want to use the <code>totalMapped</code> and <code>targetSize</code> arguments. The function <code><a href="../reference/getTotalMapped.html">getTotalMapped()</a></code> will be helpful to get this information.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="co">## Determine the files to use and fix the names</span>
<span class="no">files</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/rawFiles.html">rawFiles</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"AMY"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"derfinderData"</span>),
    <span class="kw">samplepatt</span> <span class="kw">=</span> <span class="st">"bw"</span>, <span class="kw">fileterm</span> <span class="kw">=</span> <span class="kw">NULL</span>
)
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">files</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">".bw"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">files</span>))

<span class="co">## Load the data from disk</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">fullCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fullCoverage.html">fullCoverage</a></span>(
    <span class="kw">files</span> <span class="kw">=</span> <span class="no">files</span>, <span class="kw">chrs</span> <span class="kw">=</span> <span class="st">"chr21"</span>,
    <span class="kw">totalMapped</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">files</span>)), <span class="kw">targetSize</span> <span class="kw">=</span> <span class="fl">1</span>
))</pre></body></html></div>
<pre><code>## 2020-05-14 15:56:10 fullCoverage: processing chromosome chr21</code></pre>
<pre><code>## 2020-05-14 15:56:10 loadCoverage: finding chromosome lengths</code></pre>
<pre><code>## 2020-05-14 15:56:10 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB113.bw</code></pre>
<pre><code>## 2020-05-14 15:56:11 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB123.bw</code></pre>
<pre><code>## 2020-05-14 15:56:11 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB126.bw</code></pre>
<pre><code>## 2020-05-14 15:56:11 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB130.bw</code></pre>
<pre><code>## 2020-05-14 15:56:11 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB135.bw</code></pre>
<pre><code>## 2020-05-14 15:56:11 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB136.bw</code></pre>
<pre><code>## 2020-05-14 15:56:11 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB145.bw</code></pre>
<pre><code>## 2020-05-14 15:56:12 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB153.bw</code></pre>
<pre><code>## 2020-05-14 15:56:12 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB159.bw</code></pre>
<pre><code>## 2020-05-14 15:56:12 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB178.bw</code></pre>
<pre><code>## 2020-05-14 15:56:12 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB92.bw</code></pre>
<pre><code>## 2020-05-14 15:56:12 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB97.bw</code></pre>
<pre><code>## 2020-05-14 15:56:12 loadCoverage: applying the cutoff to the merged data</code></pre>
<pre><code>## 2020-05-14 15:56:12 filterData: normalizing coverage</code></pre>
<pre><code>## 2020-05-14 15:56:13 filterData: done normalizing coverage</code></pre>
<pre><code>## 2020-05-14 15:56:13 filterData: originally there were 48129895 rows, now there are 48129895 rows. Meaning that 0 percent was filtered.</code></pre>
<pre><code>##    user  system elapsed 
##   3.045   0.064   3.110</code></pre>
<p>Alternatively, since the BigWig files are publicly available from <em>BrainSpan</em> (see <a href="http://download.alleninstitute.org/brainspan/MRF_BigWig_Gencode_v10/bigwig/">here</a>), we can extract the relevant coverage data using <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code>. Note that as of <em><a href="https://bioconductor.org/packages/3.12/rtracklayer">rtracklayer</a></em> 1.25.16 BigWig files are not supported on Windows: you can find the <code>fullCov</code> object inside <em><a href="https://bioconductor.org/packages/3.12/derfinderData">derfinderData</a></em> to follow the examples.</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="co">## Determine the files to use and fix the names</span>
<span class="no">files</span> <span class="kw">&lt;-</span> <span class="no">pheno</span>$<span class="no">file</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">files</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">".AMY"</span>, <span class="st">""</span>, <span class="no">pheno</span>$<span class="no">lab</span>)

<span class="co">## Load the data from the web</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">fullCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fullCoverage.html">fullCoverage</a></span>(
    <span class="kw">files</span> <span class="kw">=</span> <span class="no">files</span>, <span class="kw">chrs</span> <span class="kw">=</span> <span class="st">"chr21"</span>,
    <span class="kw">totalMapped</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">files</span>)), <span class="kw">targetSize</span> <span class="kw">=</span> <span class="fl">1</span>
))</pre></body></html></div>
<p>Note how loading the coverage for 12 samples from the web was quite fast. Although in this case we only retained the information for chromosome 21.</p>
<p>The result of <code>fullCov</code> is a list with one element per chromosome. If no filtering was performed, each chromosome has a <code>DataFrame</code> with the number of rows equaling the number of bases in the chromosome with one column per sample.</p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="co">## Lets explore it</span>
<span class="no">fullCov</span></pre></body></html></div>
<pre><code>## $chr21
## DataFrame with 48129895 rows and 12 columns
##          HSB113 HSB123 HSB126 HSB130 HSB135 HSB136 HSB145 HSB153 HSB159 HSB178
##           &lt;Rle&gt;  &lt;Rle&gt;  &lt;Rle&gt;  &lt;Rle&gt;  &lt;Rle&gt;  &lt;Rle&gt;  &lt;Rle&gt;  &lt;Rle&gt;  &lt;Rle&gt;  &lt;Rle&gt;
## 1             0      0      0      0      0      0      0      0      0      0
## 2             0      0      0      0      0      0      0      0      0      0
## 3             0      0      0      0      0      0      0      0      0      0
## 4             0      0      0      0      0      0      0      0      0      0
## 5             0      0      0      0      0      0      0      0      0      0
## ...         ...    ...    ...    ...    ...    ...    ...    ...    ...    ...
## 48129891      0      0      0      0      0      0      0      0      0      0
## 48129892      0      0      0      0      0      0      0      0      0      0
## 48129893      0      0      0      0      0      0      0      0      0      0
## 48129894      0      0      0      0      0      0      0      0      0      0
## 48129895      0      0      0      0      0      0      0      0      0      0
##          HSB92 HSB97
##          &lt;Rle&gt; &lt;Rle&gt;
## 1            0     0
## 2            0     0
## 3            0     0
## 4            0     0
## 5            0     0
## ...        ...   ...
## 48129891     0     0
## 48129892     0     0
## 48129893     0     0
## 48129894     0     0
## 48129895     0     0</code></pre>
<p>If filtering was performed, each chromosome also has a logical <code>Rle</code> indicating which bases of the chromosome passed the filtering. This information is useful later on to map back the results to the genome coordinates.</p>
</div>
<div id="filter-coverage" class="section level2">
<h2 class="hasAnchor">
<a href="#filter-coverage" class="anchor"></a>Filter coverage</h2>
<p>Depending on the use case, you might want to filter the base-level coverage at the time of reading it, or you might want to keep an unfiltered version. By default both <code><a href="../reference/loadCoverage.html">loadCoverage()</a></code> and <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code> will not filter.</p>
<p>If you decide to filter, set the <code>cutoff</code> argument to a positive value. This will run <code><a href="../reference/filterData.html">filterData()</a></code>. Note that you might want to standardize the library sizes prior to filtering if you didn’t already do it when creating the <code>fullCov</code> object. You can do so by by supplying the <code>totalMapped</code> and <code>targetSize</code> arguments to <code><a href="../reference/filterData.html">filterData()</a></code>. Note: don’t use these arguments twice in <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code> and <code><a href="../reference/filterData.html">filterData()</a></code>.</p>
<p>In this example, we prefer to keep both an unfiltered and filtered version. For the filtered version, we will retain the bases where at least one sample has coverage greater than 2.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="co">## Filter coverage</span>
<span class="no">filteredCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">fullCov</span>, <span class="no">filterData</span>, <span class="kw">cutoff</span> <span class="kw">=</span> <span class="fl">2</span>)</pre></body></html></div>
<pre><code>## 2020-05-14 15:56:14 filterData: originally there were 48129895 rows, now there are 130356 rows. Meaning that 99.73 percent was filtered.</code></pre>
<p>The result is similar to <code>fullCov</code> but with the genomic position index as shown below.</p>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="co">## Similar to fullCov but with $position</span>
<span class="no">filteredCov</span></pre></body></html></div>
<pre><code>## $chr21
## $chr21$coverage
## DataFrame with 130356 rows and 12 columns
##                  HSB113           HSB123             HSB126             HSB130
##                   &lt;Rle&gt;            &lt;Rle&gt;              &lt;Rle&gt;              &lt;Rle&gt;
## 1      2.00999999046326                0                  0 0.0399999991059303
## 2      2.17000007629395                0                  0 0.0399999991059303
## 3      2.21000003814697                0                  0 0.0399999991059303
## 4      2.36999988555908                0                  0 0.0399999991059303
## 5      2.36999988555908                0 0.0599999986588955 0.0399999991059303
## ...                 ...              ...                ...                ...
## 130352             1.25 1.27999997138977   2.04999995231628  0.790000021457672
## 130353 1.21000003814697 1.24000000953674   2.04999995231628  0.790000021457672
## 130354 1.21000003814697 1.20000004768372   1.92999994754791  0.790000021457672
## 130355 1.16999995708466 1.20000004768372   1.92999994754791  0.790000021457672
## 130356 1.16999995708466  1.0900000333786   1.87000000476837  0.790000021457672
##                   HSB135            HSB136            HSB145            HSB153
##                    &lt;Rle&gt;             &lt;Rle&gt;             &lt;Rle&gt;             &lt;Rle&gt;
## 1      0.230000004172325 0.129999995231628                 0 0.589999973773956
## 2      0.230000004172325 0.129999995231628                 0 0.629999995231628
## 3      0.230000004172325 0.129999995231628                 0 0.670000016689301
## 4      0.230000004172325 0.129999995231628                 0 0.709999978542328
## 5      0.230000004172325 0.129999995231628                 0              0.75
## ...                  ...               ...               ...               ...
## 130352  1.62999999523163  1.37000000476837  1.02999997138977  2.21000003814697
## 130353  1.62999999523163  1.37000000476837  1.02999997138977  2.21000003814697
## 130354  1.62999999523163  1.27999997138977 0.730000019073486  2.00999999046326
## 130355  1.62999999523163  1.27999997138977 0.600000023841858  2.00999999046326
## 130356  1.54999995231628  1.01999998092651 0.560000002384186  1.92999994754791
##                   HSB159             HSB178              HSB92            HSB97
##                    &lt;Rle&gt;              &lt;Rle&gt;              &lt;Rle&gt;            &lt;Rle&gt;
## 1      0.150000005960464 0.0500000007450581 0.0700000002980232 1.58000004291534
## 2      0.150000005960464 0.0500000007450581 0.0700000002980232 1.58000004291534
## 3      0.150000005960464 0.0500000007450581  0.100000001490116 1.61000001430511
## 4      0.150000005960464 0.0500000007450581  0.129999995231628 1.64999997615814
## 5                   0.25  0.100000001490116  0.129999995231628 1.67999994754791
## ...                  ...                ...                ...              ...
## 130352  2.46000003814697    2.0699999332428   2.23000001907349 1.50999999046326
## 130353  2.46000003814697    2.0699999332428   2.23000001907349 1.50999999046326
## 130354  2.21000003814697   2.10999989509583   2.13000011444092 1.47000002861023
## 130355  2.10999989509583   2.10999989509583   2.13000011444092 1.47000002861023
## 130356  1.96000003814697   1.77999997138977   2.05999994277954 1.47000002861023
## 
## $chr21$position
## logical-Rle of length 48129895 with 3235 runs
##   Lengths: 9825448     149       2       9 ...     172     740     598   45091
##   Values :   FALSE    TRUE   FALSE    TRUE ...    TRUE   FALSE    TRUE   FALSE</code></pre>
<p>In terms of memory, the filtered version requires less resources. Although this will depend on how rich the data set is and how aggressive was the filtering step.</p>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="co">## Compare the size in Mb</span>
<span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="kw">fullCov</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/object.size.html">object.size</a></span>(<span class="no">fullCov</span>),
    <span class="kw">filteredCov</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/object.size.html">object.size</a></span>(<span class="no">filteredCov</span>)
) / <span class="fl">1024</span>^<span class="fl">2</span>, <span class="fl">1</span>)</pre></body></html></div>
<pre><code>##     fullCov filteredCov 
##        22.7         8.5</code></pre>
<p>Note that with your own data, filtering for bases where at least one sample has coverage greater than 2 might not make sense: maybe you need a higher or lower filter. The amount of bases remaining after filtering will impact how long the analysis will take to complete. We thus recommend exploring this before proceeding.</p>
</div>
</div>
<div id="expressed-region-level-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#expressed-region-level-analysis" class="anchor"></a>Expressed region-level analysis</h1>
<div id="via-regionmatrix" class="section level2">
<h2 class="hasAnchor">
<a href="#via-regionmatrix" class="anchor"></a>Via <code>regionMatrix()</code>
</h2>
<p>Now that we have the data, we can identify expressed regions (ERs) by using a cutoff of 30 on the base-level mean coverage from these 12 samples. Once the regions have been identified, we can calculate a coverage matrix with one row per ER and one column per sample (12 in this case). For doing this calculation we need to know the length of the sequence reads, which in this study were 76 bp long.</p>
<p>Note that for this type of analysis there is no major benefit of filtering the data. Although it can be done if needed.</p>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="co">## Use regionMatrix()</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">regionMat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/regionMatrix.html">regionMatrix</a></span>(<span class="no">fullCov</span>, <span class="kw">cutoff</span> <span class="kw">=</span> <span class="fl">30</span>, <span class="kw">L</span> <span class="kw">=</span> <span class="fl">76</span>))</pre></body></html></div>
<pre><code>## By using totalMapped equal to targetSize, regionMatrix() assumes that you have normalized the data already in fullCoverage(), loadCoverage() or filterData().</code></pre>
<pre><code>## 2020-05-14 15:56:15 regionMatrix: processing chr21</code></pre>
<pre><code>## 2020-05-14 15:56:15 filterData: normalizing coverage</code></pre>
<pre><code>## 2020-05-14 15:56:15 filterData: done normalizing coverage</code></pre>
<pre><code>## 2020-05-14 15:56:17 filterData: originally there were 48129895 rows, now there are 2256 rows. Meaning that 100 percent was filtered.</code></pre>
<pre><code>## 2020-05-14 15:56:17 findRegions: identifying potential segments</code></pre>
<pre><code>## 2020-05-14 15:56:17 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:17 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:17 findRegions: identifying region clusters</code></pre>
<pre><code>## 2020-05-14 15:56:17 getRegionCoverage: processing chr21</code></pre>
<pre><code>## 2020-05-14 15:56:17 getRegionCoverage: done processing chr21</code></pre>
<pre><code>## 2020-05-14 15:56:17 regionMatrix: calculating coverageMatrix</code></pre>
<pre><code>## 2020-05-14 15:56:17 regionMatrix: adjusting coverageMatrix for 'L'</code></pre>
<pre><code>##    user  system elapsed 
##   2.595   0.073   2.668</code></pre>
<div class="sourceCode" id="cb47"><html><body><pre class="r"><span class="co">## Explore results</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span>(<span class="no">regionMat</span>)</pre></body></html></div>
<pre><code>## [1] "list"</code></pre>
<div class="sourceCode" id="cb49"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>)</pre></body></html></div>
<pre><code>## [1] "regions"        "coverageMatrix" "bpCoverage"</code></pre>
<p><code><a href="../reference/regionMatrix.html">regionMatrix()</a></code> returns a list of elements of length equal to the number of chromosomes analyzed. For each chromosome, there are three pieces of output. The actual ERs are arranged in a <code>GRanges</code> object named <code>regions</code>.</p>
<ul>
<li>
<em>regions</em> is the result from filtering with <code><a href="../reference/filterData.html">filterData()</a></code> and then defining the regions with <code><a href="../reference/findRegions.html">findRegions()</a></code>. Note that the metadata variable <code>value</code> represents the mean coverage for the given region while <code>area</code> is the sum of the base-level coverage (before adjusting for read length) from all samples.</li>
<li>
<em>bpCoverage</em> is a list with the base-level coverage from all the regions. This information can then be used with <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage()</a></code> from <em><a href="https://bioconductor.org/packages/3.12/derfinderPlot">derfinderPlot</a></em>.</li>
<li>
<em>coverageMatrix</em> is the matrix with one row per region and one column per sample. The different matrices for each of the chromosomes can then be merged prior to using <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em>, <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> or other packages. Note that the counts are generally not integers, but can easily be rounded if necessary.</li>
</ul>
<div class="sourceCode" id="cb51"><html><body><pre class="r"><span class="co">## regions output</span>
<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span></pre></body></html></div>
<pre><code>## GRanges object with 45 ranges and 6 metadata columns:
##      seqnames            ranges strand |     value      area indexStart
##         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt;
##    1    chr21   9827018-9827582      * |  313.6717 177224.53          1
##    2    chr21 15457301-15457438      * |  215.0846  29681.68        566
##    3    chr21 20230140-20230192      * |   38.8325   2058.12        704
##    4    chr21 20230445-20230505      * |   41.3245   2520.80        757
##    5    chr21 27253318-27253543      * |   34.9131   7890.37        818
##   ..      ...               ...    ... .       ...       ...        ...
##   41    chr21 33039644-33039688      * |   34.4705 1551.1742       2180
##   42    chr21 33040784-33040798      * |   32.1342  482.0133       2225
##   43    chr21          33040890      * |   30.0925   30.0925       2240
##   44    chr21 33040900-33040901      * |   30.1208   60.2417       2241
##   45    chr21 48019401-48019414      * |   31.1489  436.0850       2243
##       indexEnd cluster clusterL
##      &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt;
##    1       565       1      565
##    2       703       2      138
##    3       756       3      366
##    4       817       3      366
##    5      1043       4      765
##   ..       ...     ...      ...
##   41      2224      17       45
##   42      2239      18      118
##   43      2240      18      118
##   44      2242      18      118
##   45      2256      19       14
##   -------
##   seqinfo: 1 sequence from an unspecified genome</code></pre>
<div class="sourceCode" id="cb53"><html><body><pre class="r"><span class="co">## Number of regions</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>)</pre></body></html></div>
<pre><code>## [1] 45</code></pre>
<p><code>bpCoverage</code> is the base-level coverage list which can then be used for plotting.</p>
<div class="sourceCode" id="cb55"><html><body><pre class="r"><span class="co">## Base-level coverage matrices for each of the regions</span>
<span class="co">## Useful for plotting</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">bpCoverage</span>[<span class="fl">1</span>:<span class="fl">2</span>], <span class="no">head</span>, <span class="kw">n</span> <span class="kw">=</span> <span class="fl">2</span>)</pre></body></html></div>
<pre><code>## $`1`
##   HSB113 HSB123 HSB126 HSB130 HSB135 HSB136 HSB145 HSB153 HSB159 HSB178 HSB92
## 1  93.20   3.32  28.22   5.62 185.17  98.34   5.88  16.71   3.52  15.71 47.40
## 2 124.76   7.25  63.68  11.32 374.85 199.28  10.39  30.53   5.83  29.35 65.04
##   HSB97
## 1 36.54
## 2 51.42
## 
## $`2`
##     HSB113 HSB123 HSB126 HSB130 HSB135 HSB136 HSB145 HSB153 HSB159 HSB178 HSB92
## 566  45.59   7.94  15.92  34.75 141.61 104.21  19.87  38.61   4.97   23.2 13.95
## 567  45.59   7.94  15.92  35.15 141.64 104.30  19.87  38.65   4.97   23.2 13.95
##     HSB97
## 566 22.21
## 567 22.21</code></pre>
<div class="sourceCode" id="cb57"><html><body><pre class="r"><span class="co">## Check dimensions. First region is 565 long, second one is 138 bp long.</span>
<span class="co">## The columns match the number of samples (12 in this case).</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">bpCoverage</span>[<span class="fl">1</span>:<span class="fl">2</span>], <span class="no">dim</span>)</pre></body></html></div>
<pre><code>## $`1`
## [1] 565  12
## 
## $`2`
## [1] 138  12</code></pre>
<p>The end result of the coverage matrix is shown below. Note that the coverage has been adjusted for read length. Because reads might not fully align inside a given region, the numbers are generally not integers but can be rounded if needed.</p>
<div class="sourceCode" id="cb59"><html><body><pre class="r"><span class="co">## Dimensions of the coverage matrix</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">coverageMatrix</span>)</pre></body></html></div>
<pre><code>## [1] 45 12</code></pre>
<div class="sourceCode" id="cb61"><html><body><pre class="r"><span class="co">## Coverage for each region. This matrix can then be used with limma or other pkgs</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">coverageMatrix</span>)</pre></body></html></div>
<pre><code>##         HSB113     HSB123      HSB126      HSB130      HSB135      HSB136
## 1 3653.1093346 277.072106 1397.068687 1106.722895 8987.460401 5570.221054
## 2  333.3740816  99.987237  463.909476  267.354342 1198.713552 1162.313418
## 3   35.3828948  20.153553   30.725394   23.483947   16.786842   17.168947
## 4   42.3398681  29.931579   41.094474   24.724736   32.634080   19.309606
## 5   77.7402631 168.939342  115.059342  171.861974  180.638684   93.503158
## 6    0.7988158   1.770263    1.473421    2.231053    1.697368    1.007895
##        HSB145       HSB153     HSB159      HSB178       HSB92        HSB97
## 1 1330.158818 1461.2986829 297.939342 1407.288552 1168.519079 1325.9622371
## 2  257.114210  313.8513139  67.940131  193.695657  127.543553  200.7834228
## 3   22.895921   52.8756585  28.145395   33.127368   23.758816   20.4623685
## 4   33.802632   51.6146040  31.244343   33.576974   29.546183   28.2011836
## 5   90.950526   36.3046051  78.069605   97.151316  100.085790   35.5428946
## 6    1.171316    0.4221053   1.000132    1.139079    1.136447    0.3956579</code></pre>
<p>We can then use the coverage matrix and packages such as <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em>, <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> or <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> to identify which ERs are differentially expressed.</p>
<div id="find-ders-with-deseq2" class="section level3">
<h3 class="hasAnchor">
<a href="#find-ders-with-deseq2" class="anchor"></a>Find DERs with <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em>
</h3>
<p>Here we’ll use <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em> to identify the DERs. To use it we need to round the coverage data.</p>
<div class="sourceCode" id="cb63"><html><body><pre class="r"><span class="co">## Required</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"DESeq2"</span>)</pre></body></html></div>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: 'matrixStats'</code></pre>
<pre><code>## The following objects are masked from 'package:Biobase':
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## 
## Attaching package: 'DelayedArray'</code></pre>
<pre><code>## The following objects are masked from 'package:matrixStats':
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     aperm, apply, rowsum</code></pre>
<div class="sourceCode" id="cb74"><html><body><pre class="r"><span class="co">## Round matrix</span>
<span class="no">counts</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">coverageMatrix</span>)

<span class="co">## Round matrix and specify design</span>
<span class="no">dse</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span>(<span class="no">counts</span>, <span class="no">pheno</span>, ~ <span class="no">group</span> + <span class="no">gender</span>)</pre></body></html></div>
<pre><code>## converting counts to integer mode</code></pre>
<div class="sourceCode" id="cb76"><html><body><pre class="r"><span class="co">## Perform DE analysis</span>
<span class="no">dse</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="no">dse</span>, <span class="kw">test</span> <span class="kw">=</span> <span class="st">"LRT"</span>, <span class="kw">reduced</span> <span class="kw">=</span> ~<span class="no">gender</span>, <span class="kw">fitType</span> <span class="kw">=</span> <span class="st">"local"</span>)</pre></body></html></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb83"><html><body><pre class="r"><span class="co">## Extract results</span>
<span class="no">deseq</span> <span class="kw">&lt;-</span> <span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>
<span class="fu">mcols</span>(<span class="no">deseq</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu">mcols</span>(<span class="no">deseq</span>), <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="no">dse</span>))

<span class="co">## Explore the results</span>
<span class="no">deseq</span></pre></body></html></div>
<pre><code>## GRanges object with 45 ranges and 12 metadata columns:
##      seqnames            ranges strand |     value      area indexStart
##         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt;
##    1    chr21   9827018-9827582      * |  313.6717 177224.53          1
##    2    chr21 15457301-15457438      * |  215.0846  29681.68        566
##    3    chr21 20230140-20230192      * |   38.8325   2058.12        704
##    4    chr21 20230445-20230505      * |   41.3245   2520.80        757
##    5    chr21 27253318-27253543      * |   34.9131   7890.37        818
##   ..      ...               ...    ... .       ...       ...        ...
##   41    chr21 33039644-33039688      * |   34.4705 1551.1742       2180
##   42    chr21 33040784-33040798      * |   32.1342  482.0133       2225
##   43    chr21          33040890      * |   30.0925   30.0925       2240
##   44    chr21 33040900-33040901      * |   30.1208   60.2417       2241
##   45    chr21 48019401-48019414      * |   31.1489  436.0850       2243
##       indexEnd cluster clusterL  baseMean log2FoldChange     lfcSE      stat
##      &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt; &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##    1       565       1      565 2846.2872     -1.6903182  0.831959  0.215262
##    2       703       2      138  451.5196     -1.1640426  0.757490  0.871126
##    3       756       3      366   29.5781      0.0461488  0.458097  3.132082
##    4       817       3      366   36.0603     -0.1866200  0.390920  2.225708
##    5      1043       4      765  101.6468     -0.1387377  0.320166  3.957987
##   ..       ...     ...      ...       ...            ...       ...       ...
##   41      2224      17       45 20.782035      -0.642056  0.427661 0.6047814
##   42      2239      18      118  6.410542      -0.634321  0.512262 0.5454039
##   43      2240      18      118  0.129717      -0.859549  3.116540 0.0206273
##   44      2242      18      118  0.702291      -0.628285  2.247378 0.5825105
##   45      2256      19       14  5.293293      -1.694563  1.252290 5.7895910
##         pvalue      padj
##      &lt;numeric&gt; &lt;numeric&gt;
##    1 0.6426743  0.997155
##    2 0.3506436  0.997155
##    3 0.0767657  0.863614
##    4 0.1357305  0.997155
##    5 0.0466495  0.862040
##   ..       ...       ...
##   41 0.4367595  0.997155
##   42 0.4602018  0.997155
##   43 0.8857989  0.997155
##   44 0.4453299  0.997155
##   45 0.0161213  0.725460
##   -------
##   seqinfo: 1 sequence from an unspecified genome</code></pre>
<p>You can get similar results using <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> using these functions: <code>DGEList()</code>, <code>calcNormFactors()</code>, <code>estimateGLMRobustDisp()</code>, <code>glmFit()</code>, and <code>glmLRT()</code>.</p>
</div>
<div id="find-ders-with-limma" class="section level3">
<h3 class="hasAnchor">
<a href="#find-ders-with-limma" class="anchor"></a>Find DERs with <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em>
</h3>
<p>Alternatively, we can find DERs using <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em>. Here we’ll exemplify a type of test closer to what we’ll do later with the F-statistics approach. First of all, we need to define our models.</p>
<div class="sourceCode" id="cb85"><html><body><pre class="r"><span class="co">## Build models</span>
<span class="no">mod</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(~ <span class="no">pheno</span>$<span class="no">group</span> + <span class="no">pheno</span>$<span class="no">gender</span>)
<span class="no">mod0</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(~ <span class="no">pheno</span>$<span class="no">gender</span>)</pre></body></html></div>
<p>Next, we’ll transform the coverage information using the same default transformation from <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>.</p>
<div class="sourceCode" id="cb86"><html><body><pre class="r"><span class="co">## Transform coverage</span>
<span class="no">transformedCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">coverageMatrix</span> + <span class="fl">32</span>)</pre></body></html></div>
<p>We can then fit the models and get the F-statistic p-values and control the FDR.</p>
<div class="sourceCode" id="cb87"><html><body><pre class="r"><span class="co">## Example using limma</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"limma"</span>)</pre></body></html></div>
<pre><code>## 
## Attaching package: 'limma'</code></pre>
<pre><code>## The following object is masked from 'package:DESeq2':
## 
##     plotMA</code></pre>
<pre><code>## The following object is masked from 'package:BiocGenerics':
## 
##     plotMA</code></pre>
<div class="sourceCode" id="cb91"><html><body><pre class="r"><span class="co">## Run limma</span>
<span class="no">fit</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span>(<span class="no">transformedCov</span>, <span class="no">mod</span>)
<span class="no">fit0</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span>(<span class="no">transformedCov</span>, <span class="no">mod0</span>)

<span class="co">## Determine DE status for the regions</span>
<span class="co">## Also in https://github.com/LieberInstitute/jaffelab with help and examples</span>
<span class="no">getF</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">fit</span>, <span class="no">fit0</span>, <span class="no">theData</span>) {
    <span class="no">rss1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span>((<span class="fu"><a href="https://rdrr.io/r/stats/fitted.values.html">fitted</a></span>(<span class="no">fit</span>) - <span class="no">theData</span>)^<span class="fl">2</span>)
    <span class="no">df1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">fit</span>$<span class="no">coef</span>)
    <span class="no">rss0</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span>((<span class="fu"><a href="https://rdrr.io/r/stats/fitted.values.html">fitted</a></span>(<span class="no">fit0</span>) - <span class="no">theData</span>)^<span class="fl">2</span>)
    <span class="no">df0</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">fit0</span>$<span class="no">coef</span>)
    <span class="no">fstat</span> <span class="kw">&lt;-</span> ((<span class="no">rss0</span> - <span class="no">rss1</span>) / (<span class="no">df1</span> - <span class="no">df0</span>)) / (<span class="no">rss1</span> / (<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">theData</span>) - <span class="no">df1</span>))
    <span class="no">f_pval</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Fdist.html">pf</a></span>(<span class="no">fstat</span>, <span class="no">df1</span> - <span class="no">df0</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">theData</span>) - <span class="no">df1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
    <span class="no">fout</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html">cbind</a></span>(<span class="no">fstat</span>, <span class="no">df1</span> - <span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">theData</span>) - <span class="no">df1</span>, <span class="no">f_pval</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">fout</span>)[<span class="fl">2</span>:<span class="fl">3</span>] <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"df1"</span>, <span class="st">"df0"</span>)
    <span class="no">fout</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="no">fout</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="no">fout</span>)
}

<span class="no">ff</span> <span class="kw">&lt;-</span> <span class="fu">getF</span>(<span class="no">fit</span>, <span class="no">fit0</span>, <span class="no">transformedCov</span>)

<span class="co">## Get the p-value and assign it to the regions</span>
<span class="no">limma</span> <span class="kw">&lt;-</span> <span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>
<span class="no">limma</span>$<span class="no">fstat</span> <span class="kw">&lt;-</span> <span class="no">ff</span>$<span class="no">fstat</span>
<span class="no">limma</span>$<span class="no">pvalue</span> <span class="kw">&lt;-</span> <span class="no">ff</span>$<span class="no">f_pval</span>
<span class="no">limma</span>$<span class="no">padj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">ff</span>$<span class="no">f_pval</span>, <span class="st">"BH"</span>)

<span class="co">## Explore the results</span>
<span class="no">limma</span></pre></body></html></div>
<pre><code>## GRanges object with 45 ranges and 9 metadata columns:
##      seqnames            ranges strand |     value      area indexStart
##         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt;
##    1    chr21   9827018-9827582      * |  313.6717 177224.53          1
##    2    chr21 15457301-15457438      * |  215.0846  29681.68        566
##    3    chr21 20230140-20230192      * |   38.8325   2058.12        704
##    4    chr21 20230445-20230505      * |   41.3245   2520.80        757
##    5    chr21 27253318-27253543      * |   34.9131   7890.37        818
##   ..      ...               ...    ... .       ...       ...        ...
##   41    chr21 33039644-33039688      * |   34.4705 1551.1742       2180
##   42    chr21 33040784-33040798      * |   32.1342  482.0133       2225
##   43    chr21          33040890      * |   30.0925   30.0925       2240
##   44    chr21 33040900-33040901      * |   30.1208   60.2417       2241
##   45    chr21 48019401-48019414      * |   31.1489  436.0850       2243
##       indexEnd cluster clusterL     fstat    pvalue      padj
##      &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##    1       565       1      565  1.638455 0.2325446  0.581362
##    2       703       2      138  4.307443 0.0677644  0.324601
##    3       756       3      366  1.323342 0.2796406  0.629191
##    4       817       3      366  0.380332 0.5527044  0.863074
##    5      1043       4      765  7.249519 0.0246955  0.309532
##   ..       ...     ...      ...       ...       ...       ...
##   41      2224      17       45   3.11799 0.1112440  0.385075
##   42      2239      18      118   3.66184 0.0879543  0.329829
##   43      2240      18      118   3.87860 0.0804175  0.328981
##   44      2242      18      118   4.39338 0.0655381  0.324601
##   45      2256      19       14   6.80915 0.0282970  0.309532
##   -------
##   seqinfo: 1 sequence from an unspecified genome</code></pre>
<p>In this simple example, none of the ERs have strong differential expression signal when adjusting for an FDR of 5%.</p>
<div class="sourceCode" id="cb93"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">limma</span>$<span class="no">padj</span> <span class="kw">&lt;</span> <span class="fl">0.05</span>, <span class="no">deseq</span>$<span class="no">padj</span> <span class="kw">&lt;</span> <span class="fl">0.05</span>)</pre></body></html></div>
<pre><code>##        
##         FALSE
##   FALSE    45</code></pre>
</div>
</div>
<div id="via-railmatrix" class="section level2">
<h2 class="hasAnchor">
<a href="#via-railmatrix" class="anchor"></a>Via <code>railMatrix()</code>
</h2>
<p>If you have <a href="http://rail.bio">Rail</a> output, you can get the same results faster than with <code><a href="../reference/regionMatrix.html">regionMatrix()</a></code>. <a href="http://rail.bio">Rail</a> will create the summarized coverage BigWig file for you, but we are not including it in this package due to its size. So, lets create it.</p>
<div class="sourceCode" id="cb95"><html><body><pre class="r"><span class="co">## Calculate the mean: this step takes a long time with many samples</span>
<span class="no">meanCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html">Reduce</a></span>(<span class="st">"+"</span>, <span class="no">fullCov</span>$<span class="no">chr21</span>) / <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">fullCov</span>$<span class="no">chr21</span>)

<span class="co">## Save it on a bigwig file called meanChr21.bw</span>
<span class="fu"><a href="../reference/createBw.html">createBw</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"chr21"</span> <span class="kw">=</span> <span class="fu">DataFrame</span>(<span class="st">"meanChr21"</span> <span class="kw">=</span> <span class="no">meanCov</span>)),
    <span class="kw">keepGR</span> <span class="kw">=</span>
        <span class="fl">FALSE</span>
)</pre></body></html></div>
<pre><code>## 2020-05-14 15:56:23 coerceGR: coercing sample meanChr21</code></pre>
<pre><code>## 2020-05-14 15:56:23 createBwSample: exporting bw for sample meanChr21</code></pre>
<p>Now that we have the files <a href="http://rail.bio">Rail</a> creates for us, we can use <code><a href="../reference/railMatrix.html">railMatrix()</a></code>.</p>
<div class="sourceCode" id="cb98"><html><body><pre class="r"><span class="co">## Identify files to use</span>
<span class="no">summaryFile</span> <span class="kw">&lt;-</span> <span class="st">"meanChr21.bw"</span>
<span class="co">## We had already found the sample BigWig files and saved it in the object 'files'</span>
<span class="co">## Lets just rename it to sampleFiles for clarity.</span>
<span class="no">sampleFiles</span> <span class="kw">&lt;-</span> <span class="no">files</span>

<span class="co">## Get the regions</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(
    <span class="no">regionMat.rail</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/railMatrix.html">railMatrix</a></span>(
        <span class="kw">chrs</span> <span class="kw">=</span> <span class="st">"chr21"</span>, <span class="kw">summaryFiles</span> <span class="kw">=</span> <span class="no">summaryFile</span>,
        <span class="kw">sampleFiles</span> <span class="kw">=</span> <span class="no">sampleFiles</span>, <span class="kw">L</span> <span class="kw">=</span> <span class="fl">76</span>, <span class="kw">cutoff</span> <span class="kw">=</span> <span class="fl">30</span>, <span class="kw">maxClusterGap</span> <span class="kw">=</span> <span class="fl">3000L</span>
    )
)</pre></body></html></div>
<pre><code>## 2020-05-14 15:56:25 loadCoverage: finding chromosome lengths</code></pre>
<pre><code>## 2020-05-14 15:56:25 loadCoverage: loading BigWig file meanChr21.bw</code></pre>
<pre><code>## 2020-05-14 15:56:25 loadCoverage: applying the cutoff to the merged data</code></pre>
<pre><code>## 2020-05-14 15:56:27 filterData: originally there were 48129895 rows, now there are 48129895 rows. Meaning that 0 percent was filtered.</code></pre>
<pre><code>## 2020-05-14 15:56:29 filterData: originally there were 48129895 rows, now there are 2256 rows. Meaning that 100 percent was filtered.</code></pre>
<pre><code>## 2020-05-14 15:56:29 findRegions: identifying potential segments</code></pre>
<pre><code>## 2020-05-14 15:56:29 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:29 .getSegmentsRle: segmenting with cutoff(s) 30</code></pre>
<pre><code>## 2020-05-14 15:56:29 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:29 findRegions: identifying region clusters</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing regions 1 to 45</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB113.bw</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB123.bw</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB126.bw</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB130.bw</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB135.bw</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB136.bw</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB145.bw</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB153.bw</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB159.bw</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB178.bw</code></pre>
<pre><code>## 2020-05-14 15:56:29 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB92.bw</code></pre>
<pre><code>## 2020-05-14 15:56:30 railMatrix: processing file /usr/local/lib/R/host-site-library/derfinderData/extdata/AMY/HSB97.bw</code></pre>
<pre><code>##    user  system elapsed 
##   5.045   0.012   5.059</code></pre>
<p>When you take into account the time needed to load the data (<code><a href="../reference/fullCoverage.html">fullCoverage()</a></code>) and then creating the matrix (<code><a href="../reference/regionMatrix.html">regionMatrix()</a></code>), <code><a href="../reference/railMatrix.html">railMatrix()</a></code> is faster and less memory intensive.</p>
<p>The objects are not identical due to small rounding errors, but it’s nothing to worry about.</p>
<div class="sourceCode" id="cb123"><html><body><pre class="r"><span class="co">## Overall not identical due to small rounding errors</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="no">regionMat</span>, <span class="no">regionMat.rail</span>)</pre></body></html></div>
<pre><code>## [1] FALSE</code></pre>
<div class="sourceCode" id="cb125"><html><body><pre class="r"><span class="co">## Actual regions are the same</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="fu">ranges</span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>), <span class="fu">ranges</span>(<span class="no">regionMat.rail</span>$<span class="no">chr21</span>$<span class="no">regions</span>))</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb127"><html><body><pre class="r"><span class="co">## When you round, the small differences go away</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>$<span class="no">value</span>, <span class="fl">4</span>),
    <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="no">regionMat.rail</span>$<span class="no">chr21</span>$<span class="no">regions</span>$<span class="no">value</span>, <span class="fl">4</span>)
)</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb129"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(
    <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>$<span class="no">area</span>, <span class="fl">4</span>),
    <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="no">regionMat.rail</span>$<span class="no">chr21</span>$<span class="no">regions</span>$<span class="no">area</span>, <span class="fl">4</span>)
)</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
</div>
</div>
<div id="single-base-level-f-statistics-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#single-base-level-f-statistics-analysis" class="anchor"></a>Single base-level F-statistics analysis</h1>
<p>One form of base-level differential expression analysis implemented in <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> is to calculate F-statistics for every base and use them to define candidate differentially expressed regions. This type of analysis is further explained in this section.</p>
<div id="models" class="section level2">
<h2 class="hasAnchor">
<a href="#models" class="anchor"></a>Models</h2>
<p>Once we have the base-level coverage data for all 12 samples, we can construct the models. In this case, we want to find differences between fetal and adult samples while adjusting for gender and a proxy of the library size.</p>
<p>We can use <code><a href="../reference/sampleDepth.html">sampleDepth()</a></code> and it’s helper function <code><a href="../reference/collapseFullCoverage.html">collapseFullCoverage()</a></code> to get a proxy of the library size. Note that you would normally use the unfiltered data from all the chromosomes in this step and not just one.</p>
<div class="sourceCode" id="cb131"><html><body><pre class="r"><span class="co">## Get some idea of the library sizes</span>
<span class="no">sampleDepths</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/sampleDepth.html">sampleDepth</a></span>(<span class="fu"><a href="../reference/collapseFullCoverage.html">collapseFullCoverage</a></span>(<span class="no">fullCov</span>), <span class="fl">1</span>)</pre></body></html></div>
<pre><code>## 2020-05-14 15:56:30 sampleDepth: Calculating sample quantiles</code></pre>
<pre><code>## 2020-05-14 15:56:30 sampleDepth: Calculating sample adjustments</code></pre>
<div class="sourceCode" id="cb134"><html><body><pre class="r"><span class="no">sampleDepths</span></pre></body></html></div>
<pre><code>## HSB113.100% HSB123.100% HSB126.100% HSB130.100% HSB135.100% HSB136.100% 
##    19.82106    19.40505    19.53045    19.52017    20.33392    19.97758 
## HSB145.100% HSB153.100% HSB159.100% HSB178.100%  HSB92.100%  HSB97.100% 
##    19.49827    19.41285    19.24186    19.44252    19.55904    19.47733</code></pre>
<p><code><a href="../reference/sampleDepth.html">sampleDepth()</a></code> is similar to <code>calcNormFactors()</code> from <a href="http://www.bioconductor.org/packages/release/bioc/html/metagenomeSeq.html">metagenomeSeq</a> with some code underneath tailored for the type of data we are using. <code><a href="../reference/collapseFullCoverage.html">collapseFullCoverage()</a></code> is only needed to deal with the size of the data.</p>
<p>We can then define the nested models we want to use using <code><a href="../reference/makeModels.html">makeModels()</a></code>. This is a helper function that assumes that you will always adjust for the library size. You then need to define the variable to test, in this case we are comparing fetal vs adult samples. Optionally, you can adjust for other sample covariates, such as the gender in this case.</p>
<div class="sourceCode" id="cb136"><html><body><pre class="r"><span class="co">## Define models</span>
<span class="no">models</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/makeModels.html">makeModels</a></span>(<span class="no">sampleDepths</span>,
    <span class="kw">testvars</span> <span class="kw">=</span> <span class="no">pheno</span>$<span class="no">group</span>,
    <span class="kw">adjustvars</span> <span class="kw">=</span> <span class="no">pheno</span>[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"gender"</span>)]
)

<span class="co">## Explore the models</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">models</span>, <span class="no">head</span>)</pre></body></html></div>
<pre><code>## $mod
##   (Intercept) testvarsadult sampleDepths adjustVar1M
## 1           1             0     19.82106           0
## 2           1             0     19.40505           1
## 3           1             0     19.53045           1
## 4           1             0     19.52017           1
## 5           1             0     20.33392           0
## 6           1             0     19.97758           0
## 
## $mod0
##   (Intercept) sampleDepths adjustVar1M
## 1           1     19.82106           0
## 2           1     19.40505           1
## 3           1     19.53045           1
## 4           1     19.52017           1
## 5           1     20.33392           0
## 6           1     19.97758           0</code></pre>
<p>Note how the null model (<code>mod0</code>) is nested in the alternative model (<code>mod</code>). Use the same models for all your chromosomes unless you have a specific reason to use chromosome-specific models. Note that <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> is very flexible and works with any type of nested model.</p>
</div>
<div id="find-candidate-ders" class="section level2">
<h2 class="hasAnchor">
<a href="#find-candidate-ders" class="anchor"></a>Find candidate DERs</h2>
<p>Next, we can find candidate differentially expressed regions (DERs) using as input the segments of the genome where at least one sample has coverage greater than 2. That is, the filtered coverage version we created previously.</p>
<p>The <strong>main</strong> function in <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> for this type of analysis is <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>. It works at a chromosome level and runs behinds the scenes several other <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> functions. To use it, you have to provide the models, the grouping information, how to calculate the F-statistic cutoff and most importantly, the number of permutations.</p>
<p>By default <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> will use a theoretical cutoff. In this example, we use the cutoff that would correspond to a p-value of 0.05. To assign p-values to the candidate DERs, <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> permutes the rows of the model matrices, re-calculates the F-statistics and identifies null regions. Then it compares the area of the observed regions versus the areas from the null regions to assign an empirical p-value.</p>
<p>In this example we will use twenty permutations, although in a real case scenario you might consider a larger number of permutations.</p>
<p>In real scenario, you might consider saving the results from all the chromosomes in a given directory. Here we will use <em>analysisResults</em>. For each chromosome you analyze, a new directory with the chromosome-specific data will be created. So in this case, we will have <em>analysisResults/chr21</em>.</p>
<div class="sourceCode" id="cb138"><html><body><pre class="r"><span class="co">## Create a analysis directory</span>
<span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span>(<span class="st">"analysisResults"</span>)
<span class="no">originalWd</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span>()
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="no">originalWd</span>, <span class="st">"analysisResults"</span>))

<span class="co">## Perform differential expression analysis</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/analyzeChr.html">analyzeChr</a></span>(
    <span class="kw">chr</span> <span class="kw">=</span> <span class="st">"chr21"</span>, <span class="no">filteredCov</span>$<span class="no">chr21</span>, <span class="no">models</span>,
    <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="no">pheno</span>$<span class="no">group</span>, <span class="kw">writeOutput</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">cutoffFstat</span> <span class="kw">=</span> <span class="fl">5e-02</span>,
    <span class="kw">nPermute</span> <span class="kw">=</span> <span class="fl">20</span>, <span class="kw">seeds</span> <span class="kw">=</span> <span class="fl">20140923</span> + <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fl">20</span>), <span class="kw">returnOutput</span> <span class="kw">=</span> <span class="fl">TRUE</span>
))</pre></body></html></div>
<pre><code>## 2020-05-14 15:56:31 analyzeChr: Pre-processing the coverage data</code></pre>
<pre><code>## 2020-05-14 15:56:34 analyzeChr: Calculating statistics</code></pre>
<pre><code>## 2020-05-14 15:56:34 calculateStats: calculating the F-statistics</code></pre>
<pre><code>## 2020-05-14 15:56:34 analyzeChr: Calculating pvalues</code></pre>
<pre><code>## 2020-05-14 15:56:34 analyzeChr: Using the following theoretical cutoff for the F-statistics 5.31765507157871</code></pre>
<pre><code>## 2020-05-14 15:56:34 calculatePvalues: identifying data segments</code></pre>
<pre><code>## 2020-05-14 15:56:34 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:34 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:34 findRegions: identifying region clusters</code></pre>
<pre><code>## 2020-05-14 15:56:35 calculatePvalues: calculating F-statistics for permutation 1 and seed 20140924</code></pre>
<pre><code>## 2020-05-14 15:56:35 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:35 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:35 calculatePvalues: calculating F-statistics for permutation 2 and seed 20140925</code></pre>
<pre><code>## 2020-05-14 15:56:35 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:35 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:35 calculatePvalues: calculating F-statistics for permutation 3 and seed 20140926</code></pre>
<pre><code>## 2020-05-14 15:56:36 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:36 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:36 calculatePvalues: calculating F-statistics for permutation 4 and seed 20140927</code></pre>
<pre><code>## 2020-05-14 15:56:36 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:37 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:37 calculatePvalues: calculating F-statistics for permutation 5 and seed 20140928</code></pre>
<pre><code>## 2020-05-14 15:56:37 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:37 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:37 calculatePvalues: calculating F-statistics for permutation 6 and seed 20140929</code></pre>
<pre><code>## 2020-05-14 15:56:37 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:37 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:37 calculatePvalues: calculating F-statistics for permutation 7 and seed 20140930</code></pre>
<pre><code>## 2020-05-14 15:56:37 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:37 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:37 calculatePvalues: calculating F-statistics for permutation 8 and seed 20140931</code></pre>
<pre><code>## 2020-05-14 15:56:37 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:38 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:38 calculatePvalues: calculating F-statistics for permutation 9 and seed 20140932</code></pre>
<pre><code>## 2020-05-14 15:56:38 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:38 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:38 calculatePvalues: calculating F-statistics for permutation 10 and seed 20140933</code></pre>
<pre><code>## 2020-05-14 15:56:38 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:38 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:38 calculatePvalues: calculating F-statistics for permutation 11 and seed 20140934</code></pre>
<pre><code>## 2020-05-14 15:56:38 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:38 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:38 calculatePvalues: calculating F-statistics for permutation 12 and seed 20140935</code></pre>
<pre><code>## 2020-05-14 15:56:39 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:39 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:39 calculatePvalues: calculating F-statistics for permutation 13 and seed 20140936</code></pre>
<pre><code>## 2020-05-14 15:56:39 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:39 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:39 calculatePvalues: calculating F-statistics for permutation 14 and seed 20140937</code></pre>
<pre><code>## 2020-05-14 15:56:39 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:39 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:39 calculatePvalues: calculating F-statistics for permutation 15 and seed 20140938</code></pre>
<pre><code>## 2020-05-14 15:56:39 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:39 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:39 calculatePvalues: calculating F-statistics for permutation 16 and seed 20140939</code></pre>
<pre><code>## 2020-05-14 15:56:40 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:40 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:40 calculatePvalues: calculating F-statistics for permutation 17 and seed 20140940</code></pre>
<pre><code>## 2020-05-14 15:56:40 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:40 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:40 calculatePvalues: calculating F-statistics for permutation 18 and seed 20140941</code></pre>
<pre><code>## 2020-05-14 15:56:40 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:40 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:40 calculatePvalues: calculating F-statistics for permutation 19 and seed 20140942</code></pre>
<pre><code>## 2020-05-14 15:56:40 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:40 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:40 calculatePvalues: calculating F-statistics for permutation 20 and seed 20140943</code></pre>
<pre><code>## 2020-05-14 15:56:41 findRegions: segmenting information</code></pre>
<pre><code>## 2020-05-14 15:56:41 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-05-14 15:56:41 calculatePvalues: calculating the p-values</code></pre>
<pre><code>## 2020-05-14 15:56:41 analyzeChr: Annotating regions</code></pre>
<pre><code>## No annotationPackage supplied. Trying org.Hs.eg.db.</code></pre>
<pre><code>## Loading required package: org.Hs.eg.db</code></pre>
<pre><code>## Warning in library(package, lib.loc = lib.loc, character.only = TRUE,
## logical.return = TRUE, : there is no package called 'org.Hs.eg.db'</code></pre>
<pre><code>## Could not load org.Hs.eg.db. Will continue without annotation</code></pre>
<pre><code>## Getting TSS and TSE.</code></pre>
<pre><code>## Getting CSS and CSE.</code></pre>
<pre><code>## Getting exons.</code></pre>
<pre><code>## .....</code></pre>
<pre><code>##    user  system elapsed 
##  50.750   2.521  50.640</code></pre>
<p>To speed up <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>, you might need to use several cores via the <code>mc.cores</code> argument. If memory is limiting, you might want to use a smaller <code>chunksize</code> (default is 5 million). Note that if you use too many cores, you might hit the input/output ceiling of your data network and/or hard drives speed.</p>
<p>Before running with a large number of permutations we recommend exploring how long each permutation cycle takes using a single permutation.</p>
<p>Note that analyzing each chromosome with a large number of permutations and a rich data set can take several hours, so we recommend running one job running <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> per chromosome, and then merging the results via <code><a href="../reference/mergeResults.html">mergeResults()</a></code>. This process is further described in the advanced <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> vignette.</p>
</div>
<div id="explore-results" class="section level2">
<h2 class="hasAnchor">
<a href="#explore-results" class="anchor"></a>Explore results</h2>
<p>When using <code>returnOutput = TRUE</code>, <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> will return a list with the results to explore interactively. However, by default it writes the results to disk (one .Rdata file per result).</p>
<p>The following code explores the results.</p>
<div class="sourceCode" id="cb219"><html><body><pre class="r"><span class="co">## Explore</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">results</span>)</pre></body></html></div>
<pre><code>## [1] "timeinfo"     "optionsStats" "coveragePrep" "fstats"       "regions"     
## [6] "annotation"</code></pre>
<div id="optionstats" class="section level3">
<h3 class="hasAnchor">
<a href="#optionstats" class="anchor"></a>optionStats</h3>
<p><code>optionStats</code> stores the main options used in the <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> call including the models used, the type of cutoff, number of permutations, seeds for the permutations. All this information can be useful to reproduce the analysis.</p>
<div class="sourceCode" id="cb221"><html><body><pre class="r"><span class="co">## Explore optionsStats</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">results</span>$<span class="no">optionsStats</span>)</pre></body></html></div>
<pre><code>##  [1] "models"          "cutoffPre"       "scalefac"        "chunksize"      
##  [5] "cutoffFstat"     "cutoffType"      "nPermute"        "seeds"          
##  [9] "groupInfo"       "lowMemDir"       "analyzeCall"     "cutoffFstatUsed"
## [13] "smooth"          "smoothFunction"  "weights"         "returnOutput"</code></pre>
<div class="sourceCode" id="cb223"><html><body><pre class="r"><span class="co">## Call used</span>
<span class="no">results</span>$<span class="no">optionsStats</span>$<span class="no">analyzeCall</span></pre></body></html></div>
<pre><code>## analyzeChr(chr = "chr21", coverageInfo = filteredCov$chr21, models = models, 
##     cutoffFstat = 0.05, nPermute = 20, seeds = 20140923 + seq_len(20), 
##     groupInfo = pheno$group, writeOutput = TRUE, returnOutput = TRUE)</code></pre>
</div>
<div id="coverageprep" class="section level3">
<h3 class="hasAnchor">
<a href="#coverageprep" class="anchor"></a>coveragePrep</h3>
<p><code>coveragePrep</code> has the result from the <code><a href="../reference/preprocessCoverage.html">preprocessCoverage()</a></code> step. This includes the genomic position index, the mean coverage (after scaling and the log2 transformation) for all the samples, and the group mean coverages. By default, the chunks are written to disk in <code>optionsStats$lowMemDir</code> (chr21/chunksDir in this example) to help reduce the required memory resources. Otherwise it is stored in <code>coveragePrep$coverageProcessed</code>.</p>
<div class="sourceCode" id="cb225"><html><body><pre class="r"><span class="co">## Explore coveragePrep</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">results</span>$<span class="no">coveragePrep</span>)</pre></body></html></div>
<pre><code>## [1] "coverageProcessed" "mclapplyIndex"     "position"         
## [4] "meanCoverage"      "groupMeans"</code></pre>
<div class="sourceCode" id="cb227"><html><body><pre class="r"><span class="co">## Group means</span>
<span class="no">results</span>$<span class="no">coveragePrep</span>$<span class="no">groupMeans</span></pre></body></html></div>
<pre><code>## $fetal
## numeric-Rle of length 130356 with 116452 runs
##   Lengths:                 1                 1 ...                 1
##   Values : 0.401666664828857 0.428333345800638 ...  1.24833332498868
## 
## $adult
## numeric-Rle of length 130356 with 119226 runs
##   Lengths:                 1                 1 ...                 1
##   Values : 0.406666670615474  0.41333334085842 ...   1.6266666551431</code></pre>
</div>
<div id="fstats" class="section level3">
<h3 class="hasAnchor">
<a href="#fstats" class="anchor"></a>fstats</h3>
<p>The F-statistics are then stored in <code>fstats</code>. These are calculated using <code><a href="../reference/calculateStats.html">calculateStats()</a></code>.</p>
<div class="sourceCode" id="cb229"><html><body><pre class="r"><span class="co">## Explore optionsStats</span>
<span class="no">results</span>$<span class="no">fstats</span></pre></body></html></div>
<pre><code>## numeric-Rle of length 130356 with 126807 runs
##   Lengths:                   1                   1 ...                   1
##   Values :  0.0192260952849047  0.0299693665742318 ...    2.32463844414497</code></pre>
<div class="sourceCode" id="cb231"><html><body><pre class="r"><span class="co">## Note that the length matches the number of bases used</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">results</span>$<span class="no">fstats</span>), <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">results</span>$<span class="no">coveragePrep</span>$<span class="no">position</span>))</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
</div>
<div id="regions" class="section level3">
<h3 class="hasAnchor">
<a href="#regions" class="anchor"></a>regions</h3>
<p>The candidate DERs and summary results from the permutations is then stored in <code>regions</code>. This is the output from <code><a href="../reference/calculatePvalues.html">calculatePvalues()</a></code> which uses several underneath other functions including <code><a href="../reference/calculateStats.html">calculateStats()</a></code> and <code><a href="../reference/findRegions.html">findRegions()</a></code>.</p>
<div class="sourceCode" id="cb233"><html><body><pre class="r"><span class="co">## Explore regions</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">results</span>$<span class="no">regions</span>)</pre></body></html></div>
<pre><code>## [1] "regions"         "nullStats"       "nullWidths"      "nullPermutation"</code></pre>
<p>For the null regions, the summary information is composed of the mean F-statistic for the null regions (<code>regions$nullStats</code>), the width of the null regions (<code>regions$nullWidths</code>), and the permutation number under which they were identified (<code>regions$nullPermutation</code>).</p>
<div class="sourceCode" id="cb235"><html><body><pre class="r"><span class="co">## Permutation summary information</span>
<span class="no">results</span>$<span class="no">regions</span>[<span class="fl">2</span>:<span class="fl">4</span>]</pre></body></html></div>
<pre><code>## $nullStats
## numeric-Rle of length 13994 with 13994 runs
##   Lengths:                1                1 ...                1
##   Values : 5.43461202789059 5.71737636013832 ... 5.62516039511788
## 
## $nullWidths
## integer-Rle of length 13994 with 12365 runs
##   Lengths:   2   1   1   1   1   1   3   1 ...   1   1   1   2   1   1   4   1
##   Values :   1  24   7   1  32   2   1  11 ...   1   3  45   4  28   6   1   2
## 
## $nullPermutation
## integer-Rle of length 13994 with 20 runs
##   Lengths:  246  350  574  554  396  462  482 ...  746  114 1460  428  802  278
##   Values :    1    2    3    4    5    6    7 ...   15   16   17   18   19   20</code></pre>
<p>The most important part of the output is the <code>GRanges</code> object with the candidate DERs shown below.</p>
<div class="sourceCode" id="cb237"><html><body><pre class="r"><span class="co">## Candidate DERs</span>
<span class="no">results</span>$<span class="no">regions</span>$<span class="no">regions</span></pre></body></html></div>
<pre><code>## GRanges object with 591 ranges and 14 metadata columns:
##      seqnames            ranges strand |     value      area indexStart
##         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;  &lt;integer&gt;
##   up    chr21 47610386-47610682      * |  11.10304   3297.60     122158
##   up    chr21 40196145-40196444      * |  10.06142   3018.43      76110
##   up    chr21 27253616-27253948      * |   8.43488   2808.82      22019
##   up    chr21 22115534-22115894      * |   7.23645   2612.36      12274
##   up    chr21 22914853-22915064      * |   9.78066   2073.50      17318
##   ..      ...               ...    ... .       ...       ...        ...
##   up    chr21          35889784      * |   5.31952   5.31952      60088
##   up    chr21          47610093      * |   5.31912   5.31912     121865
##   up    chr21          16333728      * |   5.31881   5.31881       5048
##   up    chr21          34001896      * |   5.31871   5.31871      32577
##   up    chr21          34809571      * |   5.31801   5.31801      43694
##       indexEnd cluster clusterL meanCoverage meanfetal meanadult
##      &lt;integer&gt;   &lt;Rle&gt;    &lt;Rle&gt;    &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##   up    122454     138      933     1.597952   0.82289  2.373013
##   up     76409      71     1323     1.303508   2.02532  0.581694
##   up     22351      28      407    33.657858  42.46704 24.848674
##   up     12634       9      694     0.964464   1.71906  0.209871
##   up     17529      21      217     2.838978   4.23593  1.442028
##   ..       ...     ...      ...          ...       ...       ...
##   up     60088      51      742      2.75417   3.36000   2.14833
##   up    121865     138      933      1.45583   0.77500   2.13667
##   up      5048       1        9      1.19500   1.23167   1.15833
##   up     32577      38     1428      1.71250   2.33333   1.09167
##   up     43694      46      149      2.95000   2.87833   3.02167
##      log2FoldChangeadultvsfetal    pvalues significant   qvalues
##                       &lt;numeric&gt;  &lt;numeric&gt;    &lt;factor&gt; &lt;numeric&gt;
##   up                   1.527949 0.00278671        TRUE  0.738407
##   up                  -1.799818 0.00378707        TRUE  0.738407
##   up                  -0.773175 0.00464452        TRUE  0.738407
##   up                  -3.034045 0.00535906        TRUE  0.738407
##   up                  -1.554578 0.00793140        TRUE  0.738407
##   ..                        ...        ...         ...       ...
##   up                 -0.6452433   0.997856       FALSE  0.974463
##   up                  1.4630937   0.998285       FALSE  0.974463
##   up                 -0.0885613   0.998714       FALSE  0.974463
##   up                 -1.0958600   0.998714       FALSE  0.974463
##   up                  0.0701108   0.999571       FALSE  0.974463
##      significantQval
##             &lt;factor&gt;
##   up           FALSE
##   up           FALSE
##   up           FALSE
##   up           FALSE
##   up           FALSE
##   ..             ...
##   up           FALSE
##   up           FALSE
##   up           FALSE
##   up           FALSE
##   up           FALSE
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>The metadata columns are:</p>
<ul>
<li>
<em>value</em> is the mean F-statistics for the candidate DER.</li>
<li>
<em>area</em> is the sum of the F-statistics for the candidate DER.</li>
<li>
<em>indexStart</em> Relates the genomic start coordinate with the filtered genomic index start coordinate.</li>
<li>
<em>indexEnd</em> Similarly as above but for the end coordinates.</li>
<li>
<em>cluster</em> The cluster id to which this candidate DER belongs to.</li>
<li>
<em>clusterL</em> The length of the cluster to which this candidate DER belongs to.</li>
<li>
<em>meanCoverage</em> The base level mean coverage for the candidate DER.</li>
<li>
<em>meanfetal</em> In this example, the mean coverage for the fetal samples.</li>
<li>
<em>meanadult</em> In this example, the mean coverage for the adult samples.</li>
<li>
<em>log2FoldChangeadultvsfetal</em> In this example, the log2 fold change between adult vs fetal samples.</li>
<li>
<em>pvalues</em> The p-value for the candidate DER.</li>
<li>
<em>significant</em> By default, whether the p-value is less than 0.05 or not.</li>
<li>
<em>qvalues</em> The q-value for the candidate DER calculated with <a href="http://www.bioconductor.org/packages/release/bioc/html/qvalue.html">qvalue</a>.</li>
<li>
<em>significantQval</em> By default, whether the q-value is less than 0.10 or not.</li>
</ul>
<p>Note that for this type of analysis you might want to try a few coverage cutoffs and/or F-statistic cutoffs. One quick way to evaluate the results is to compare the width of the regions.</p>
<div class="sourceCode" id="cb239"><html><body><pre class="r"><span class="co">## Width of potential DERs</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/summary.html">summary</a></span>(<span class="fu">width</span>(<span class="no">results</span>$<span class="no">regions</span>$<span class="no">regions</span>))</pre></body></html></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    1.00    1.00    4.00   17.98   17.00  361.00</code></pre>
<div class="sourceCode" id="cb241"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="fu">width</span>(<span class="no">results</span>$<span class="no">regions</span>$<span class="no">regions</span>) <span class="kw">&gt;</span> <span class="fl">50</span>)</pre></body></html></div>
<pre><code>## [1] 68</code></pre>
<div class="sourceCode" id="cb243"><html><body><pre class="r"><span class="co">## Width of candidate DERs</span>
<span class="no">sig</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/logical.html">as.logical</a></span>(<span class="no">results</span>$<span class="no">regions</span>$<span class="no">regions</span>$<span class="no">significant</span>)
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/summary.html">summary</a></span>(<span class="fu">width</span>(<span class="no">results</span>$<span class="no">regions</span>$<span class="no">regions</span>[<span class="no">sig</span>]))</pre></body></html></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    65.0    81.5    97.0   127.8   122.0   361.0</code></pre>
<div class="sourceCode" id="cb245"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="fu">width</span>(<span class="no">results</span>$<span class="no">regions</span>$<span class="no">regions</span>[<span class="no">sig</span>]) <span class="kw">&gt;</span> <span class="fl">50</span>)</pre></body></html></div>
<pre><code>## [1] 35</code></pre>
</div>
<div id="nearest-annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#nearest-annotation" class="anchor"></a>Nearest annotation</h3>
<p><code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> will find the nearest annotation feature using <code><a href="https://rdrr.io/pkg/bumphunter/man/matchGenes.html">matchGenes()</a></code> from <em><a href="https://bioconductor.org/packages/3.12/bumphunter">bumphunter</a></em> (version &gt;= 1.7.3). This information is useful considering that the candidate DERs were identified without relying on annotation. Yet at the end, we are interested to check if they are inside a known exon, upstream a gene, etc.</p>
<div class="sourceCode" id="cb247"><html><body><pre class="r"><span class="co">## Nearest annotation</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">results</span>$<span class="no">annotation</span>)</pre></body></html></div>
<pre><code>##   name annotation description     region distance   subregion insideDistance
## 1 &lt;NA&gt;       &lt;NA&gt; inside exon     inside    38056 inside exon              0
## 2 &lt;NA&gt;       &lt;NA&gt; inside exon     inside    18390 inside exon              0
## 3 &lt;NA&gt;       &lt;NA&gt; inside exon     inside   289498 inside exon              0
## 4 &lt;NA&gt;       &lt;NA&gt; inside exon     inside    59532 inside exon              0
## 5 &lt;NA&gt;       &lt;NA&gt;  downstream downstream   544220        &lt;NA&gt;             NA
## 6 &lt;NA&gt;       &lt;NA&gt; inside exon     inside    59315 inside exon              0
##   exonnumber nexons                         UTR strand  geneL codingL Geneid
## 1         22     22                       3'UTR      -  39700      NA   4047
## 2         10     10                       3'UTR      +  19123      NA   2114
## 3         16     16                       3'UTR      - 290585      NA    351
## 4          7      7 inside transcription region      -  60513      NA 387486
## 5         NA     15                        &lt;NA&gt;      + 541884      NA   4685
## 6          7      7 inside transcription region      -  60513      NA 387486
##   subjectHits
## 1       34356
## 2       17340
## 3       31161
## 4       32699
## 5       37133
## 6       32699</code></pre>
<p>For more details on the output please check the <em><a href="https://bioconductor.org/packages/3.12/bumphunter">bumphunter</a></em> package.</p>
<p>Check the section about non-human data (specifically, using annotation different from <strong>hg19</strong>) on the advanced vignette.</p>
</div>
<div id="time-spent" class="section level3">
<h3 class="hasAnchor">
<a href="#time-spent" class="anchor"></a>Time spent</h3>
<p>The final piece is the wallclock time spent during each of the steps in <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>. You can then make a plot with this information as shown in Figure @ref(fig:exploreTime).</p>
<div class="sourceCode" id="cb249"><html><body><pre class="r"><span class="co">## Time spent</span>
<span class="no">results</span>$<span class="no">timeinfo</span></pre></body></html></div>
<pre><code>##                      init                     setup                  prepData 
## "2020-05-14 15:56:31 UTC" "2020-05-14 15:56:31 UTC" "2020-05-14 15:56:34 UTC" 
##                  savePrep            calculateStats                 saveStats 
## "2020-05-14 15:56:34 UTC" "2020-05-14 15:56:34 UTC" "2020-05-14 15:56:34 UTC" 
##             saveStatsOpts          calculatePvalues                  saveRegs 
## "2020-05-14 15:56:34 UTC" "2020-05-14 15:56:41 UTC" "2020-05-14 15:56:41 UTC" 
##                  annotate                  saveAnno 
## "2020-05-14 15:57:21 UTC" "2020-05-14 15:57:21 UTC"</code></pre>
<div class="sourceCode" id="cb251"><html><body><pre class="r"><span class="co">## Use this information to make a plot</span>
<span class="no">timed</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/diff.html">diff</a></span>(<span class="no">results</span>$<span class="no">timeinfo</span>)
<span class="no">timed.df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">Seconds</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(<span class="no">timed</span>), <span class="kw">Step</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">timed</span>),
    <span class="kw">levels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">timed</span>))
))
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"ggplot2"</span>)
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">timed.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">y</span> <span class="kw">=</span> <span class="no">Step</span>, <span class="kw">x</span> <span class="kw">=</span> <span class="no">Seconds</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>()</pre></body></html></div>
<div class="figure">
<img src="derfinder-users-guide_files/figure-html/exploreTime-1.png" alt="Seconds used to run each step in analyzeChr()." width="700"><p class="caption">
Seconds used to run each step in analyzeChr().
</p>
</div>
</div>
</div>
<div id="merge-results" class="section level2">
<h2 class="hasAnchor">
<a href="#merge-results" class="anchor"></a>Merge results</h2>
<p>Once you have analyzed each chromosome using <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>, you can use <code><a href="../reference/mergeResults.html">mergeResults()</a></code> to merge the results. This function does not return an object in R but instead creates several Rdata files with the main results from the different chromosomes.</p>
<div class="sourceCode" id="cb252"><html><body><pre class="r"><span class="co">## Go back to the original directory</span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span>(<span class="no">originalWd</span>)

<span class="co">## Merge results from several chromosomes. In this case we only have one.</span>
<span class="fu"><a href="../reference/mergeResults.html">mergeResults</a></span>(
    <span class="kw">chrs</span> <span class="kw">=</span> <span class="st">"chr21"</span>, <span class="kw">prefix</span> <span class="kw">=</span> <span class="st">"analysisResults"</span>,
    <span class="kw">genomicState</span> <span class="kw">=</span> <span class="no">genomicState</span>$<span class="no">fullGenome</span>,
    <span class="kw">optionsStats</span> <span class="kw">=</span> <span class="no">results</span>$<span class="no">optionsStats</span>
)</pre></body></html></div>
<pre><code>## 2020-05-14 15:57:22 mergeResults: Saving options used</code></pre>
<pre><code>## 2020-05-14 15:57:22 Loading chromosome chr21</code></pre>
<pre><code>## 2020-05-14 15:57:22 mergeResults: calculating FWER</code></pre>
<pre><code>## 2020-05-14 15:57:22 mergeResults: Saving fullNullSummary</code></pre>
<pre><code>## 2020-05-14 15:57:22 mergeResults: Re-calculating the p-values</code></pre>
<pre><code>## 2020-05-14 15:57:22 mergeResults: Saving fullRegions</code></pre>
<pre><code>## 2020-05-14 15:57:22 mergeResults: assigning genomic states</code></pre>
<pre><code>## 2020-05-14 15:57:22 annotateRegions: counting</code></pre>
<pre><code>## 2020-05-14 15:57:22 annotateRegions: annotating</code></pre>
<pre><code>## 2020-05-14 15:57:22 mergeResults: Saving fullAnnotatedRegions</code></pre>
<pre><code>## 2020-05-14 15:57:22 mergeResults: Saving fullFstats</code></pre>
<pre><code>## 2020-05-14 15:57:22 mergeResults: Saving fullTime</code></pre>
<div class="sourceCode" id="cb265"><html><body><pre class="r"><span class="co">## Files created by mergeResults()</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="st">"analysisResults"</span>, <span class="kw">pattern</span> <span class="kw">=</span> <span class="st">".Rdata"</span>)</pre></body></html></div>
<pre><code>## [1] "fullAnnotatedRegions.Rdata" "fullFstats.Rdata"          
## [3] "fullNullSummary.Rdata"      "fullRegions.Rdata"         
## [5] "fullTime.Rdata"             "optionsMerge.Rdata"</code></pre>
<ul>
<li>
<em>fullFstats.Rdata</em> contains a list with one element per chromosome. Per chromosome it has the F-statistics.</li>
<li>
<em>fullNullSummary.Rdata</em> is a list with the summary information from the null regions stored for each chromosome.</li>
<li>
<em>fullTime.Rdata</em> has the timing information for each chromosome as a list.</li>
</ul>
<div id="optionsmerge" class="section level3">
<h3 class="hasAnchor">
<a href="#optionsmerge" class="anchor"></a>optionsMerge</h3>
<p>For reproducibility purposes, the options used the merge the results are stored in <code>optionsMerge</code>.</p>
<div class="sourceCode" id="cb267"><html><body><pre class="r"><span class="co">## Options used to merge</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"analysisResults"</span>, <span class="st">"optionsMerge.Rdata"</span>))

<span class="co">## Contents</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">optionsMerge</span>)</pre></body></html></div>
<pre><code>## [1] "chrs"            "significantCut"  "minoverlap"      "mergeCall"      
## [5] "cutoffFstatUsed" "optionsStats"</code></pre>
<div class="sourceCode" id="cb269"><html><body><pre class="r"><span class="co">## Merge call</span>
<span class="no">optionsMerge</span>$<span class="no">mergeCall</span></pre></body></html></div>
<pre><code>## mergeResults(chrs = "chr21", prefix = "analysisResults", genomicState = genomicState$fullGenome, 
##     optionsStats = results$optionsStats)</code></pre>
</div>
<div id="fullregions" class="section level3">
<h3 class="hasAnchor">
<a href="#fullregions" class="anchor"></a>fullRegions</h3>
<p>The main result from <code><a href="../reference/mergeResults.html">mergeResults()</a></code> is in <code>fullRegions</code>. This is a <code>GRanges</code> object with the candidate DERs from all the chromosomes. It also includes the nearest annotation metadata as well as FWER adjusted p-values (<em>fwer</em>) and whether the FWER adjusted p-value is less than 0.05 (<em>significantFWER</em>).</p>
<div class="sourceCode" id="cb271"><html><body><pre class="r"><span class="co">## Load all the regions</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"analysisResults"</span>, <span class="st">"fullRegions.Rdata"</span>))

<span class="co">## Metadata columns</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="fu">mcols</span>(<span class="no">fullRegions</span>))</pre></body></html></div>
<pre><code>##  [1] "value"                      "area"                      
##  [3] "indexStart"                 "indexEnd"                  
##  [5] "cluster"                    "clusterL"                  
##  [7] "meanCoverage"               "meanfetal"                 
##  [9] "meanadult"                  "log2FoldChangeadultvsfetal"
## [11] "pvalues"                    "significant"               
## [13] "qvalues"                    "significantQval"           
## [15] "name"                       "annotation"                
## [17] "description"                "region"                    
## [19] "distance"                   "subregion"                 
## [21] "insideDistance"             "exonnumber"                
## [23] "nexons"                     "UTR"                       
## [25] "annoStrand"                 "geneL"                     
## [27] "codingL"                    "Geneid"                    
## [29] "subjectHits"                "fwer"                      
## [31] "significantFWER"</code></pre>
<p>Note that <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> only has the information for a given chromosome at a time, so <code><a href="../reference/mergeResults.html">mergeResults()</a></code> re-calculates the p-values and q-values using the information from all the chromosomes.</p>
</div>
<div id="fullannotatedregions" class="section level3">
<h3 class="hasAnchor">
<a href="#fullannotatedregions" class="anchor"></a>fullAnnotatedRegions</h3>
<p>In preparation for visually exploring the results, <code><a href="../reference/mergeResults.html">mergeResults()</a></code> will run <code><a href="../reference/annotateRegions.html">annotateRegions()</a></code> which counts how many known exons, introns and intergenic segments each candidate DER overlaps (by default with a minimum overlap of 20bp). <code><a href="../reference/annotateRegions.html">annotateRegions()</a></code> uses a summarized version of the genome annotation created with <code><a href="../reference/makeGenomicState.html">makeGenomicState()</a></code>. For this example, we can use the data included in <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> which is the summarized annotation of hg19 for chromosome 21.</p>
<div class="sourceCode" id="cb273"><html><body><pre class="r"><span class="co">## Load annotateRegions() output</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"analysisResults"</span>, <span class="st">"fullAnnotatedRegions.Rdata"</span>))

<span class="co">## Information stored</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">fullAnnotatedRegions</span>)</pre></body></html></div>
<pre><code>## [1] "countTable"     "annotationList"</code></pre>
<div class="sourceCode" id="cb275"><html><body><pre class="r"><span class="co">## Take a peak</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">fullAnnotatedRegions</span>, <span class="no">head</span>)</pre></body></html></div>
<pre><code>## $countTable
##   exon intergenic intron
## 1    1          0      0
## 2    1          0      0
## 3    1          0      0
## 4    1          0      0
## 5    0          1      0
## 6    1          0      0
## 
## $annotationList
## GRangesList object of length 6:
## $`1`
## GRanges object with 1 range and 4 metadata columns:
##        seqnames            ranges strand |   theRegion                 tx_id
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;         &lt;IntegerList&gt;
##   4871    chr21 47609038-47611149      - |        exon 73448,73449,73450,...
##                                     tx_name          gene
##                             &lt;CharacterList&gt; &lt;IntegerList&gt;
##   4871 uc002zij.3,uc002zik.2,uc002zil.2,...           170
##   -------
##   seqinfo: 1 sequence from hg19 genome
## 
## $`2`
## GRanges object with 1 range and 4 metadata columns:
##        seqnames            ranges strand |   theRegion         tx_id
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;IntegerList&gt;
##   1189    chr21 40194598-40196878      + |        exon   72757,72758
##                      tx_name          gene
##              &lt;CharacterList&gt; &lt;IntegerList&gt;
##   1189 uc002yxf.3,uc002yxg.4            77
##   -------
##   seqinfo: 1 sequence from hg19 genome
## 
## $`3`
## GRanges object with 1 range and 4 metadata columns:
##        seqnames            ranges strand |   theRegion                 tx_id
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;         &lt;IntegerList&gt;
##   2965    chr21 27252861-27254082      - |        exon 73066,73067,73068,...
##                                     tx_name          gene
##                             &lt;CharacterList&gt; &lt;IntegerList&gt;
##   2965 uc002ylz.3,uc002yma.3,uc002ymb.3,...           139
##   -------
##   seqinfo: 1 sequence from hg19 genome
## 
## ...
## &lt;3 more elements&gt;</code></pre>
</div>
</div>
</div>
<div id="chip-seq-differential-binding" class="section level1">
<h1 class="hasAnchor">
<a href="#chip-seq-differential-binding" class="anchor"></a>ChIP-seq differential binding</h1>
<p>As of version 1.5.27 <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> has parameters that allow smoothing of the single base-level F-statistics before determining DERs. This allows finding differentially bounded regions (peaks) using ChIP-seq data. In general, ChIP-seq studies are smaller than RNA-seq studies which means that the single base-level F-statistics approach is well suited for differential binding analysis.</p>
<p>To smooth the F-statistics use <code>smooth = TRUE</code> in <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>. The default smoothing function is <code><a href="https://rdrr.io/pkg/bumphunter/man/locfitByCluster.html">bumphunter::locfitByCluster()</a></code> and all its parameters can be passed specified in the call to <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>. In particular, the <code>minNum</code> and <code>bpSpan</code> arguments are important. We recommend setting <code>minNum</code> to the minimum read length and <code>bpSpan</code> to the average peak length expected in the ChIP-seq data being analyzed. Smoothing the F-statistics will take longer but not use significantly more memory than the default behavior. So take this into account when choosing the number of permutations to run.</p>
</div>
<div id="visually-explore-results" class="section level1">
<h1 class="hasAnchor">
<a href="#visually-explore-results" class="anchor"></a>Visually explore results</h1>
<p>Optionally, we can use the addon package <em><a href="https://bioconductor.org/packages/3.12/derfinderPlot">derfinderPlot</a></em> to visually explore the results.</p>
<p>To make the region level plots, we will need to extract the region level coverage data. We can do so using <code><a href="../reference/getRegionCoverage.html">getRegionCoverage()</a></code> as shown below.</p>
<div class="sourceCode" id="cb277"><html><body><pre class="r"><span class="co">## Find overlaps between regions and summarized genomic annotation</span>
<span class="no">annoRegs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/annotateRegions.html">annotateRegions</a></span>(<span class="no">fullRegions</span>, <span class="no">genomicState</span>$<span class="no">fullGenome</span>)</pre></body></html></div>
<pre><code>## 2020-05-14 15:57:22 annotateRegions: counting</code></pre>
<pre><code>## 2020-05-14 15:57:23 annotateRegions: annotating</code></pre>
<div class="sourceCode" id="cb280"><html><body><pre class="r"><span class="co">## Indeed, the result is the same because we only used chr21</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="no">annoRegs</span>, <span class="no">fullAnnotatedRegions</span>)</pre></body></html></div>
<pre><code>## [1] FALSE</code></pre>
<div class="sourceCode" id="cb282"><html><body><pre class="r"><span class="co">## Get the region coverage</span>
<span class="no">regionCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/getRegionCoverage.html">getRegionCoverage</a></span>(<span class="no">fullCov</span>, <span class="no">fullRegions</span>)</pre></body></html></div>
<pre><code>## 2020-05-14 15:57:23 getRegionCoverage: processing chr21</code></pre>
<pre><code>## 2020-05-14 15:57:23 getRegionCoverage: done processing chr21</code></pre>
<div class="sourceCode" id="cb285"><html><body><pre class="r"><span class="co">## Explore the result</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">regionCov</span><span class="kw">[[</span><span class="fl">1</span>]])</pre></body></html></div>
<pre><code>##   HSB113 HSB123 HSB126 HSB130 HSB135 HSB136 HSB145 HSB153 HSB159 HSB178 HSB92
## 1   0.68   0.44   0.48   0.36   0.19   2.34   1.29   1.77   2.21   2.69  1.89
## 2   0.60   0.44   0.48   0.36   0.19   2.30   1.29   1.77   2.21   2.64  1.86
## 3   0.60   0.40   0.48   0.32   0.19   2.39   1.37   1.81   2.31   2.69  1.89
## 4   0.64   0.40   0.48   0.32   0.19   2.61   1.42   1.89   2.36   2.88  1.89
## 5   0.64   0.40   0.48   0.36   0.19   2.65   1.42   1.93   2.36   2.88  1.96
## 6   0.60   0.44   0.48   0.39   0.23   2.65   1.59   1.93   2.36   2.83  1.93
##   HSB97
## 1  3.57
## 2  3.60
## 3  3.60
## 4  3.60
## 5  3.70
## 6  3.70</code></pre>
<p>With this, we are all set to visually explore the results.</p>
<div class="sourceCode" id="cb287"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinderPlot"</span>)

<span class="co">## Overview of the candidate DERs in the genome</span>
<span class="fu"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotOverview.html">plotOverview</a></span>(
    <span class="kw">regions</span> <span class="kw">=</span> <span class="no">fullRegions</span>, <span class="kw">annotation</span> <span class="kw">=</span> <span class="no">results</span>$<span class="no">annotation</span>,
    <span class="kw">type</span> <span class="kw">=</span> <span class="st">"fwer"</span>
)

<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"TxDb.Hsapiens.UCSC.hg19.knownGene"</span>))
<span class="no">txdb</span> <span class="kw">&lt;-</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>

<span class="co">## Base-levle coverage plots for the first 10 regions</span>
<span class="fu"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage</a></span>(
    <span class="kw">regions</span> <span class="kw">=</span> <span class="no">fullRegions</span>, <span class="kw">regionCoverage</span> <span class="kw">=</span> <span class="no">regionCov</span>,
    <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="no">pheno</span>$<span class="no">group</span>, <span class="kw">nearestAnnotation</span> <span class="kw">=</span> <span class="no">results</span>$<span class="no">annotation</span>,
    <span class="kw">annotatedRegions</span> <span class="kw">=</span> <span class="no">annoRegs</span>, <span class="kw">whichRegions</span> <span class="kw">=</span> <span class="fl">1</span>:<span class="fl">10</span>, <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">txdb</span>, <span class="kw">scalefac</span> <span class="kw">=</span> <span class="fl">1</span>,
    <span class="kw">ask</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)

<span class="co">## Cluster plot for the first region</span>
<span class="fu"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotCluster.html">plotCluster</a></span>(
    <span class="kw">idx</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">regions</span> <span class="kw">=</span> <span class="no">fullRegions</span>, <span class="kw">annotation</span> <span class="kw">=</span> <span class="no">results</span>$<span class="no">annotation</span>,
    <span class="kw">coverageInfo</span> <span class="kw">=</span> <span class="no">fullCov</span>$<span class="no">chr21</span>, <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">txdb</span>, <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="no">pheno</span>$<span class="no">group</span>,
    <span class="kw">titleUse</span> <span class="kw">=</span> <span class="st">"fwer"</span>
)</pre></body></html></div>
<p>The <a href="quickstart-derfinder.html">quick start to using derfinder</a> has example plots for the expressed regions-level approach. The vignette for <em><a href="https://bioconductor.org/packages/3.12/derfinderPlot">derfinderPlot</a></em> has even more examples.</p>
</div>
<div id="interactive-html-reports" class="section level1">
<h1 class="hasAnchor">
<a href="#interactive-html-reports" class="anchor"></a>Interactive HTML reports</h1>
<p>We have also developed an addon package called <em><a href="https://bioconductor.org/packages/3.12/regionReport">regionReport</a></em> available via <a href="http://bioconductor.org/packages/regionReport">Bioconductor</a>.</p>
<p>The function <code>derfinderReport()</code> in <em><a href="https://bioconductor.org/packages/3.12/regionReport">regionReport</a></em> basically takes advantage of the results from <code><a href="../reference/mergeResults.html">mergeResults()</a></code> and plotting functions available in <em><a href="https://bioconductor.org/packages/3.12/derfinderPlot">derfinderPlot</a></em> as well as other neat features from <em><a href="https://CRAN.R-project.org/package=knitrBootstrap">knitrBootstrap</a></em>. It then generates a customized report for single-base level F-statistics DER finding analyses.</p>
<p>For results from <code><a href="../reference/regionMatrix.html">regionMatrix()</a></code> or <code><a href="../reference/railMatrix.html">railMatrix()</a></code> use <code>renderReport()</code> from <em><a href="https://bioconductor.org/packages/3.12/regionReport">regionReport</a></em>. In both cases, the resulting HTML report promotes reproducibility of the analysis and allows you to explore in more detail the results through some diagnostic plots.</p>
<p>We think that these reports are very important when you are exploring the resulting DERs after changing a key parameter in <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>, <code><a href="../reference/regionMatrix.html">regionMatrix()</a></code> or <code><a href="../reference/railMatrix.html">railMatrix()</a></code>.</p>
<p>Check out the vignette for <em><a href="https://bioconductor.org/packages/3.12/regionReport">regionReport</a></em> for example reports generated with it.</p>
</div>
<div id="miscellaneous-features" class="section level1">
<h1 class="hasAnchor">
<a href="#miscellaneous-features" class="anchor"></a>Miscellaneous features</h1>
<p>In this section we go over some other features of <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> which can be useful for performing feature-counts based analyses, exploring the results, or exporting data.</p>
<div id="feature-level-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#feature-level-analysis" class="anchor"></a>Feature level analysis</h2>
<p>Similar to the expressed region-level analysis, you might be interested in performing a feature-level analysis. More specifically, this means getting a count matrix at the exon-level (or gene-level). <code><a href="../reference/coverageToExon.html">coverageToExon()</a></code> allows you to get such a matrix by taking advantage of the summarized annotation produced by <code><a href="../reference/makeGenomicState.html">makeGenomicState()</a></code>.</p>
<p>In this example, we use the genomic state included in the package which has the information for chr21 <em><a href="https://bioconductor.org/packages/3.12/TxDb.Hsapiens.UCSC.hg19.knownGene">TxDb.Hsapiens.UCSC.hg19.knownGene</a></em> annotation.</p>
<div class="sourceCode" id="cb288"><html><body><pre class="r"><span class="co">## Get the exon-level matrix</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">exonCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/coverageToExon.html">coverageToExon</a></span>(<span class="no">fullCov</span>, <span class="no">genomicState</span>$<span class="no">fullGenome</span>, <span class="kw">L</span> <span class="kw">=</span> <span class="fl">76</span>))</pre></body></html></div>
<pre><code>## class: SerialParam
##   bpisup: FALSE; bpnworkers: 1; bptasks: 0; bpjobname: BPJOB
##   bplog: FALSE; bpthreshold: INFO; bpstopOnError: TRUE
##   bpRNGseed: ; bptimeout: 2592000; bpprogressbar: FALSE
##   bpexportglobals: TRUE
##   bplogdir: NA
##   bpresultdir: NA
## class: SerialParam
##   bpisup: FALSE; bpnworkers: 1; bptasks: 0; bpjobname: BPJOB
##   bplog: FALSE; bpthreshold: INFO; bpstopOnError: TRUE
##   bpRNGseed: ; bptimeout: 2592000; bpprogressbar: FALSE
##   bpexportglobals: TRUE
##   bplogdir: NA
##   bpresultdir: NA</code></pre>
<pre><code>## 2020-05-14 15:57:27 coverageToExon: processing chromosome chr21</code></pre>
<pre><code>## class: SerialParam
##   bpisup: FALSE; bpnworkers: 1; bptasks: 0; bpjobname: BPJOB
##   bplog: FALSE; bpthreshold: INFO; bpstopOnError: TRUE
##   bpRNGseed: ; bptimeout: 2592000; bpprogressbar: FALSE
##   bpexportglobals: TRUE
##   bplogdir: NA
##   bpresultdir: NA</code></pre>
<pre><code>## 2020-05-14 15:57:30 coverageToExon: processing chromosome chr21</code></pre>
<pre><code>##    user  system elapsed 
##   5.936   0.881   6.819</code></pre>
<div class="sourceCode" id="cb294"><html><body><pre class="r"><span class="co">## Dimensions of the matrix</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">exonCov</span>)</pre></body></html></div>
<pre><code>## [1] 2658   12</code></pre>
<div class="sourceCode" id="cb296"><html><body><pre class="r"><span class="co">## Explore a little bit</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span>(<span class="no">exonCov</span>)</pre></body></html></div>
<pre><code>##         HSB113      HSB123      HSB126      HSB130       HSB135       HSB136
## 4983 0.1173684   0.1382895   0.0000000   0.0000000   0.07894737   0.03947368
## 4985 4.6542105   1.4557895   1.6034210   0.7798684   1.29592103   0.72986842
## 4986 7.1510526 291.0205261 252.4892106 152.1905258 439.69500020 263.63486853
## 4988 0.0000000   0.7063158   0.7223684   0.2639474   0.48657895   0.37657895
## 4990 2.3064474  64.9423686  69.6584212  35.5769737 101.76394746  76.96736838
## 4992 0.1652632   8.1834211   9.7538158   2.4193421  10.08618420  14.41013157
##            HSB145     HSB153      HSB159    HSB178       HSB92       HSB97
## 4983 5.263158e-03  0.1052632  0.04934211 0.1289474  0.03986842  0.44605264
## 4985 9.964474e-01 10.4906578  4.39013167 7.3119736  1.65184211 10.87723678
## 4986 2.345414e+02  4.5310526 10.64368415 4.9807895 11.40447366  6.23315790
## 4988 6.156579e-01  0.0000000  0.00000000 0.0000000  0.05644737  0.00000000
## 4990 5.978842e+01  1.2284210  2.67486840 0.9542105  3.98013159  1.96131579
## 4992 5.325658e+00  0.2402632  0.73039474 0.2601316  0.66907895  0.07473684</code></pre>
<p>With this matrix, rounded if necessary, you can proceed to use packages such as <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em>, <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em>, <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> among others.</p>
</div>
<div id="compare-results-visually" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-results-visually" class="anchor"></a>Compare results visually</h2>
<p>We can certainly make region-level plots using <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage()</a></code> or cluster plots using <code><a href="https://rdrr.io/pkg/derfinderPlot/man/plotCluster.html">plotCluster()</a></code> or overview plots using <code>plotOveview()</code>, all from <em><a href="https://bioconductor.org/packages/3.12/derfinderPlot">derfinderPlot</a></em>.</p>
<p>First we need to get the relevant annotation information.</p>
<div class="sourceCode" id="cb298"><html><body><pre class="r"><span class="co">## Annotate regions as exonic, intronic or intergenic</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">annoGenome</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/annotateRegions.html">annotateRegions</a></span>(
    <span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>,
    <span class="no">genomicState</span>$<span class="no">fullGenome</span>
))</pre></body></html></div>
<pre><code>## 2020-05-14 15:57:31 annotateRegions: counting</code></pre>
<pre><code>## 2020-05-14 15:57:31 annotateRegions: annotating</code></pre>
<pre><code>##    user  system elapsed 
##   0.178   0.000   0.178</code></pre>
<div class="sourceCode" id="cb302"><html><body><pre class="r"><span class="co">## Note that the genomicState object included in derfinder only has information</span>
<span class="co">## for chr21 (hg19).</span>

<span class="co">## Identify closest genes to regions</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"bumphunter"</span>))
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"TxDb.Hsapiens.UCSC.hg19.knownGene"</span>))
<span class="no">txdb</span> <span class="kw">&lt;-</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>
<span class="no">genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html">annotateTranscripts</a></span>(<span class="no">txdb</span>)</pre></body></html></div>
<pre><code>## No annotationPackage supplied. Trying org.Hs.eg.db.</code></pre>
<pre><code>## Loading required package: org.Hs.eg.db</code></pre>
<pre><code>## Warning in library(package, lib.loc = lib.loc, character.only = TRUE,
## logical.return = TRUE, : there is no package called 'org.Hs.eg.db'</code></pre>
<pre><code>## Could not load org.Hs.eg.db. Will continue without annotation</code></pre>
<pre><code>## Getting TSS and TSE.</code></pre>
<pre><code>## Getting CSS and CSE.</code></pre>
<pre><code>## Getting exons.</code></pre>
<div class="sourceCode" id="cb310"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">annoNear</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bumphunter/man/matchGenes.html">matchGenes</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>, <span class="no">genes</span>))</pre></body></html></div>
<pre><code>##    user  system elapsed 
##   2.491   0.001   2.489</code></pre>
<p>Now we can proceed to use <em><a href="https://bioconductor.org/packages/3.12/derfinderPlot">derfinderPlot</a></em> to make the region-level plots for the top 100 regions.</p>
<div class="sourceCode" id="cb312"><html><body><pre class="r"><span class="co">## Identify the top regions by highest total coverage</span>
<span class="no">top</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>$<span class="no">area</span>, <span class="kw">decreasing</span> <span class="kw">=</span> <span class="fl">TRUE</span>)[<span class="fl">1</span>:<span class="fl">100</span>]

<span class="co">## Base-level plots for the top 100 regions with transcript information</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinderPlot"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html">plotRegionCoverage</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>,
    <span class="kw">regionCoverage</span> <span class="kw">=</span> <span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">bpCoverage</span>,
    <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="no">pheno</span>$<span class="no">group</span>, <span class="kw">nearestAnnotation</span> <span class="kw">=</span> <span class="no">annoNear</span>,
    <span class="kw">annotatedRegions</span> <span class="kw">=</span> <span class="no">annoGenome</span>, <span class="kw">whichRegions</span> <span class="kw">=</span> <span class="no">top</span>, <span class="kw">scalefac</span> <span class="kw">=</span> <span class="fl">1</span>,
    <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">txdb</span>, <span class="kw">ask</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<p>However, we can alternatively use <em><a href="https://bioconductor.org/packages/3.12/epivizr">epivizr</a></em> to view the candidate DERs and the region matrix results in a genome browser.</p>
<div class="sourceCode" id="cb313"><html><body><pre class="r"><span class="co">## Load epivizr, it's available from Bioconductor</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"epivizr"</span>)

<span class="co">## Load data to your browser</span>
<span class="no">mgr</span> <span class="kw">&lt;-</span> <span class="fu">startEpiviz</span>()
<span class="no">ders_dev</span> <span class="kw">&lt;-</span> <span class="no">mgr</span>$<span class="fu">addDevice</span>(
    <span class="no">fullRegions</span>[<span class="fu"><a href="https://rdrr.io/r/base/logical.html">as.logical</a></span>(<span class="no">fullRegions</span>$<span class="no">significantFWER</span>)], <span class="st">"Candidate DERs"</span>
)
<span class="no">ders_potential_dev</span> <span class="kw">&lt;-</span> <span class="no">mgr</span>$<span class="fu">addDevice</span>(
    <span class="no">fullRegions</span>[!<span class="fu"><a href="https://rdrr.io/r/base/logical.html">as.logical</a></span>(<span class="no">fullRegions</span>$<span class="no">significantFWER</span>)], <span class="st">"Potential DERs"</span>
)
<span class="no">regs_dev</span> <span class="kw">&lt;-</span> <span class="no">mgr</span>$<span class="fu">addDevice</span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>, <span class="st">"Region Matrix"</span>)

<span class="co">## Go to a place you like in the genome</span>
<span class="no">mgr</span>$<span class="fu">navigate</span>(
    <span class="st">"chr21"</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>[<span class="no">top</span>[<span class="fl">1</span>]]) - <span class="fl">100</span>,
    <span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span>(<span class="no">regionMat</span>$<span class="no">chr21</span>$<span class="no">regions</span>[<span class="no">top</span>[<span class="fl">1</span>]]) + <span class="fl">100</span>
)

<span class="co">## Stop the navigation</span>
<span class="no">mgr</span>$<span class="fu">stopServer</span>()</pre></body></html></div>
</div>
<div id="export-coverage-to-bigwig-files" class="section level2">
<h2 class="hasAnchor">
<a href="#export-coverage-to-bigwig-files" class="anchor"></a>Export coverage to BigWig files</h2>
<p><em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> also includes <code><a href="../reference/createBw.html">createBw()</a></code> with related functions <code><a href="../reference/createBwSample.html">createBwSample()</a></code> and <code><a href="../reference/coerceGR.html">coerceGR()</a></code> to export the output of <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code> to BigWig files. These functions can be useful in the case where you start with BAM files and later on want to save the coverage data into BigWig files, which are generally smaller.</p>
<div class="sourceCode" id="cb314"><html><body><pre class="r"><span class="co">## Subset only the first sample</span>
<span class="no">fullCovSmall</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">fullCov</span>, <span class="st">"["</span>, <span class="fl">1</span>)

<span class="co">## Export to BigWig</span>
<span class="no">bw</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/createBw.html">createBw</a></span>(<span class="no">fullCovSmall</span>)</pre></body></html></div>
<pre><code>## 2020-05-14 15:57:41 coerceGR: coercing sample HSB113</code></pre>
<pre><code>## 2020-05-14 15:57:41 createBwSample: exporting bw for sample HSB113</code></pre>
<div class="sourceCode" id="cb317"><html><body><pre class="r"><span class="co">## See the file. Note that the sample name is used to name the file.</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="kw">pattern</span> <span class="kw">=</span> <span class="st">".bw"</span>)</pre></body></html></div>
<pre><code>## [1] "HSB113.bw"    "meanChr21.bw"</code></pre>
<div class="sourceCode" id="cb319"><html><body><pre class="r"><span class="co">## Internally createBw() coerces each sample to a GRanges object before</span>
<span class="co">## exporting to a BigWig file. If more than one sample was exported, the</span>
<span class="co">## GRangesList would have more elements.</span>
<span class="no">bw</span></pre></body></html></div>
<pre><code>## GRangesList object of length 1:
## $HSB113
## GRanges object with 155950 ranges and 1 metadata column:
##         seqnames            ranges strand |     score
##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   chr21    chr21   9458667-9458741      * |      0.04
##   chr21    chr21   9540957-9540971      * |      0.04
##   chr21    chr21   9543719-9543778      * |      0.04
##   chr21    chr21   9651480-9651554      * |      0.04
##   chr21    chr21   9653397-9653471      * |      0.04
##     ...      ...               ...    ... .       ...
##   chr21    chr21 48093246-48093255      * |      0.04
##   chr21    chr21 48093257-48093331      * |      0.04
##   chr21    chr21 48093350-48093424      * |      0.04
##   chr21    chr21 48112194-48112268      * |      0.04
##   chr21    chr21 48115056-48115130      * |      0.04
##   -------
##   seqinfo: 1 sequence from an unspecified genome</code></pre>
</div>
</div>
<div id="advanced-arguments" class="section level1">
<h1 class="hasAnchor">
<a href="#advanced-arguments" class="anchor"></a>Advanced arguments</h1>
<p>If you are interested in using advanced arguments in <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em>, they are described in the manual pages of each function. Some of the most common advanced arguments are:</p>
<ul>
<li>
<code>chrsStyle</code> (default is <code>UCSC</code>)</li>
<li>
<code>verbose</code> (by default <code>TRUE</code>).</li>
</ul>
<p><code>verbose</code> controls whether to print status updates for nearly all the functions. <code>chrsStyle</code> is used to determine the chromosome naming style and is powered by <em><a href="https://bioconductor.org/packages/3.12/GenomeInfoDb">GenomeInfoDb</a></em>. Note that <code>chrsStyle</code> is used in any of the functions that call <code><a href="../reference/extendedMapSeqlevels.html">extendedMapSeqlevels()</a></code>. If you are working with a different organism than <em>Homo sapiens</em> set the global <code>species</code> option using <code><a href="https://rdrr.io/r/base/options.html">options(species = 'your species')</a></code> with the syntax used in <code><a href="https://rdrr.io/r/base/names.html">names(GenomeInfoDb::genomeStyles())</a></code>. If you want to disable <code><a href="../reference/extendedMapSeqlevels.html">extendedMapSeqlevels()</a></code> set <code>chrsStyle</code> to <code>NULL</code>, which can be useful if your organism is not part of <em><a href="https://bioconductor.org/packages/3.12/GenomeInfoDb">GenomeInfoDb</a></em>.</p>
<p>The third commonly used advanced argument is <code>mc.cores</code>. It controls the number of cores to use for the functions that can run with more than one core to speed up. In nearly all the cases, the maximum number of cores depends on the number of chromosomes. One notable exception is <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> where the maximum number of cores depends on the <code>chunksize</code> used and the dimensions of the data for the chromosome under study.</p>
<p>Note that using the <code>...</code> argument allows you to specify some of the documented arguments. For example, you might want to control the <code>maxClusterGap</code> from <code><a href="../reference/findRegions.html">findRegions()</a></code> in the <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> call.</p>
</div>
<div id="non-human-data" class="section level1">
<h1 class="hasAnchor">
<a href="#non-human-data" class="anchor"></a>Non-human data</h1>
<p>If you are working with data from an organism that is not <em>Homo sapiens</em>, then set the global options defining the <code>species</code> and the <code>chrsStyle</code> used. For example, if you are working with <em>Arabidopsis Thaliana</em> and the <em>NCBI</em> naming style, then set the options using the following code:</p>
<div class="sourceCode" id="cb321"><html><body><pre class="r"><span class="co">## Set global species and chrsStyle options</span>
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="kw">species</span> <span class="kw">=</span> <span class="st">"arabidopsis_thaliana"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="kw">chrsStyle</span> <span class="kw">=</span> <span class="st">"NCBI"</span>)

<span class="co">## Then proceed to load and analyze the data</span></pre></body></html></div>
<p>Internally <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> uses <code><a href="../reference/extendedMapSeqlevels.html">extendedMapSeqlevels()</a></code> to use the appropriate chromosome naming style given a species in all functions involving chromosome names.</p>
<p>Further note that the argument <code>txdb</code> from <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> is passed to <code><a href="https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html">bumphunter::annotateTranscripts(txdb)</a></code>. So if you are using a genome different from <strong>hg19</strong> remember to provide the appropriate annotation data or simply use <code><a href="../reference/analyzeChr.html">analyzeChr(runAnnotation = FALSE)</a></code>.</p>
<p>So, in the <em>Arabidopsis Thaliana</em> example, your <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> call would look like this:</p>
<div class="sourceCode" id="cb322"><html><body><pre class="r"><span class="co">## Load transcript database information</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"TxDb.Athaliana.BioMart.plantsmart28"</span>)

<span class="co">## Set organism options</span>
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="kw">species</span> <span class="kw">=</span> <span class="st">"arabidopsis_thaliana"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span>(<span class="kw">chrsStyle</span> <span class="kw">=</span> <span class="st">"NCBI"</span>)

<span class="co">## Run command with more arguments</span>
<span class="fu"><a href="../reference/analyzeChr.html">analyzeChr</a></span>(<span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Athaliana.BioMart.plantsmart28</span>)</pre></body></html></div>
<p>You might find the discussion <a href="https://support.bioconductor.org/p/62781/">Using bumphunter with non-human genomes</a> useful.</p>
</div>
<div id="functions-that-use-multiple-cores" class="section level1">
<h1 class="hasAnchor">
<a href="#functions-that-use-multiple-cores" class="anchor"></a>Functions that use multiple cores</h1>
<p>Currently, the following functions can use multiple cores, several of which are called inside <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>.</p>
<ul>
<li>
<code><a href="../reference/calculatePvalues.html">calculatePvalues()</a></code>: 1 core per chunk of data to process.</li>
<li>
<code><a href="../reference/calculateStats.html">calculateStats()</a></code>: 1 core per chunk of data to process.</li>
<li>
<code><a href="../reference/coerceGR.html">coerceGR()</a></code>: 1 core per chromosome. This function is used by <code><a href="../reference/createBw.html">createBw()</a></code>.</li>
<li>
<code><a href="../reference/coverageToExon.html">coverageToExon()</a></code>: 1 core per strand, then 1 core per chromosome.</li>
<li>
<code><a href="../reference/loadCoverage.html">loadCoverage()</a></code>: up to 1 core per tile when loading the data with <a href="http://www.bioconductor.org/packages/release/bioc/html/GenomicFiles.html">GenomicFiles</a>. Otherwise, no parallelization is used.</li>
<li>
<code><a href="../reference/fullCoverage.html">fullCoverage()</a></code>: 1 core per chromosome. In general, try to avoid using more than 10 cores as you might reach your maximum network speed and/or hard disk input/output seed. For the case described in <code><a href="../reference/loadCoverage.html">loadCoverage()</a></code>, you can specify how many cores to use per chromosome for the tiles using the <code>mc.cores.load</code> argument effectively resulting in <code>mc.cores</code> times <code>mc.cores.load</code> used (otherwise it’s <code>mc.cores</code> squared).</li>
<li>
<code><a href="../reference/getRegionCoverage.html">getRegionCoverage()</a></code>: 1 core per chromosome.</li>
<li>
<code><a href="../reference/regionMatrix.html">regionMatrix()</a></code>: 1 core per chromosome.</li>
<li>
<code><a href="../reference/railMatrix.html">railMatrix()</a></code>: 1 core per chromosome.</li>
</ul>
<p>All parallel operations use <code>SnowParam()</code> from <em><a href="https://bioconductor.org/packages/3.12/BiocParallel">BiocParallel</a></em> when more than 1 core is being used. Otherwise, <code>SerialParam()</code> is used. Note that if you prefer to specify other types of parallelization you can do so by specifying the <code>BPPARAM.custom</code> advanced argument.</p>
<p>Because <code>SnowParam()</code> requires <code>R</code> to load the necessary packages on each worker, the key function <code>fstats.apply()</code> was isolated in the <em><a href="https://bioconductor.org/packages/3.12/derfinderHelper">derfinderHelper</a></em> package. This package has much faster loading speeds than <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> which greatly impacts performance on cases where the actual step of calculating the F-statistics is fast.</p>
<p>You may prefer to use <code>MulticoreParam()</code> described in the <em><a href="https://bioconductor.org/packages/3.12/BiocParallel">BiocParallel</a></em> vignette. In that case, when using these functions use <code>BPPARAM.custom = MulticoreParam(workers = x)</code> where <code>x</code> is the number of cores you want to use. Note that in some systems, as is the case of the cluster used by <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em>’s developers, the system tools for assessing memory usage can be misleading, thus resulting in much higher memory loads when using <code>MulticoreParam()</code> instead of the default <code>SnowParam()</code>.</p>
</div>
<div id="loading-data-details" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-data-details" class="anchor"></a>Loading data details</h1>
<div id="controlling-loading-from-bam-files" class="section level2">
<h2 class="hasAnchor">
<a href="#controlling-loading-from-bam-files" class="anchor"></a>Controlling loading from BAM files</h2>
<p>If you are loading data from BAM files, you might want to specify some criteria to decide which reads to include or not. For example, your data might have been generated by a strand-specific protocol. You can do so by specifying the arguments of <code>scanBamFlag()</code> from <em><a href="https://bioconductor.org/packages/3.12/Rsamtools">Rsamtools</a></em>.</p>
<p>You can also control whether to include or exclude bases with <code>CIGAR</code> string <code>D</code> (deletion from the reference) by setting the advanced argument <code>drop.D = TRUE</code> in your <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code> or <code><a href="../reference/loadCoverage.html">loadCoverage()</a></code> call.</p>
</div>
<div id="unfiltered-base-level-coverage" class="section level2">
<h2 class="hasAnchor">
<a href="#unfiltered-base-level-coverage" class="anchor"></a>Unfiltered base-level coverage</h2>
<p>Note that in most scenarios, the <code>fullCov</code> object illustrated in the introductory vignette can be large in memory. When making plots or calculating the region-level coverage, we don’t need the full information. In such situations, it might pay off to create a smaller version by loading only the required data. This can be achieved using the advanced argument <code>which</code> to <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code> or <code><a href="../reference/loadCoverage.html">loadCoverage()</a></code>.</p>
<p>However, it is important to consider that when reading the data from BAM files, a read might align partially inside the region of interest. By default such a read would be discarded and thus the base-level coverage would be lower than what it is in reality. The advanced argument <code>protectWhich</code> extends regions by 30 kbp (15 kbp each side) to help mitigate this issue.</p>
<p>We can illustrate this issue with the example data from <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em>. First, we load in the data and generate some regions of interest.</p>
<div class="sourceCode" id="cb323"><html><body><pre class="r"><span class="co">## Find some regions to work with</span>
<span class="fu"><a href="https://rdrr.io/r/utils/example.html">example</a></span>(<span class="st">"loadCoverage"</span>, <span class="st">"derfinder"</span>)</pre></body></html></div>
<pre><code>## 
## ldCvrg&gt; datadir &lt;- system.file("extdata", "genomeData", package = "derfinder")
## 
## ldCvrg&gt; files &lt;- rawFiles(
## ldCvrg+     datadir = datadir, samplepatt = "*accepted_hits.bam$",
## ldCvrg+     fileterm = NULL
## ldCvrg+ )
## 
## ldCvrg&gt; ## Shorten the column names
## ldCvrg&gt; names(files) &lt;- gsub("_accepted_hits.bam", "", names(files))
## 
## ldCvrg&gt; ## Read and filter the data, only for 2 files
## ldCvrg&gt; dataSmall &lt;- loadCoverage(files = files[1:2], chr = "21", cutoff = 0)</code></pre>
<pre><code>## 2020-05-14 15:57:42 loadCoverage: finding chromosome lengths</code></pre>
<pre><code>## 2020-05-14 15:57:42 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009101_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:42 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009102_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:42 loadCoverage: applying the cutoff to the merged data</code></pre>
<pre><code>## 2020-05-14 15:57:42 filterData: originally there were 48129895 rows, now there are 454 rows. Meaning that 100 percent was filtered.</code></pre>
<pre><code>## 
## ldCvrg&gt; ## Not run: 
## ldCvrg&gt; ##D ## Export to BigWig files
## ldCvrg&gt; ##D createBw(list("chr21" = dataSmall))
## ldCvrg&gt; ##D 
## ldCvrg&gt; ##D ## Load data from BigWig files
## ldCvrg&gt; ##D dataSmall.bw &lt;- loadCoverage(c(
## ldCvrg&gt; ##D     ERR009101 = "ERR009101.bw", ERR009102 =
## ldCvrg&gt; ##D         "ERR009102.bw"
## ldCvrg&gt; ##D ), chr = "chr21")
## ldCvrg&gt; ##D 
## ldCvrg&gt; ##D ## Compare them
## ldCvrg&gt; ##D mapply(function(x, y) {
## ldCvrg&gt; ##D     x - y
## ldCvrg&gt; ##D }, dataSmall$coverage, dataSmall.bw$coverage)
## ldCvrg&gt; ##D 
## ldCvrg&gt; ##D ## Note that the only difference is the type of Rle (integer vs numeric) used
## ldCvrg&gt; ##D ## to store the data.
## ldCvrg&gt; ## End(Not run)
## ldCvrg&gt; 
## ldCvrg&gt; 
## ldCvrg&gt; 
## ldCvrg&gt;</code></pre>
<div class="sourceCode" id="cb331"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/example.html">example</a></span>(<span class="st">"getRegionCoverage"</span>, <span class="st">"derfinder"</span>)</pre></body></html></div>
<pre><code>## 
## gtRgnC&gt; ## Obtain fullCov object
## gtRgnC&gt; fullCov &lt;- list("21" = genomeDataRaw$coverage)
## 
## gtRgnC&gt; ## Assign chr lengths using hg19 information, use only first two regions
## gtRgnC&gt; library("GenomicRanges")
## 
## gtRgnC&gt; regions &lt;- genomeRegions$regions[1:2]
## 
## gtRgnC&gt; seqlengths(regions) &lt;- seqlengths(getChromInfoFromUCSC("hg19",
## gtRgnC+     as.Seqinfo = TRUE
## gtRgnC+ ))[
## gtRgnC+     mapSeqlevels(names(seqlengths(regions)), "UCSC")
## gtRgnC+ ]
## 
## gtRgnC&gt; ## Finally, get the region coverage
## gtRgnC&gt; regionCov &lt;- getRegionCoverage(fullCov = fullCov, regions = regions)</code></pre>
<pre><code>## extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</code></pre>
<pre><code>## 2020-05-14 15:57:43 getRegionCoverage: processing chr21</code></pre>
<pre><code>## 2020-05-14 15:57:43 getRegionCoverage: done processing chr21</code></pre>
<p>Next, we load the coverage again using <code>which</code> but without any padding. We can see how the coverage is not the same by looking at the maximum coverage for each sample.</p>
<div class="sourceCode" id="cb336"><html><body><pre class="r"><span class="co">## Illustrate reading data from a set of regions</span>
<span class="no">test</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/loadCoverage.html">loadCoverage</a></span>(
    <span class="kw">files</span> <span class="kw">=</span> <span class="no">files</span>, <span class="kw">chr</span> <span class="kw">=</span> <span class="st">"21"</span>, <span class="kw">cutoff</span> <span class="kw">=</span> <span class="kw">NULL</span>, <span class="kw">which</span> <span class="kw">=</span> <span class="no">regions</span>,
    <span class="kw">protectWhich</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">fileStyle</span> <span class="kw">=</span> <span class="st">"NCBI"</span>
)</pre></body></html></div>
<pre><code>## extendedMapSeqlevels: sequence names mapped from UCSC to NCBI for species homo_sapiens</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: finding chromosome lengths</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009101_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009102_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009105_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009107_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009108_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009112_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009115_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009116_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009131_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009138_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009144_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009145_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009148_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009151_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:43 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009152_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009153_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009159_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009161_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009163_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009164_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009167_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031812_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031835_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031867_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031868_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031900_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031904_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031914_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031936_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031958_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031960_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: applying the cutoff to the merged data</code></pre>
<pre><code>## 2020-05-14 15:57:44 filterData: originally there were 48129895 rows, now there are 48129895 rows. Meaning that 0 percent was filtered.</code></pre>
<div class="sourceCode" id="cb372"><html><body><pre class="r"><span class="co">## Some reads were ignored and thus the coverage is lower as can be seen below:</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">test</span>$<span class="no">coverage</span>, <span class="no">max</span>) - <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">genomeDataRaw</span>$<span class="no">coverage</span>, <span class="no">max</span>)</pre></body></html></div>
<pre><code>## ERR009101 ERR009102 ERR009105 ERR009107 ERR009108 ERR009112 ERR009115 ERR009116 
##         0         0         0         0        -1         0        -1        -2 
## ERR009131 ERR009138 ERR009144 ERR009145 ERR009148 ERR009151 ERR009152 ERR009153 
##        -2        -2        -1        -1        -3        -3         0        -3 
## ERR009159 ERR009161 ERR009163 ERR009164 ERR009167 SRR031812 SRR031835 SRR031867 
##        -3        -3        -1        -3        -3         0        -1         0 
## SRR031868 SRR031900 SRR031904 SRR031914 SRR031936 SRR031958 SRR031960 
##         0         0        -1         0         0         0         0</code></pre>
<p>When we re-load the data using some padding to the regions, we find that the coverage matches at all the bases.</p>
<div class="sourceCode" id="cb374"><html><body><pre class="r"><span class="co">## Illustrate reading data from a set of regions</span>

<span class="no">test2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/loadCoverage.html">loadCoverage</a></span>(
    <span class="kw">files</span> <span class="kw">=</span> <span class="no">files</span>, <span class="kw">chr</span> <span class="kw">=</span> <span class="st">"21"</span>, <span class="kw">cutoff</span> <span class="kw">=</span> <span class="kw">NULL</span>,
    <span class="kw">which</span> <span class="kw">=</span> <span class="no">regions</span>, <span class="kw">protectWhich</span> <span class="kw">=</span> <span class="fl">3e4</span>, <span class="kw">fileStyle</span> <span class="kw">=</span> <span class="st">"NCBI"</span>
)</pre></body></html></div>
<pre><code>## extendedMapSeqlevels: sequence names mapped from UCSC to NCBI for species homo_sapiens</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: finding chromosome lengths</code></pre>
<pre><code>## 2020-05-14 15:57:44 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009101_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009102_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009105_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009107_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009108_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009112_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009115_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009116_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009131_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009138_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009144_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009145_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009148_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009151_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009152_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009153_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009159_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009161_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009163_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:45 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009164_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/ERR009167_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031812_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031835_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031867_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031868_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031900_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031904_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031914_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031936_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031958_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: loading BAM file /usr/local/lib/R/host-site-library/derfinder/extdata/genomeData/SRR031960_accepted_hits.bam</code></pre>
<pre><code>## 2020-05-14 15:57:46 loadCoverage: applying the cutoff to the merged data</code></pre>
<pre><code>## 2020-05-14 15:57:46 filterData: originally there were 48129895 rows, now there are 48129895 rows. Meaning that 0 percent was filtered.</code></pre>
<div class="sourceCode" id="cb410"><html><body><pre class="r"><span class="co">## Adding some padding to the regions helps get the same coverage</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">test2</span>$<span class="no">coverage</span>, <span class="no">max</span>), <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">genomeDataRaw</span>$<span class="no">coverage</span>, <span class="no">max</span>))</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb412"><html><body><pre class="r"><span class="co">## A more detailed test reveals that the coverage matches at every base</span>
<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span>(<span class="kw">function</span>(<span class="no">x</span>, <span class="no">y</span>) {
    <span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="no">x</span>, <span class="no">y</span>)
}, <span class="no">test2</span>$<span class="no">coverage</span>, <span class="no">genomeDataRaw</span>$<span class="no">coverage</span>))</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<p>How much padding you need to use will depend on your specific data set, and you might be comfortable getting approximately the same coverage values for the sake of greatly reducing the memory resources needed.</p>
</div>
<div id="input-files-in-a-different-naming-style" class="section level2">
<h2 class="hasAnchor">
<a href="#input-files-in-a-different-naming-style" class="anchor"></a>Input files in a different naming style</h2>
<p>If you are under the case where you like to use a specific chromosome naming style but the raw data files use another one, you might need to use the <code>fileStyle</code> argument.</p>
<p>For example, you could be working with <em>Homo sapiens</em> data and your preferred naming style is <em>UCSC</em> (chr1, chr2, …, chrX, chrY) but the raw data uses <em>NCBI</em> style names (1, 2, …, X, Y). In that case, use <code><a href="../reference/fullCoverage.html">fullCoverage(fileStyle = 'NCBI')</a></code> or <code><a href="../reference/loadCoverage.html">loadCoverage(fileStyle = 'NCBI')</a></code> depending if you are loading one chromosome or multiple at a time.</p>
</div>
<div id="loading-data-in-chunks" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-data-in-chunks" class="anchor"></a>Loading data in chunks</h2>
<p>If you prefer to do so, <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code> and <code><a href="../reference/loadCoverage.html">loadCoverage()</a></code> can load the data of a chromosome in chunks using <em><a href="https://bioconductor.org/packages/3.12/GenomicFiles">GenomicFiles</a></em>. This is controlled by whether you specify the <code>tilewidth</code> argument.</p>
<p>Notice that you might run into slight coverage errors near the borders of the tiles for the same reason that was illustrated previously when loading specific regions.</p>
<p>This approach is not necessarily more efficient and can be significantly time consuming if you use a small <code>tilewidth</code>.</p>
</div>
<div id="large-number-of-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#large-number-of-samples" class="anchor"></a>Large number of samples</h2>
<p>If you have a large number of samples (say thousands), it might be best to submit cluster jobs that run <code><a href="../reference/loadCoverage.html">loadCoverage()</a></code> or <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code> for only one chromosome at a time.</p>
<p>In the case of working with <a href="http://rail.bio">Rail</a> output, you can either use <code><a href="../reference/railMatrix.html">railMatrix()</a></code> with the argument <code>file.cores</code> greater than 1 or specify the advanced argument <code>BPPARAM.railChr</code> to control the parallel environment used for loading the BigWig files. Doing so, you can then submit cluster jobs that run <code><a href="../reference/railMatrix.html">railMatrix()</a></code> for one chromosome at a time, yet read in the data fast. You can do all of from R by using <code>BPPARAM.custom</code> and <code>BPPARAM.railChr</code> at the same time where you use a <code>BatchJobsParam()</code> for the first one.</p>
<p>Another option when working with <a href="http://rail.bio">Rail</a> output is to simply load the summary BigWig data (mean, median), then define the ERs using <code><a href="../reference/findRegions.html">findRegions()</a></code> and write them to a BED file. You can then use <a href="https://github.com/CRG-Barcelona/bwtool/wiki">bwtool</a> to create the coverage matrix.</p>
</div>
</div>
<div id="flow-charts" class="section level1">
<h1 class="hasAnchor">
<a href="#flow-charts" class="anchor"></a>Flow charts</h1>
<div id="der-analysis-flow-chart" class="section level2">
<h2 class="hasAnchor">
<a href="#der-analysis-flow-chart" class="anchor"></a>DER analysis flow chart</h2>
<p>Figure @ref(fig:Figure1) illustrates how most of <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em>’s functions interact when performing a base-level differential expression analysis by calculating base-level F-statistics.</p>
<div class="figure" style="text-align: center">
<img src="http://lcolladotor.github.io/derfinder/fig/DERpathway.png" alt="derfinder F-statistics flow chart." width="100%"><p class="caption">
derfinder F-statistics flow chart.
</p>
</div>
<p>Flow chart of the different processing steps (black boxes) that can be carried out using <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> and the functions that perform these actions (in red). Input and output is shown in green boxes. Functions in blue are those applied to the results from multiple chromosomes (<code><a href="../reference/mergeResults.html">mergeResults()</a></code> and <code>derfinderReport</code>). <em><a href="https://bioconductor.org/packages/3.12/regionReport">regionReport</a></em> functions are shown in orange while <em><a href="https://bioconductor.org/packages/3.12/derfinderPlot">derfinderPlot</a></em> functions are shown in dark purple. Purple dotted arrow marks functions that require unfiltered base-level coverage.</p>
</div>
<div id="analyzechr-flow-chart" class="section level2">
<h2 class="hasAnchor">
<a href="#analyzechr-flow-chart" class="anchor"></a><code>analyzeChr()</code> flow chart</h2>
<div class="figure" style="text-align: center">
<img src="http://lcolladotor.github.io/derfinder/fig/analyzeChr.png" alt="analyzeChr() flow chart." width="100%"><p class="caption">
analyzeChr() flow chart.
</p>
</div>
<p>Figure @ref(fig:Figure2) shows in more detail the processing flow in <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>, which is the main function for identifying candidate differentially expressed regions (DERs) from the base-level F-statistics.</p>
<p>Many fine-tunning arguments can be passed to <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> to feed into the other functions. For example, you might want to use a smaller <code>chunksize</code> when pre-processing the coverage data (the default is 5 million): specially if you have hundreds of samples.</p>
<p>Another useful argument is <code>scalefac</code> (by default it’s 32) which controls the scaling factor to use before the log2 transformation.</p>
<p>Furthermore, you might want to specify <code>maxClusterGap</code> to control the maximum gap between two regions before they are considered to be part of the same cluster.</p>
</div>
<div id="regionmatrix-flow-chart" class="section level2">
<h2 class="hasAnchor">
<a href="#regionmatrix-flow-chart" class="anchor"></a><code>regionMatrix()</code> flow chart</h2>
<div class="figure" style="text-align: center">
<img src="http://lcolladotor.github.io/derfinder/fig/regionMatrix.png" alt="regionMatrix() flow chart." width="100%"><p class="caption">
regionMatrix() flow chart.
</p>
</div>
<p>Figure @ref(fig:Figure3) shows the functions used internally by <code><a href="../reference/regionMatrix.html">regionMatrix()</a></code> and processing steps for identifying expressed regions. Overall, it is much simpler than <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>. The flow is even simpler in <code><a href="../reference/railMatrix.html">railMatrix()</a></code>.</p>
</div>
</div>
<div id="base-level-f-statistics-projects" class="section level1">
<h1 class="hasAnchor">
<a href="#base-level-f-statistics-projects" class="anchor"></a>Base-level F-statistics projects</h1>
<p>For each project where you will calculating base-level F-statistics, we recommend the following organization. You might be interested in a similar structure when using <code><a href="../reference/regionMatrix.html">regionMatrix()</a></code> if you have BAM files. The notable exception is when you are analyzing output data from <a href="http://rail.bio">Rail</a> in which case you won’t be needing this type of organization. Remember that <code><a href="../reference/railMatrix.html">railMatrix()</a></code> is much less computationally intensive than <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>.</p>
<div id="file-organization" class="section level2">
<h2 class="hasAnchor">
<a href="#file-organization" class="anchor"></a>File organization</h2>
<p>This is our recommended file organization when using <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>.</p>
<pre><code>ProjectDir
|-derCoverageInfo
|-derAnalysis
|---analysis01
|---analysis02</code></pre>
<p>We start with a main project directory that has two initial directories. One for storing the coverage data, and one for storing each analysis: you might explore different models, cutoffs or other parameters.</p>
<p>You can then use <code><a href="../reference/fullCoverage.html">fullCoverage()</a></code>, save the result and also save the filtered coverage information for each chromosome separately. Doing so will result in the following structure.</p>
<pre><code>ProjectDir
|-derCoverageInfo
|---Chr1Cov.Rdata
|---Chr2Cov.Rdata
...
|---ChrYCov.Rdata
|---fullCov.Rdata
|-derAnalysis
|---analysis01
|---analysis02</code></pre>
<p>Next, you can use <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> for each of the chromosomes of a specific analysis (say <em>analysis01</em>). Doing so will create several Rdata files per chromosome as shown below. <code>bash</code> scripts can be useful if you wish to submit one cluster job per chromosome. In general, you will use the same model and group information for each chromosome, so saving the information can be useful.</p>
<pre><code>ProjectDir
|-derCoverageInfo
|---Chr1Cov.Rdata
|---Chr2Cov.Rdata
...
|---ChrYCov.Rdata
|---fullCov.Rdata
|-derAnalysis
|---analysis01
|-----models.Rdata
|-----groupInfo.Rdata
|-----chr1/
|-------chunksDir/
|-------logs/
|-------annotation.Rdata
|-------coveragePrep.Rdata
|-------fstats.Rdata
|-------optionsStats.Rdata
|-------regions.Rdata
|-------timeinfo.Rdata
|-----chr2/
...
|-----chrY/
|---analysis02</code></pre>
<p>Then use <code><a href="../reference/mergeResults.html">mergeResults()</a></code> to pool together the results from all the chromosomes for a given analysis (here <em>analysis01</em>).</p>
<pre><code>ProjectDir
|-derCoverageInfo
|---Chr1Cov.Rdata
|---Chr2Cov.Rdata
...
|---ChrYCov.Rdata
|---fullCov.Rdata
|-derAnalysis
|---analysis01
|-----logs/
|-----fullAnnotatedRegions.Rdata
|-----fullFstats.Rdata
|-----fullNullSummary.Rdata
|-----fullRegions.Rdata
|-----fullTime.Rdata
|-----optionsMerge.Rdata
|-----chr1/
|-------chunksDir/
|-------logs/
|-------annotation.Rdata
|-------coveragePrep.Rdata
|-------fstats.Rdata
|-------optionsStats.Rdata
|-------regions.Rdata
|-------timeinfo.Rdata
|-----chr2/
...
|-----chrY/
|---analysis02</code></pre>
<p>Finally, you might want to use <code>derfinderReport()</code> from <em><a href="https://bioconductor.org/packages/3.12/regionReport">regionReport</a></em> to create a HTML report of the results.</p>
</div>
<div id="bash-scripts" class="section level2">
<h2 class="hasAnchor">
<a href="#bash-scripts" class="anchor"></a>bash scripts</h2>
<p>If you are following our recommended file organization for a base-level F-statistics DER finding analysis, you might be interested in the following <code>bash</code> scripts templates. Note that there are plenty of other solutions, such as writing a master <code>R</code> script and using <em><a href="https://bioconductor.org/packages/3.12/BiocParallel">BiocParallel</a></em> to submit jobs and administer them for you. Check out the <code>BatchJobsParam()</code> for more information. If you want to use <code>bash</code> scripts, take a look below.</p>
<p>For interacting between <code>bash</code> and <code>R</code> we have found quite useful the <em><a href="https://CRAN.R-project.org/package=getopt">getopt</a></em> package. Here we include an example <code>R</code> script that is controlled by a <code>bash</code> script which submits a job for each chromosome to analyze for a given analysis.</p>
<p>The two files, <em>derfinderAnalysis.R</em> and <em>derAnalysis.sh</em> should live under the <em>derAnalysis</em> directory.</p>
<pre><code>ProjectDir
|-derCoverageInfo
|---Chr1Cov.Rdata
|---Chr2Cov.Rdata
...
|---ChrYCov.Rdata
|---fullCov.Rdata
|-derAnalysis
|---derfinder-Analysis.R
|---derAnalysis.sh
|---analysis01
|---analysis02</code></pre>
<p>Then, you can simply use:</p>
<div class="sourceCode" id="cb419"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb419-1" data-line-number="1"><span class="bu">cd</span> /ProjectDir/derAnalysis</a>
<a class="sourceLine" id="cb419-2" data-line-number="2"><span class="fu">sh</span> derAnalysis.sh analysis01</a></code></pre></div>
<p>to run <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code> on all your chromosomes.</p>
<div id="derfinder-analysis-r" class="section level3">
<h3 class="hasAnchor">
<a href="#derfinder-analysis-r" class="anchor"></a>derfinder-analysis.R</h3>
<div class="sourceCode" id="cb420"><html><body><pre class="r"><span class="co">## Run derfinder's analysis steps with timing info</span>

<span class="co">## Load libraries</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"getopt"</span>)

<span class="co">## Available at http:/bioconductor.org/packages/derfinder</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinder"</span>)

<span class="co">## Specify parameters</span>
<span class="no">spec</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"DFfile"</span>, <span class="st">"d"</span>, <span class="fl">1</span>, <span class="st">"character"</span>, <span class="st">"path to the .Rdata file with the results from loadCoverage()"</span>,
    <span class="st">"chr"</span>, <span class="st">"c"</span>, <span class="fl">1</span>, <span class="st">"character"</span>, <span class="st">"Chromosome under analysis. Use X instead of chrX."</span>,
    <span class="st">"mcores"</span>, <span class="st">"m"</span>, <span class="fl">1</span>, <span class="st">"integer"</span>, <span class="st">"Number of cores"</span>,
    <span class="st">"verbose"</span>, <span class="st">"v"</span>, <span class="fl">2</span>, <span class="st">"logical"</span>, <span class="st">"Print status updates"</span>,
    <span class="st">"help"</span>, <span class="st">"h"</span>, <span class="fl">0</span>, <span class="st">"logical"</span>, <span class="st">"Display help"</span>
), <span class="kw">byrow</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">ncol</span> <span class="kw">=</span> <span class="fl">5</span>)
<span class="no">opt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/getopt/man/getopt.html">getopt</a></span>(<span class="no">spec</span>)

<span class="co">## Testing the script</span>
<span class="no">test</span> <span class="kw">&lt;-</span> <span class="fl">FALSE</span>
<span class="kw">if</span> (<span class="no">test</span>) {
    <span class="co">## Speficy it using an interactive R session and testing</span>
    <span class="no">test</span> <span class="kw">&lt;-</span> <span class="fl">TRUE</span>
}

<span class="co">## Test values</span>
<span class="kw">if</span> (<span class="no">test</span>) {
    <span class="no">opt</span> <span class="kw">&lt;-</span> <span class="kw">NULL</span>
    <span class="no">opt</span>$<span class="no">DFfile</span> <span class="kw">&lt;-</span> <span class="st">"/ProjectDir/derCoverageInfo/chr21Cov.Rdata"</span>
    <span class="no">opt</span>$<span class="no">chr</span> <span class="kw">&lt;-</span> <span class="st">"21"</span>
    <span class="no">opt</span>$<span class="no">mcores</span> <span class="kw">&lt;-</span> <span class="fl">1</span>
    <span class="no">opt</span>$<span class="no">verbose</span> <span class="kw">&lt;-</span> <span class="kw">NULL</span>
}

<span class="co">## if help was asked for print a friendly message</span>
<span class="co">## and exit with a non-zero error code</span>
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span>(<span class="no">opt</span>$<span class="no">help</span>)) {
    <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/getopt/man/getopt.html">getopt</a></span>(<span class="no">spec</span>, <span class="kw">usage</span> <span class="kw">=</span> <span class="fl">TRUE</span>))
    <span class="fu"><a href="https://rdrr.io/r/base/quit.html">q</a></span>(<span class="kw">status</span> <span class="kw">=</span> <span class="fl">1</span>)
}

<span class="co">## Default value for verbose = TRUE</span>
<span class="kw">if</span> (<span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span>(<span class="no">opt</span>$<span class="no">verbose</span>)) <span class="no">opt</span>$<span class="no">verbose</span> <span class="kw">&lt;-</span> <span class="fl">TRUE</span>

<span class="kw">if</span> (<span class="no">opt</span>$<span class="no">verbose</span>) <span class="fu"><a href="https://rdrr.io/r/base/message.html">message</a></span>(<span class="st">"Loading Rdata file with the output from loadCoverage()"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="no">opt</span>$<span class="no">DFfile</span>)

<span class="co">## Make it easy to use the name later. Here I'm assuming the names were generated using output='auto' in loadCoverage()</span>
<span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/Vector-class-leftovers.html">eval</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/parse.html">parse</a></span>(<span class="kw">text</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"data &lt;- "</span>, <span class="st">"chr"</span>, <span class="no">opt</span>$<span class="no">chr</span>, <span class="st">"CovInfo"</span>)))
<span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/Vector-class-leftovers.html">eval</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/parse.html">parse</a></span>(<span class="kw">text</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"rm(chr"</span>, <span class="no">opt</span>$<span class="no">chr</span>, <span class="st">"CovInfo)"</span>)))

<span class="co">## Just for testing purposes</span>
<span class="kw">if</span> (<span class="no">test</span>) {
    <span class="no">tmp</span> <span class="kw">&lt;-</span> <span class="no">data</span>
    <span class="no">tmp</span>$<span class="no">coverage</span> <span class="kw">&lt;-</span> <span class="no">tmp</span>$<span class="no">coverage</span>[<span class="fl">1</span>:<span class="fl">1e6</span>, ]
    <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"IRanges"</span>)
    <span class="no">tmp</span>$<span class="no">position</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">tmp</span>$<span class="no">pos</span>)[<span class="fl">1e6</span> + <span class="fl">1</span>]:<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tmp</span>$<span class="no">pos</span>)] <span class="kw">&lt;-</span> <span class="fl">FALSE</span>
    <span class="no">data</span> <span class="kw">&lt;-</span> <span class="no">tmp</span>
}

<span class="co">## Load the models</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="st">"models.Rdata"</span>)

<span class="co">## Load group information</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="st">"groupInfo.Rdata"</span>)


<span class="co">## Run the analysis with lowMemDir</span>
<span class="fu"><a href="../reference/analyzeChr.html">analyzeChr</a></span>(
    <span class="kw">chr</span> <span class="kw">=</span> <span class="no">opt</span>$<span class="no">chr</span>, <span class="kw">coverageInfo</span> <span class="kw">=</span> <span class="no">data</span>, <span class="kw">models</span> <span class="kw">=</span> <span class="no">models</span>,
    <span class="kw">cutoffFstat</span> <span class="kw">=</span> <span class="fl">1e-06</span>, <span class="kw">cutoffType</span> <span class="kw">=</span> <span class="st">"theoretical"</span>, <span class="kw">nPermute</span> <span class="kw">=</span> <span class="fl">1000</span>,
    <span class="kw">seeds</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fl">1000</span>), <span class="kw">maxClusterGap</span> <span class="kw">=</span> <span class="fl">3000</span>, <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="no">groupInfo</span>,
    <span class="kw">subject</span> <span class="kw">=</span> <span class="st">"hg19"</span>, <span class="kw">mc.cores</span> <span class="kw">=</span> <span class="no">opt</span>$<span class="no">mcores</span>,
    <span class="kw">lowMemDir</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"chr"</span>, <span class="no">opt</span>$<span class="no">chr</span>), <span class="st">"chunksDir"</span>),
    <span class="kw">verbose</span> <span class="kw">=</span> <span class="no">opt</span>$<span class="no">verbose</span>, <span class="kw">chunksize</span> <span class="kw">=</span> <span class="fl">1e5</span>
)

<span class="co">## Done</span>
<span class="kw">if</span> (<span class="no">opt</span>$<span class="no">verbose</span>) {
    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/proc.time.html">proc.time</a></span>())
    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>(), <span class="kw">locale</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
}</pre></body></html></div>
<p>Remember to modify the the script to fit your project.</p>
</div>
<div id="deranalysis-sh" class="section level3">
<h3 class="hasAnchor">
<a href="#deranalysis-sh" class="anchor"></a>derAnalysis.sh</h3>
<div class="sourceCode" id="cb421"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb421-1" data-line-number="1"><span class="co">#!/bin/sh</span></a>
<a class="sourceLine" id="cb421-2" data-line-number="2"></a>
<a class="sourceLine" id="cb421-3" data-line-number="3"><span class="co">## Usage</span></a>
<a class="sourceLine" id="cb421-4" data-line-number="4"><span class="co"># sh derAnalysis.sh analysis01</span></a>
<a class="sourceLine" id="cb421-5" data-line-number="5"></a>
<a class="sourceLine" id="cb421-6" data-line-number="6"><span class="co"># Directories</span></a>
<a class="sourceLine" id="cb421-7" data-line-number="7"><span class="va">MAINDIR=</span>/ProjectDir</a>
<a class="sourceLine" id="cb421-8" data-line-number="8"><span class="va">WDIR=${MAINDIR}</span>/derAnalysis</a>
<a class="sourceLine" id="cb421-9" data-line-number="9"><span class="va">DATADIR=${MAINDIR}</span>/derCoverageInfo</a>
<a class="sourceLine" id="cb421-10" data-line-number="10"></a>
<a class="sourceLine" id="cb421-11" data-line-number="11"><span class="co"># Define variables</span></a>
<a class="sourceLine" id="cb421-12" data-line-number="12"><span class="va">SHORT=</span><span class="st">'derA-01'</span></a>
<a class="sourceLine" id="cb421-13" data-line-number="13"><span class="va">PREFIX=$1</span></a>
<a class="sourceLine" id="cb421-14" data-line-number="14"></a>
<a class="sourceLine" id="cb421-15" data-line-number="15"><span class="co"># Construct shell files</span></a>
<a class="sourceLine" id="cb421-16" data-line-number="16"><span class="kw">for</span> <span class="ex">chrnum</span> in 22 21 Y 20 19 18 17 16 15 14 13 12 11 10 9 8 X 7 6 5 4 3 2 1</a>
<a class="sourceLine" id="cb421-17" data-line-number="17"><span class="kw">do</span></a>
<a class="sourceLine" id="cb421-18" data-line-number="18">    <span class="bu">echo</span> <span class="st">"Creating script for chromosome </span><span class="va">${chrnum}</span><span class="st">"</span></a>
<a class="sourceLine" id="cb421-19" data-line-number="19">    </a>
<a class="sourceLine" id="cb421-20" data-line-number="20">    <span class="kw">if [[</span> <span class="va">${chrnum}</span> <span class="ot">==</span> <span class="st">"Y"</span><span class="kw"> ]]</span></a>
<a class="sourceLine" id="cb421-21" data-line-number="21">    <span class="kw">then</span></a>
<a class="sourceLine" id="cb421-22" data-line-number="22">        <span class="va">CORES=</span>2</a>
<a class="sourceLine" id="cb421-23" data-line-number="23">    <span class="kw">else</span></a>
<a class="sourceLine" id="cb421-24" data-line-number="24">        <span class="va">CORES=</span>6</a>
<a class="sourceLine" id="cb421-25" data-line-number="25">    <span class="kw">fi</span></a>
<a class="sourceLine" id="cb421-26" data-line-number="26">    </a>
<a class="sourceLine" id="cb421-27" data-line-number="27">    <span class="va">chr=</span><span class="st">"chr</span><span class="va">${chrnum}</span><span class="st">"</span></a>
<a class="sourceLine" id="cb421-28" data-line-number="28">    <span class="va">outdir=</span><span class="st">"</span><span class="va">${PREFIX}</span><span class="st">/</span><span class="va">${chr}</span><span class="st">"</span></a>
<a class="sourceLine" id="cb421-29" data-line-number="29">    <span class="va">sname=</span><span class="st">"</span><span class="va">${SHORT}</span><span class="st">.</span><span class="va">${PREFIX}</span><span class="st">.</span><span class="va">${chr}</span><span class="st">"</span></a>
<a class="sourceLine" id="cb421-30" data-line-number="30">    <span class="fu">cat</span> <span class="op">&gt;</span> <span class="va">${WDIR}</span>/.<span class="va">${sname}</span>.sh <span class="op">&lt;&lt;EOF</span></a>
<a class="sourceLine" id="cb421-31" data-line-number="31">#!/bin/bash</a>
<a class="sourceLine" id="cb421-32" data-line-number="32">#$ -cwd</a>
<a class="sourceLine" id="cb421-33" data-line-number="33">#$ -m e</a>
<a class="sourceLine" id="cb421-34" data-line-number="34">#$ -l mem_free=8G,h_vmem=10G,h_fsize=10G</a>
<a class="sourceLine" id="cb421-35" data-line-number="35">#$ -N <span class="va">${sname}</span></a>
<a class="sourceLine" id="cb421-36" data-line-number="36">#$ -pe local <span class="va">${CORES}</span></a>
<a class="sourceLine" id="cb421-37" data-line-number="37"></a>
<a class="sourceLine" id="cb421-38" data-line-number="38">echo "**** Job starts ****"</a>
<a class="sourceLine" id="cb421-39" data-line-number="39">date</a>
<a class="sourceLine" id="cb421-40" data-line-number="40"></a>
<a class="sourceLine" id="cb421-41" data-line-number="41"># Create output directory </a>
<a class="sourceLine" id="cb421-42" data-line-number="42">mkdir -p <span class="va">${WDIR}</span>/<span class="va">${outdir}</span></a>
<a class="sourceLine" id="cb421-43" data-line-number="43"># Make logs directory</a>
<a class="sourceLine" id="cb421-44" data-line-number="44">mkdir -p <span class="va">${WDIR}</span>/<span class="va">${outdir}</span>/logs</a>
<a class="sourceLine" id="cb421-45" data-line-number="45"></a>
<a class="sourceLine" id="cb421-46" data-line-number="46"># run derfinder-analysis.R</a>
<a class="sourceLine" id="cb421-47" data-line-number="47">cd <span class="va">${WDIR}</span>/<span class="va">${PREFIX}</span>/</a>
<a class="sourceLine" id="cb421-48" data-line-number="48"></a>
<a class="sourceLine" id="cb421-49" data-line-number="49"># specific to our cluster</a>
<a class="sourceLine" id="cb421-50" data-line-number="50"># see http://www.jhpce.jhu.edu/knowledge-base/environment-modules/</a>
<a class="sourceLine" id="cb421-51" data-line-number="51">module load R/3.3.x</a>
<a class="sourceLine" id="cb421-52" data-line-number="52">Rscript <span class="va">${WDIR}</span>/derfinder-analysis.R -d "<span class="va">${DATADIR}</span>/<span class="va">${chr}</span>Cov.Rdata" -c "<span class="va">${chrnum}</span>" -m <span class="va">${CORES}</span> -v TRUE</a>
<a class="sourceLine" id="cb421-53" data-line-number="53"></a>
<a class="sourceLine" id="cb421-54" data-line-number="54"># Move log files into the logs directory</a>
<a class="sourceLine" id="cb421-55" data-line-number="55">mv <span class="va">${WDIR}</span>/<span class="va">${sname}</span>.* <span class="va">${WDIR}</span>/<span class="va">${outdir}</span>/logs/</a>
<a class="sourceLine" id="cb421-56" data-line-number="56"></a>
<a class="sourceLine" id="cb421-57" data-line-number="57">echo "**** Job ends ****"</a>
<a class="sourceLine" id="cb421-58" data-line-number="58">date</a>
<a class="sourceLine" id="cb421-59" data-line-number="59"><span class="op">EOF</span></a>
<a class="sourceLine" id="cb421-60" data-line-number="60">    <span class="va">call=</span><span class="st">"qsub .</span><span class="va">${sname}</span><span class="st">.sh"</span></a>
<a class="sourceLine" id="cb421-61" data-line-number="61">    <span class="va">$call</span></a>
<a class="sourceLine" id="cb421-62" data-line-number="62"><span class="kw">done</span></a></code></pre></div>
<p>Your cluster might specify memory requirements differently and you might need to use fewer or more cores depending on your data set.</p>
</div>
</div>
</div>
<div id="expressed-regions-level-projects" class="section level1">
<h1 class="hasAnchor">
<a href="#expressed-regions-level-projects" class="anchor"></a>Expressed regions-level projects</h1>
<p>If you are using <code><a href="../reference/regionMatrix.html">regionMatrix()</a></code>, an organization similar to the one we mentioned for the single base-level F-statistics implementation is recommended. If you have a very large number of samples, it might be best to store the coverage data separately by chromosome instead of a single <code>fullCov</code> object. This is help with the memory load since you can then write a script for loading the coverage data for the chromosome of interest and running <code><a href="../reference/regionMatrix.html">regionMatrix()</a></code> on it. Once all your jobs are done, merge the results from them.</p>
<p>If you are using <code><a href="../reference/railMatrix.html">railMatrix()</a></code>, you don’t need any fancy setup. You only need the paths to the summary files and the sample files, all of which are assumed to be BigWig files.</p>
</div>
<div id="summary" class="section level1">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
<p>We have illustrated how to identify candidate differentially expressed regions without using annotation by using <code><a href="../reference/analyzeChr.html">analyzeChr()</a></code>. Furthermore, we covered how to perform the expressed region-level matrix analysis with <code><a href="../reference/regionMatrix.html">regionMatrix()</a></code> or <code><a href="../reference/railMatrix.html">railMatrix()</a></code>. We also highlighted other uses of the <em><a href="https://bioconductor.org/packages/3.12/derfinder">derfinder</a></em> package.</p>
<p>Latter sections of this vignette covered the most commonly used advanced arguments, details on how to load data, flow charts explaining the relationships between the functions, the recommended output organization, and example shell scripts for running the analysis.</p>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>This package was made possible thanks to:</p>
<ul>
<li>R <a id="cite-R_Core_Team_2020"></a>(<a href="https://www.R-project.org/">R Core Team, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/AnnotationDbi">AnnotationDbi</a></em> <a id="cite-AnnotationDbi"></a>(<a href="https://doi.org/10.18129/B9.bioc.AnnotationDbi">Pagès, Carlson, Falcon, and Li, 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/BiocParallel">BiocParallel</a></em> <a id="cite-Morgan_2020"></a>(<a href="https://github.com/Bioconductor/BiocParallel">Morgan, Obenchain, Lang, Thompson, et al., 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/bumphunter">bumphunter</a></em> <a id="cite-bumphunter"></a>(<a href="https://doi.org/10.1093/ije/dyr238">Jaffe, Murakami, Lee, Leek, et al., 2012</a>) and <a id="cite-bumphunterPaper"></a>(<a href="#bib-bumphunterPaper">Jaffe, Murakami, Lee, Leek, et al., 2012</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/derfinderHelper">derfinderHelper</a></em></li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/GenomeInfoDb">GenomeInfoDb</a></em> <a id="cite-GenomeInfoDb"></a>(<a href="https://doi.org/10.18129/B9.bioc.GenomeInfoDb">Arora, Morgan, Carlson, and Pagès, 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/GenomicAlignments">GenomicAlignments</a></em> <a id="cite-Lawrence_2013"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/GenomicFeatures">GenomicFeatures</a></em> <a id="cite-Lawrence_2013a"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/GenomicFiles">GenomicFiles</a></em> <a id="cite-Bioconductor_Package_Maintainer_2020"></a>(<a href="#bib-Bioconductor_Package_Maintainer_2020">Bioconductor Package Maintainer, Obenchain, Love, Shepherd, et al., 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/GenomicRanges">GenomicRanges</a></em> <a id="cite-Lawrence_2013ab"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=Hmisc">Hmisc</a></em> <a id="cite-Harrell_Jr_2020"></a>(<a href="https://CRAN.R-project.org/package=Hmisc">Harrell Jr, Dupont, and others., 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/IRanges">IRanges</a></em> <a id="cite-Lawrence_2013abc"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/qvalue">qvalue</a></em> <a id="cite-Storey_2020"></a>(<a href="http://github.com/jdstorey/qvalue">Storey, Bass, Dabney, and Robinson, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/Rsamtools">Rsamtools</a></em> <a id="cite-Morgan_2020a"></a>(<a href="http://bioconductor.org/packages/Rsamtools">Morgan, Pagès, Obenchain, and Hayden, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/rtracklayer">rtracklayer</a></em> <a id="cite-Lawrence_2009"></a>(<a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract">Lawrence, Gentleman, and Carey, 2009</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/S4Vectors">S4Vectors</a></em> <a id="cite-S4Vectors"></a>(<a href="https://doi.org/10.18129/B9.bioc.S4Vectors">Pagès, Lawrence, and Aboyoun, 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/derfinderData">derfinderData</a></em> (<a href="https://github.com/leekgroup/derfinderData">Collado-Torres, Jaffe, and Leek, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> <a id="cite-Csardi_2018"></a>(<a href="https://CRAN.R-project.org/package=sessioninfo">Csárdi, core, Wickham, Chang, et al., 2018</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a></em> <a id="cite-Wickham_2016"></a>(<a href="https://ggplot2.tidyverse.org">Wickham, 2016</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> <a id="cite-Boettiger_2019"></a>(<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> <a id="cite-Xie_2014"></a>(<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> <a id="cite-Oles_2020"></a>(<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> <a id="cite-Allaire_2020"></a>(<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat">testthat</a></em> <a id="cite-Wickham_2011"></a>(<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf">Wickham, 2011</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/TxDb.Hsapiens.UCSC.hg19.knownGene">TxDb.Hsapiens.UCSC.hg19.knownGene</a></em> <a id="cite-Carlson_2015"></a>(<a href="#bib-Carlson_2015">Carlson and Maintainer, 2015</a>)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb422"><html><body><pre class="r"><span class="co">## Create the vignette</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"rmarkdown"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"derfinder-users-guide.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span>))

<span class="co">## Extract the R code</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"knitr"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span>(<span class="st">"derfinder-users-guide.Rmd"</span>, <span class="kw">tangle</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb423"><html><body><pre class="r"><span class="co">## Clean up</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span>(<span class="st">"derfinderUsersGuideRef.bib"</span>)</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb425"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span>(<span class="st">"analysisResults"</span>, <span class="kw">recursive</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span>(<span class="st">"HSB113.bw"</span>)</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb427"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span>(<span class="st">"meanChr21.bw"</span>)</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<p>Date the vignette was generated.</p>
<pre><code>## [1] "2020-05-14 15:57:47 UTC"</code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code>## Time difference of 1.825 mins</code></pre>
<p><code>R</code> session information.</p>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value                                      
##  version  R version 4.0.0 Patched (2020-05-07 r78385)
##  os       Ubuntu 18.04.4 LTS                         
##  system   x86_64, linux-gnu                          
##  ui       X11                                        
##  language (EN)                                       
##  collate  en_US.UTF-8                                
##  ctype    en_US.UTF-8                                
##  tz       UTC                                        
##  date     2020-05-14                                 
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package                           * version    date       lib source                        
##  acepack                             1.4.1      2016-10-29 [1] CRAN (R 4.0.0)                
##  annotate                            1.67.0     2020-04-27 [1] Bioconductor                  
##  AnnotationDbi                     * 1.51.0     2020-04-27 [1] Bioconductor                  
##  askpass                             1.1        2019-01-13 [2] CRAN (R 4.0.0)                
##  assertthat                          0.2.1      2019-03-21 [2] CRAN (R 4.0.0)                
##  backports                           1.1.7      2020-05-13 [1] CRAN (R 4.0.0)                
##  base64enc                           0.1-3      2015-07-28 [1] CRAN (R 4.0.0)                
##  bibtex                              0.4.2.2    2020-01-02 [1] CRAN (R 4.0.0)                
##  Biobase                           * 2.49.0     2020-04-27 [1] Bioconductor                  
##  BiocFileCache                       1.13.0     2020-04-27 [1] Bioconductor                  
##  BiocGenerics                      * 0.35.2     2020-05-06 [1] Bioconductor                  
##  BiocManager                         1.30.10    2019-11-16 [2] CRAN (R 4.0.0)                
##  BiocParallel                        1.23.0     2020-04-27 [1] Bioconductor                  
##  BiocStyle                         * 2.17.0     2020-04-27 [1] Bioconductor                  
##  biomaRt                             2.45.0     2020-04-27 [1] Bioconductor                  
##  Biostrings                          2.57.0     2020-04-27 [1] Bioconductor                  
##  bit                                 1.1-15.2   2020-02-10 [1] CRAN (R 4.0.0)                
##  bit64                               0.9-7      2017-05-08 [1] CRAN (R 4.0.0)                
##  bitops                              1.0-6      2013-08-17 [1] CRAN (R 4.0.0)                
##  blob                                1.2.1      2020-01-20 [1] CRAN (R 4.0.0)                
##  bookdown                            0.18       2020-03-05 [1] CRAN (R 4.0.0)                
##  BSgenome                            1.57.0     2020-04-27 [1] Bioconductor                  
##  bumphunter                        * 1.31.0     2020-04-28 [1] Bioconductor                  
##  checkmate                           2.0.0      2020-02-06 [1] CRAN (R 4.0.0)                
##  cli                                 2.0.2      2020-02-28 [2] CRAN (R 4.0.0)                
##  cluster                             2.1.0      2019-06-19 [3] CRAN (R 4.0.0)                
##  codetools                           0.2-16     2018-12-24 [3] CRAN (R 4.0.0)                
##  colorspace                          1.4-1      2019-03-18 [1] CRAN (R 4.0.0)                
##  crayon                              1.3.4      2017-09-16 [2] CRAN (R 4.0.0)                
##  curl                                4.3        2019-12-02 [2] CRAN (R 4.0.0)                
##  data.table                          1.12.8     2019-12-09 [1] CRAN (R 4.0.0)                
##  DBI                                 1.1.0      2019-12-15 [1] CRAN (R 4.0.0)                
##  dbplyr                              1.4.3      2020-04-19 [1] CRAN (R 4.0.0)                
##  DelayedArray                      * 0.15.1     2020-05-01 [1] Bioconductor                  
##  derfinder                         * 1.23.1     2020-05-14 [1] Bioconductor                  
##  derfinderData                     * 2.7.1      2020-05-11 [1] Bioconductor                  
##  derfinderHelper                     1.23.1     2020-05-10 [1] Bioconductor                  
##  desc                                1.2.0      2018-05-01 [2] CRAN (R 4.0.0)                
##  DESeq2                            * 1.29.1     2020-05-12 [1] Bioconductor                  
##  digest                              0.6.25     2020-02-23 [2] CRAN (R 4.0.0)                
##  doRNG                               1.8.2      2020-01-27 [1] CRAN (R 4.0.0)                
##  dplyr                               0.8.5      2020-03-07 [1] CRAN (R 4.0.0)                
##  ellipsis                            0.3.0      2019-09-20 [2] CRAN (R 4.0.0)                
##  evaluate                            0.14       2019-05-28 [2] CRAN (R 4.0.0)                
##  fansi                               0.4.1      2020-01-08 [2] CRAN (R 4.0.0)                
##  farver                              2.0.3      2020-01-16 [1] CRAN (R 4.0.0)                
##  foreach                           * 1.5.0      2020-03-30 [1] CRAN (R 4.0.0)                
##  foreign                             0.8-79     2020-04-26 [3] CRAN (R 4.0.0)                
##  Formula                             1.2-3      2018-05-03 [1] CRAN (R 4.0.0)                
##  fs                                  1.4.1      2020-04-04 [2] CRAN (R 4.0.0)                
##  genefilter                          1.71.0     2020-04-27 [1] Bioconductor                  
##  geneplotter                         1.67.0     2020-04-27 [1] Bioconductor                  
##  generics                            0.0.2      2018-11-29 [1] CRAN (R 4.0.0)                
##  GenomeInfoDb                      * 1.25.0     2020-04-27 [1] Bioconductor                  
##  GenomeInfoDbData                    1.2.3      2020-05-10 [1] Bioconductor                  
##  GenomicAlignments                   1.25.0     2020-04-27 [1] Bioconductor                  
##  GenomicFeatures                   * 1.41.0     2020-04-27 [1] Bioconductor                  
##  GenomicFiles                        1.25.0     2020-04-27 [1] Bioconductor                  
##  GenomicRanges                     * 1.41.1     2020-05-02 [1] Bioconductor                  
##  ggplot2                           * 3.3.0      2020-03-05 [1] CRAN (R 4.0.0)                
##  glue                                1.4.1      2020-05-13 [1] CRAN (R 4.0.0)                
##  gridExtra                           2.3        2017-09-09 [1] CRAN (R 4.0.0)                
##  gtable                              0.3.0      2019-03-25 [1] CRAN (R 4.0.0)                
##  highr                               0.8        2019-03-20 [2] CRAN (R 4.0.0)                
##  Hmisc                               4.4-0      2020-03-23 [1] CRAN (R 4.0.0)                
##  hms                                 0.5.3      2020-01-08 [1] CRAN (R 4.0.0)                
##  htmlTable                           1.13.3     2019-12-04 [1] CRAN (R 4.0.0)                
##  htmltools                           0.4.0      2019-10-04 [2] CRAN (R 4.0.0)                
##  htmlwidgets                         1.5.1      2019-10-08 [2] CRAN (R 4.0.0)                
##  httr                                1.4.1      2019-08-05 [2] CRAN (R 4.0.0)                
##  IRanges                           * 2.23.4     2020-05-03 [1] Bioconductor                  
##  iterators                         * 1.0.12     2019-07-26 [1] CRAN (R 4.0.0)                
##  jpeg                                0.1-8.1    2019-10-24 [1] CRAN (R 4.0.0)                
##  jsonlite                            1.6.1      2020-02-02 [2] CRAN (R 4.0.0)                
##  knitcitations                     * 1.0.10     2019-09-15 [1] CRAN (R 4.0.0)                
##  knitr                             * 1.28       2020-02-06 [2] CRAN (R 4.0.0)                
##  labeling                            0.3        2014-08-23 [1] CRAN (R 4.0.0)                
##  lattice                             0.20-41    2020-04-02 [3] CRAN (R 4.0.0)                
##  latticeExtra                        0.6-29     2019-12-19 [1] CRAN (R 4.0.0)                
##  lifecycle                           0.2.0      2020-03-06 [2] CRAN (R 4.0.0)                
##  limma                             * 3.45.0     2020-04-27 [1] Bioconductor                  
##  locfit                            * 1.5-9.4    2020-03-25 [1] CRAN (R 4.0.0)                
##  lubridate                           1.7.8      2020-04-06 [1] CRAN (R 4.0.0)                
##  magrittr                            1.5        2014-11-22 [2] CRAN (R 4.0.0)                
##  MASS                                7.3-51.6   2020-04-26 [3] CRAN (R 4.0.0)                
##  Matrix                              1.2-18     2019-11-27 [3] CRAN (R 4.0.0)                
##  matrixStats                       * 0.56.0     2020-03-13 [1] CRAN (R 4.0.0)                
##  memoise                             1.1.0      2017-04-21 [2] CRAN (R 4.0.0)                
##  munsell                             0.5.0      2018-06-12 [1] CRAN (R 4.0.0)                
##  nnet                                7.3-14     2020-04-26 [3] CRAN (R 4.0.0)                
##  openssl                             1.4.1      2019-07-18 [2] CRAN (R 4.0.0)                
##  pillar                              1.4.4      2020-05-05 [2] CRAN (R 4.0.0)                
##  pkgconfig                           2.0.3      2019-09-22 [2] CRAN (R 4.0.0)                
##  pkgdown                             1.5.1.9000 2020-05-10 [1] Github (r-lib/pkgdown@c181e5c)
##  plyr                                1.8.6      2020-03-03 [1] CRAN (R 4.0.0)                
##  png                                 0.1-7      2013-12-03 [1] CRAN (R 4.0.0)                
##  prettyunits                         1.1.1      2020-01-24 [2] CRAN (R 4.0.0)                
##  progress                            1.2.2      2019-05-16 [1] CRAN (R 4.0.0)                
##  purrr                               0.3.4      2020-04-17 [2] CRAN (R 4.0.0)                
##  qvalue                              2.21.0     2020-04-27 [1] Bioconductor                  
##  R6                                  2.4.1      2019-11-12 [2] CRAN (R 4.0.0)                
##  rappdirs                            0.3.1      2016-03-28 [1] CRAN (R 4.0.0)                
##  RColorBrewer                        1.1-2      2014-12-07 [1] CRAN (R 4.0.0)                
##  Rcpp                                1.0.4.6    2020-04-09 [2] CRAN (R 4.0.0)                
##  RCurl                               1.98-1.2   2020-04-18 [1] CRAN (R 4.0.0)                
##  RefManageR                          1.2.12     2019-04-03 [1] CRAN (R 4.0.0)                
##  reshape2                            1.4.4      2020-04-09 [1] CRAN (R 4.0.0)                
##  rlang                               0.4.6      2020-05-02 [2] CRAN (R 4.0.0)                
##  rmarkdown                           2.1        2020-01-20 [1] CRAN (R 4.0.0)                
##  rngtools                            1.5        2020-01-23 [1] CRAN (R 4.0.0)                
##  rpart                               4.1-15     2019-04-12 [3] CRAN (R 4.0.0)                
##  rprojroot                           1.3-2      2018-01-03 [2] CRAN (R 4.0.0)                
##  Rsamtools                           2.5.0      2020-04-27 [1] Bioconductor                  
##  RSQLite                             2.2.0      2020-01-07 [1] CRAN (R 4.0.0)                
##  rstudioapi                          0.11       2020-02-07 [2] CRAN (R 4.0.0)                
##  rtracklayer                         1.49.1     2020-05-02 [1] Bioconductor                  
##  S4Vectors                         * 0.27.5     2020-05-06 [1] Bioconductor                  
##  scales                              1.1.1      2020-05-11 [1] CRAN (R 4.0.0)                
##  sessioninfo                       * 1.1.1      2018-11-05 [2] CRAN (R 4.0.0)                
##  stringi                             1.4.6      2020-02-17 [2] CRAN (R 4.0.0)                
##  stringr                             1.4.0      2019-02-10 [2] CRAN (R 4.0.0)                
##  SummarizedExperiment              * 1.19.2     2020-05-01 [1] Bioconductor                  
##  survival                            3.1-12     2020-04-10 [3] CRAN (R 4.0.0)                
##  tibble                              3.0.1      2020-04-20 [2] CRAN (R 4.0.0)                
##  tidyselect                          1.1.0      2020-05-11 [1] CRAN (R 4.0.0)                
##  TxDb.Hsapiens.UCSC.hg19.knownGene * 3.2.2      2020-05-10 [1] Bioconductor                  
##  VariantAnnotation                   1.35.0     2020-04-27 [1] Bioconductor                  
##  vctrs                               0.3.0      2020-05-11 [1] CRAN (R 4.0.0)                
##  withr                               2.2.0      2020-04-20 [2] CRAN (R 4.0.0)                
##  xfun                                0.13       2020-04-13 [2] CRAN (R 4.0.0)                
##  XML                                 3.99-0.3   2020-01-20 [1] CRAN (R 4.0.0)                
##  xml2                                1.3.2      2020-04-23 [2] CRAN (R 4.0.0)                
##  xtable                              1.8-4      2019-04-21 [1] CRAN (R 4.0.0)                
##  XVector                             0.29.0     2020-04-27 [1] Bioconductor                  
##  yaml                                2.2.1      2020-02-01 [2] CRAN (R 4.0.0)                
##  zlibbioc                            1.35.0     2020-04-27 [1] Bioconductor                  
## 
## [1] /usr/local/lib/R/host-site-library
## [2] /usr/local/lib/R/site-library
## [3] /usr/local/lib/R/library</code></pre>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> (<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>) with <em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>) and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>).</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, J. Luraschi, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version 2.1. 2020. &lt;URL: <a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] S. Arora, M. Morgan, M. Carlson, and H. Pagès. <em>GenomeInfoDb: Utilities for manipulating chromosome and other ‘seqname’ identifiers</em>. 2017. DOI: 10.18129/B9.bioc.GenomeInfoDb.</p>
<p>[3] Bioconductor Package Maintainer, V. Obenchain, M. Love, L. Shepherd, et al. <em>GenomicFiles: Distributed computing by file or by range</em>. R package version 1.25.0. 2020.</p>
<p>[4] C. Boettiger. <em>knitcitations: Citations for ‘Knitr’ Markdown Files</em>. R package version 1.0.10. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=knitcitations" class="uri">https://CRAN.R-project.org/package=knitcitations</a>&gt;.</p>
<p>[5] BrainSpan. “Atlas of the Developing Human Brain [Internet]. Funded by ARRA Awards 1RC2MH089921-01, 1RC2MH090047-01, and 1RC2MH089929-01.” 2011. &lt;URL: <a href="http://www.brainspan.org/" class="uri">http://www.brainspan.org/</a>&gt;.</p>
<p>[6] M. Carlson and B. P. Maintainer. <em>TxDb.Hsapiens.UCSC.hg19.knownGene: Annotation package for TxDb object(s)</em>. R package version 3.2.2. 2015.</p>
<p>[7] L. Collado-Torres, A. Jaffe, and J. Leek. <em>derfinderData: Processed BigWigs from BrainSpan for examples</em>. R package version 2.7.1. 2020. &lt;URL: <a href="https://github.com/leekgroup/derfinderData" class="uri">https://github.com/leekgroup/derfinderData</a>&gt;.</p>
<p>[8] G. Csárdi, R. core, H. Wickham, W. Chang, et al. <em>sessioninfo: R Session Information</em>. R package version 1.1.1. 2018. &lt;URL: <a href="https://CRAN.R-project.org/package=sessioninfo" class="uri">https://CRAN.R-project.org/package=sessioninfo</a>&gt;.</p>
<p>[9] A. C. Frazee, S. Sabunciyan, K. D. Hansen, R. A. Irizarry, et al. “Differential expression analysis of RNA-seq data at single-base resolution”. In: <em>Biostatistics</em> 15 (3) (2014), pp. 413-426. DOI: 10.1093/biostatistics/kxt053. &lt;URL: <a href="http://biostatistics.oxfordjournals.org/content/15/3/413.long" class="uri">http://biostatistics.oxfordjournals.org/content/15/3/413.long</a>&gt;.</p>
<p>[10] F. E. Harrell Jr, w. c. f. C. Dupont, and m. others. <em>Hmisc: Harrell Miscellaneous</em>. R package version 4.4-0. 2020. &lt;URL: <a href="https://CRAN.R-project.org/package=Hmisc" class="uri">https://CRAN.R-project.org/package=Hmisc</a>&gt;.</p>
<p>[11] A. E. Jaffe, P. Murakami, H. Lee, J. T. Leek, et al. “Bump hunting to identify differentially methylated regions in epigenetic epidemiology studies”. In: <em>International journal of epidemiology</em> 41.1 (2012), pp. 200-209. DOI: 10.1093/ije/dyr238.</p>
<p>[12] A. E. Jaffe, P. Murakami, H. Lee, J. T. Leek, et al. “Bump hunting to identify differentially methylated regions in epigenetic epidemiology studies”. In: <em>International Journal of Epidemiology</em> (2012).</p>
<p>[13] M. Lawrence, R. Gentleman, and V. Carey. “rtracklayer: an R package for interfacing with genome browsers”. In: <em>Bioinformatics</em> 25 (2009), pp. 1841-1842. DOI: 10.1093/bioinformatics/btp328. &lt;URL: <a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract" class="uri">http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract</a>&gt;.</p>
<p>[14] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[15] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[16] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[17] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[18] M. Morgan, V. Obenchain, M. Lang, R. Thompson, et al. <em>BiocParallel: Bioconductor facilities for parallel evaluation</em>. R package version 1.23.0. 2020. &lt;URL: <a href="https://github.com/Bioconductor/BiocParallel" class="uri">https://github.com/Bioconductor/BiocParallel</a>&gt;.</p>
<p>[19] M. Morgan, H. Pagès, V. Obenchain, and N. Hayden. <em>Rsamtools: Binary alignment (BAM), FASTA, variant call (BCF), and tabix file import</em>. R package version 2.5.0. 2020. &lt;URL: <a href="http://bioconductor.org/packages/Rsamtools" class="uri">http://bioconductor.org/packages/Rsamtools</a>&gt;.</p>
<p>[20] A. Oleś, M. Morgan, and W. Huber. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.17.0. 2020. &lt;URL: <a href="https://github.com/Bioconductor/BiocStyle" class="uri">https://github.com/Bioconductor/BiocStyle</a>&gt;.</p>
<p>[21] H. Pagès, M. Carlson, S. Falcon, and N. Li. <em>AnnotationDbi: Annotation Database Interface</em>. 2017. DOI: 10.18129/B9.bioc.AnnotationDbi.</p>
<p>[22] H. Pagès, M. Lawrence, and P. Aboyoun. <em>S4Vectors: S4 implementation of vector-like and list-like objects</em>. 2017. DOI: 10.18129/B9.bioc.S4Vectors.</p>
<p>[23] R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2020. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[24] J. D. Storey, A. J. Bass, A. Dabney, and D. Robinson. <em>qvalue: Q-value estimation for false discovery rate control</em>. R package version 2.21.0. 2020. &lt;URL: <a href="http://github.com/jdstorey/qvalue" class="uri">http://github.com/jdstorey/qvalue</a>&gt;.</p>
<p>[25] H. Wickham. <em>ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York, 2016. ISBN: 978-3-319-24277-4. &lt;URL: <a href="https://ggplot2.tidyverse.org" class="uri">https://ggplot2.tidyverse.org</a>&gt;.</p>
<p>[26] H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5-10. &lt;URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="uri">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>&gt;.</p>
<p>[27] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V. Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and Hall/CRC, 2014. &lt;URL: <a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
